{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pybedtools as pbt\n",
    "import pandas as pd\n",
    "import seaborn as sns\n",
    "import glob\n",
    "import matplotlib.pyplot as plt\n",
    "import numpy as np"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "def merge_peaks(peaks, n):\n",
    "    return peaks.sort().merge(d=n, s=True, c=[4, 5, 6], o=['distinct', 'sum', 'distinct'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "set1_2_merged30_150motifs = pbt.BedTool('/home/aram/Dropbox (The Francis Crick)/ucl/aram/martina_tdp43/binding_sites_set_1and2/genome_larger_motif_groups/set1_2_merged30_150motifs.bed')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "intersect_merged30_all = set1_2_merged30_150motifs.intersect(pbt.BedTool(\n",
    "    '/home/aram/data/martina_tdp/thresholded_xl_p05/all_sites_merged_sets_1_2.bed.gz'), \n",
    "        s=True,\n",
    "        wb=True,\n",
    "        nonamecheck=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_intersect_merged30_all = intersect_merged30_all.to_dataframe()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_out_all = pd.DataFrame()\n",
    "df_out_all['chrome'] = df_intersect_merged30_all.chrom\n",
    "df_out_all['start'] = df_intersect_merged30_all.thickEnd\n",
    "df_out_all['end'] = df_intersect_merged30_all.itemRgb\n",
    "df_out_all['name'] = df_intersect_merged30_all.name\n",
    "df_out_all['score'] = df_intersect_merged30_all.blockSizes\n",
    "df_out_all['strand'] = df_intersect_merged30_all.strand"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>name</th>\n",
       "      <th>score</th>\n",
       "      <th>strand</th>\n",
       "      <th>thickStart</th>\n",
       "      <th>thickEnd</th>\n",
       "      <th>itemRgb</th>\n",
       "      <th>blockCount</th>\n",
       "      <th>blockSizes</th>\n",
       "      <th>blockStarts</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GL000008.2</td>\n",
       "      <td>118900</td>\n",
       "      <td>118901</td>\n",
       "      <td>.</td>\n",
       "      <td>16</td>\n",
       "      <td>-</td>\n",
       "      <td>GL000008.2</td>\n",
       "      <td>118900</td>\n",
       "      <td>118901</td>\n",
       "      <td>.</td>\n",
       "      <td>2</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GL000008.2</td>\n",
       "      <td>131468</td>\n",
       "      <td>131469</td>\n",
       "      <td>.</td>\n",
       "      <td>30</td>\n",
       "      <td>+</td>\n",
       "      <td>GL000008.2</td>\n",
       "      <td>131468</td>\n",
       "      <td>131469</td>\n",
       "      <td>.</td>\n",
       "      <td>1</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GL000008.2</td>\n",
       "      <td>131470</td>\n",
       "      <td>131471</td>\n",
       "      <td>.</td>\n",
       "      <td>30</td>\n",
       "      <td>+</td>\n",
       "      <td>GL000008.2</td>\n",
       "      <td>131470</td>\n",
       "      <td>131471</td>\n",
       "      <td>.</td>\n",
       "      <td>6</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GL000008.2</td>\n",
       "      <td>131471</td>\n",
       "      <td>131472</td>\n",
       "      <td>.</td>\n",
       "      <td>30</td>\n",
       "      <td>+</td>\n",
       "      <td>GL000008.2</td>\n",
       "      <td>131471</td>\n",
       "      <td>131472</td>\n",
       "      <td>.</td>\n",
       "      <td>5</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GL000008.2</td>\n",
       "      <td>131472</td>\n",
       "      <td>131473</td>\n",
       "      <td>.</td>\n",
       "      <td>30</td>\n",
       "      <td>+</td>\n",
       "      <td>GL000008.2</td>\n",
       "      <td>131472</td>\n",
       "      <td>131473</td>\n",
       "      <td>.</td>\n",
       "      <td>3</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        chrom   start     end name  score strand  thickStart  thickEnd  \\\n",
       "0  GL000008.2  118900  118901    .     16      -  GL000008.2    118900   \n",
       "1  GL000008.2  131468  131469    .     30      +  GL000008.2    131468   \n",
       "2  GL000008.2  131470  131471    .     30      +  GL000008.2    131470   \n",
       "3  GL000008.2  131471  131472    .     30      +  GL000008.2    131471   \n",
       "4  GL000008.2  131472  131473    .     30      +  GL000008.2    131472   \n",
       "\n",
       "   itemRgb blockCount  blockSizes blockStarts  \n",
       "0   118901          .           2           -  \n",
       "1   131469          .           1           +  \n",
       "2   131471          .           6           +  \n",
       "3   131472          .           5           +  \n",
       "4   131473          .           3           +  "
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_intersect_merged30_all.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "#df_intersect_merged30_all.to_csv('intersect_merged30_all.bed')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "bed_out_all = pbt.BedTool.from_dataframe(df_out_all)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "bed_out_all_unfiltered = merge_peaks(bed_out_all, 30)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_all_unfiltered = bed_out_all_unfiltered.to_dataframe()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "files2 = glob.glob('/home/aram/data/martina_tdp/second_big_heatmap/*')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "files1 = glob.glob('/home/aram/data/martina_tdp/test_idr_mutants/*')\n",
    "files1 = [f for f in files1 if '_peaks' not in f]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "files = files1 + files2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['/home/aram/data/martina_tdp/test_idr_mutants/tardbp-316del346-egfp-hek293-hd1-6-4-20180829-ju_trimmed_single.bed.gz',\n",
       " '/home/aram/data/martina_tdp/test_idr_mutants/tardbp-egfp-hek293-hd2-5-3-20180829-ju_trimmed_single.bed.gz',\n",
       " '/home/aram/data/martina_tdp/test_idr_mutants/tardbp-egfp-hek293-hd2-5-on-bead-rt-1-20180829-ju_trimmed_single.bed.gz',\n",
       " '/home/aram/data/martina_tdp/test_idr_mutants/tardbp-316del346-egfp-hek293-hd2-5-3-20180829-ju_trimmed_single.bed.gz',\n",
       " '/home/aram/data/martina_tdp/test_idr_mutants/tardbp-egfp-hek293-3-20181001-ju_trimmed_single.bed.gz',\n",
       " '/home/aram/data/martina_tdp/test_idr_mutants/tardbp-q331k-egfp-hek293-no-uv-1-20181001-ju_trimmed_single.bed.gz',\n",
       " '/home/aram/data/martina_tdp/test_idr_mutants/tardbp-320del414-egfp-hek293-hd1-6-on-bead-rt-1-20180829-ju_trimmed_single.bed.gz',\n",
       " '/home/aram/data/martina_tdp/test_idr_mutants/tardbp-316del346-egfp-hek293-4-20181001-ju_trimmed_single.bed.gz',\n",
       " '/home/aram/data/martina_tdp/test_idr_mutants/tardbp-egfp-hek293-no-uv-1-20180829-ju_trimmed_single.bed.gz',\n",
       " '/home/aram/data/martina_tdp/test_idr_mutants/tardbp-egfp-hek293-4-20181001-ju_trimmed_single.bed.gz',\n",
       " '/home/aram/data/martina_tdp/test_idr_mutants/tardbp-egfp-hek293-hd1-6-4-20180829-ju_trimmed_single.bed.gz',\n",
       " '/home/aram/data/martina_tdp/test_idr_mutants/tardbp-a326p-egfp-hek293-4-20181001-ju_trimmed_single.bed.gz',\n",
       " '/home/aram/data/martina_tdp/test_idr_mutants/tardbp-egfp-hek293-hd1-6-3-20180829-ju_trimmed_single.bed.gz',\n",
       " '/home/aram/data/martina_tdp/test_idr_mutants/tardbp-320del414-egfp-hek293-hd1-6-4-20180829-ju_trimmed_single.bed.gz',\n",
       " '/home/aram/data/martina_tdp/test_idr_mutants/tardbp-q331k-egfp-hek293-3-20181001-ju_trimmed_single.bed.gz',\n",
       " '/home/aram/data/martina_tdp/test_idr_mutants/tardbp-316del346-egfp-hek293-hd1-6-3-20180829-ju_trimmed_single.bed.gz',\n",
       " '/home/aram/data/martina_tdp/test_idr_mutants/tardbp-320del414-egfp-hek293-hd1-6-3-20180829-ju_trimmed_single.bed.gz',\n",
       " '/home/aram/data/martina_tdp/test_idr_mutants/tardbp-egfp-hek293-hd1-6-on-bead-rt-1-20180829-ju_trimmed_single.bed.gz',\n",
       " '/home/aram/data/martina_tdp/test_idr_mutants/tardbp-316del346-egfp-hek293-hd1-6-on-bead-rt-1-20180829-ju_trimmed_single.bed.gz',\n",
       " '/home/aram/data/martina_tdp/test_idr_mutants/tardbp-m337p-egfp-hek293-4-20181001-ju_trimmed_single.bed.gz',\n",
       " '/home/aram/data/martina_tdp/test_idr_mutants/tardbp-m337p-egfp-hek293-3-20181001-ju_trimmed_single.bed.gz',\n",
       " '/home/aram/data/martina_tdp/test_idr_mutants/tardbp-a326p-egfp-hek293-3-20181001-ju_trimmed_single.bed.gz',\n",
       " '/home/aram/data/martina_tdp/test_idr_mutants/tardbp-316del346-egfp-hek293-hd2-5-4-20180829-ju_trimmed_single.bed.gz',\n",
       " '/home/aram/data/martina_tdp/test_idr_mutants/tardbp-q331k-egfp-hek293-4-20181001-ju_trimmed_single.bed.gz',\n",
       " '/home/aram/data/martina_tdp/test_idr_mutants/tardbp-g335a-egfp-hek293-3-20181001-ju_trimmed_single.bed.gz',\n",
       " '/home/aram/data/martina_tdp/test_idr_mutants/tardbp-g335a-egfp-hek293-4-20181001-ju_trimmed_single.bed.gz',\n",
       " '/home/aram/data/martina_tdp/test_idr_mutants/tardbp-g294a-egfp-hek293-3-20181001-ju_trimmed_single.bed.gz',\n",
       " '/home/aram/data/martina_tdp/test_idr_mutants/tardbp-316del346-egfp-hek293-hd2-5-on-bead-rt-1-20180829-ju_trimmed_single.bed.gz',\n",
       " '/home/aram/data/martina_tdp/test_idr_mutants/tardbp-316del346-egfp-hek293-3-20181001-ju_trimmed_single.bed.gz',\n",
       " '/home/aram/data/martina_tdp/test_idr_mutants/tardbp-egfp-hek293-hd2-5-4-20180829-ju_trimmed_single.bed.gz',\n",
       " '/home/aram/data/martina_tdp/test_idr_mutants/tardbp-g294a-egfp-hek293-4-20181001-ju_trimmed_single.bed.gz',\n",
       " '/home/aram/data/martina_tdp/second_big_heatmap/tardbp-egfp-293flp-l2-20171010-ju_trimmed_single.bed.gz',\n",
       " '/home/aram/data/martina_tdp/second_big_heatmap/tardbp-egfp-293flp-l1-20171010-ju_trimmed_single.bed.gz',\n",
       " '/home/aram/data/martina_tdp/second_big_heatmap/tardbp-316del346-egfp-293flp-l1-20171010-ju_trimmed_single.bed.gz',\n",
       " '/home/aram/data/martina_tdp/second_big_heatmap/tardbp-316del346-egfp-293flp-l2-20171010-ju_trimmed_single.bed.gz',\n",
       " '/home/aram/data/martina_tdp/second_big_heatmap/tardbp-egfp-293flp-m1-20171010-ju_trimmed_single.bed.gz',\n",
       " '/home/aram/data/martina_tdp/second_big_heatmap/tardbp-316del346-egfp-293flp-m2-20171010-ju_trimmed_single.bed.gz',\n",
       " '/home/aram/data/martina_tdp/second_big_heatmap/tardbp-316del346-egfp-293flp-m1-20171010-ju_trimmed_single.bed.gz',\n",
       " '/home/aram/data/martina_tdp/second_big_heatmap/tardbp-egfp-293flp-m2-20171010-ju_trimmed_single.bed.gz']"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "files_old = ['/home/aram/data/martina_tdp/test_idr_mutants/tardbp-316del346-egfp-hek293-4-20181001-ju_trimmed_single.bed.gz',\n",
    " '/home/aram/data/martina_tdp/test_idr_mutants/tardbp-316del346-egfp-hek293-3-20181001-ju_trimmed_single.bed.gz',\n",
    " '/home/aram/data/martina_tdp/test_idr_mutants/tardbp-egfp-hek293-hd2-5-4-20180829-ju_trimmed_single.bed.gz',\n",
    " '/home/aram/data/martina_tdp/test_idr_mutants/tardbp-a326p-egfp-hek293-4-20181001-ju_trimmed_single.bed.gz',\n",
    " '/home/aram/data/martina_tdp/test_idr_mutants/tardbp-egfp-hek293-hd1-6-on-bead-rt-1-20180829-ju_trimmed_single.bed.gz',\n",
    " '/home/aram/data/martina_tdp/test_idr_mutants/tardbp-g294a-egfp-hek293-4-20181001-ju_trimmed_single.bed.gz',\n",
    " '/home/aram/data/martina_tdp/test_idr_mutants/tardbp-316del346-egfp-hek293-hd1-6-3-20180829-ju_trimmed_single.bed.gz',\n",
    " '/home/aram/data/martina_tdp/test_idr_mutants/tardbp-320del414-egfp-hek293-hd1-6-3-20180829-ju_trimmed_single.bed.gz',\n",
    " '/home/aram/data/martina_tdp/test_idr_mutants/tardbp-egfp-hek293-no-uv-1-20180829-ju_trimmed_single.bed.gz',\n",
    " '/home/aram/data/martina_tdp/test_idr_mutants/tardbp-egfp-hek293-hd2-5-on-bead-rt-1-20180829-ju_trimmed_single.bed.gz',\n",
    " '/home/aram/data/martina_tdp/test_idr_mutants/tardbp-egfp-hek293-hd1-6-3-20180829-ju_trimmed_single.bed.gz',\n",
    " '/home/aram/data/martina_tdp/test_idr_mutants/tardbp-316del346-egfp-hek293-hd1-6-4-20180829-ju_trimmed_single.bed.gz',\n",
    " '/home/aram/data/martina_tdp/test_idr_mutants/tardbp-316del346-egfp-hek293-hd2-5-4-20180829-ju_trimmed_single.bed.gz',\n",
    " '/home/aram/data/martina_tdp/test_idr_mutants/tardbp-a326p-egfp-hek293-3-20181001-ju_trimmed_single.bed.gz',\n",
    " '/home/aram/data/martina_tdp/test_idr_mutants/tardbp-g335a-egfp-hek293-3-20181001-ju_trimmed_single.bed.gz',\n",
    " '/home/aram/data/martina_tdp/test_idr_mutants/tardbp-g294a-egfp-hek293-3-20181001-ju_trimmed_single.bed.gz',\n",
    " '/home/aram/data/martina_tdp/test_idr_mutants/tardbp-320del414-egfp-hek293-hd1-6-on-bead-rt-1-20180829-ju_trimmed_single.bed.gz',\n",
    " '/home/aram/data/martina_tdp/test_idr_mutants/tardbp-316del346-egfp-hek293-hd1-6-on-bead-rt-1-20180829-ju_trimmed_single.bed.gz',\n",
    " '/home/aram/data/martina_tdp/test_idr_mutants/tardbp-m337p-egfp-hek293-3-20181001-ju_trimmed_single.bed.gz',\n",
    " '/home/aram/data/martina_tdp/test_idr_mutants/tardbp-q331k-egfp-hek293-4-20181001-ju_trimmed_single.bed.gz',\n",
    " '/home/aram/data/martina_tdp/test_idr_mutants/tardbp-egfp-hek293-hd2-5-3-20180829-ju_trimmed_single.bed.gz',\n",
    " '/home/aram/data/martina_tdp/test_idr_mutants/tardbp-316del346-egfp-hek293-hd2-5-3-20180829-ju_trimmed_single.bed.gz',\n",
    " '/home/aram/data/martina_tdp/test_idr_mutants/tardbp-320del414-egfp-hek293-hd1-6-4-20180829-ju_trimmed_single.bed.gz',\n",
    " '/home/aram/data/martina_tdp/test_idr_mutants/tardbp-egfp-hek293-3-20181001-ju_trimmed_single.bed.gz',\n",
    " '/home/aram/data/martina_tdp/test_idr_mutants/tardbp-q331k-egfp-hek293-no-uv-1-20181001-ju_trimmed_single.bed.gz',\n",
    " '/home/aram/data/martina_tdp/test_idr_mutants/tardbp-g335a-egfp-hek293-4-20181001-ju_trimmed_single.bed.gz',\n",
    " '/home/aram/data/martina_tdp/test_idr_mutants/tardbp-316del346-egfp-hek293-hd2-5-on-bead-rt-1-20180829-ju_trimmed_single.bed.gz',\n",
    " '/home/aram/data/martina_tdp/test_idr_mutants/tardbp-q331k-egfp-hek293-3-20181001-ju_trimmed_single.bed.gz',\n",
    " '/home/aram/data/martina_tdp/test_idr_mutants/tardbp-egfp-hek293-hd1-6-4-20180829-ju_trimmed_single.bed.gz',\n",
    " '/home/aram/data/martina_tdp/test_idr_mutants/tardbp-egfp-hek293-4-20181001-ju_trimmed_single.bed.gz',\n",
    " '/home/aram/data/martina_tdp/test_idr_mutants/tardbp-m337p-egfp-hek293-4-20181001-ju_trimmed_single.bed.gz',\n",
    " '/home/aram/data/martina_tdp/second_big_heatmap/tardbp-egfp-293flp-m2-20171010-ju_trimmed_single.bed.gz',\n",
    " '/home/aram/data/martina_tdp/second_big_heatmap/tardbp-316del346-egfp-293flp-l1-20171010-ju_trimmed_single.bed.gz',\n",
    " '/home/aram/data/martina_tdp/second_big_heatmap/tardbp-316del346-egfp-293flp-l2-20171010-ju_trimmed_single.bed.gz',\n",
    " '/home/aram/data/martina_tdp/second_big_heatmap/tardbp-egfp-293flp-l2-20171010-ju_trimmed_single.bed.gz',\n",
    " '/home/aram/data/martina_tdp/second_big_heatmap/tardbp-316del346-egfp-293flp-m1-20171010-ju_trimmed_single.bed.gz',\n",
    " '/home/aram/data/martina_tdp/second_big_heatmap/tardbp-egfp-293flp-l1-20171010-ju_trimmed_single.bed.gz',\n",
    " '/home/aram/data/martina_tdp/second_big_heatmap/tardbp-316del346-egfp-293flp-m2-20171010-ju_trimmed_single.bed.gz',\n",
    " '/home/aram/data/martina_tdp/second_big_heatmap/tardbp-egfp-293flp-m1-20171010-ju_trimmed_single.bed.gz']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_out = df_all_unfiltered.copy()\n",
    "for file in files_old:\n",
    "    temp_name = file.split('/')[-1].replace('_trimmed_single.bed.gz', '')\n",
    "    temp = pbt.BedTool(\n",
    "        file).intersect(\n",
    "            bed_out_all_unfiltered,\n",
    "            s=True,\n",
    "            wb=True,\n",
    "            nonamecheck=True)\n",
    "    df_temp_unmerged = temp.to_dataframe()\n",
    "    df_temp = pd.DataFrame()\n",
    "    df_temp['chrome'] = df_temp_unmerged.chrom\n",
    "    df_temp['start'] = df_temp_unmerged.thickEnd\n",
    "    df_temp['end'] = df_temp_unmerged.itemRgb\n",
    "    df_temp['name'] = df_temp_unmerged.name\n",
    "    df_temp['score'] = df_temp_unmerged.blockSizes\n",
    "    df_temp['strand'] = df_temp_unmerged.strand\n",
    "    temp_merged = merge_peaks(pbt.BedTool.from_dataframe(df_temp), 0)\n",
    "    df_temp_merged = temp_merged.to_dataframe()\n",
    "    df_temp_merged = df_temp_merged.rename(columns={'score': f'{temp_name}'})\n",
    "    df_out = pd.merge(df_out, df_temp_merged, on=['chrom', 'start', 'end','name', 'strand'], how='outer')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>name</th>\n",
       "      <th>score</th>\n",
       "      <th>strand</th>\n",
       "      <th>tardbp-316del346-egfp-hek293-4-20181001-ju</th>\n",
       "      <th>tardbp-316del346-egfp-hek293-3-20181001-ju</th>\n",
       "      <th>tardbp-egfp-hek293-hd2-5-4-20180829-ju</th>\n",
       "      <th>tardbp-a326p-egfp-hek293-4-20181001-ju</th>\n",
       "      <th>...</th>\n",
       "      <th>tardbp-egfp-hek293-4-20181001-ju</th>\n",
       "      <th>tardbp-m337p-egfp-hek293-4-20181001-ju</th>\n",
       "      <th>tardbp-egfp-293flp-m2-20171010-ju</th>\n",
       "      <th>tardbp-316del346-egfp-293flp-l1-20171010-ju</th>\n",
       "      <th>tardbp-316del346-egfp-293flp-l2-20171010-ju</th>\n",
       "      <th>tardbp-egfp-293flp-l2-20171010-ju</th>\n",
       "      <th>tardbp-316del346-egfp-293flp-m1-20171010-ju</th>\n",
       "      <th>tardbp-egfp-293flp-l1-20171010-ju</th>\n",
       "      <th>tardbp-316del346-egfp-293flp-m2-20171010-ju</th>\n",
       "      <th>tardbp-egfp-293flp-m1-20171010-ju</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GL000008.2</td>\n",
       "      <td>118900</td>\n",
       "      <td>118901</td>\n",
       "      <td>.</td>\n",
       "      <td>2</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GL000008.2</td>\n",
       "      <td>131468</td>\n",
       "      <td>131481</td>\n",
       "      <td>.</td>\n",
       "      <td>40</td>\n",
       "      <td>+</td>\n",
       "      <td>40.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>80.0</td>\n",
       "      <td>160.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>40.0</td>\n",
       "      <td>40.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GL000008.2</td>\n",
       "      <td>132030</td>\n",
       "      <td>132032</td>\n",
       "      <td>.</td>\n",
       "      <td>5</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GL000008.2</td>\n",
       "      <td>154464</td>\n",
       "      <td>154475</td>\n",
       "      <td>.</td>\n",
       "      <td>113</td>\n",
       "      <td>+</td>\n",
       "      <td>339.0</td>\n",
       "      <td>113.0</td>\n",
       "      <td>339.0</td>\n",
       "      <td>226.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>226.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>113.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>226.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GL000008.2</td>\n",
       "      <td>155093</td>\n",
       "      <td>155130</td>\n",
       "      <td>.</td>\n",
       "      <td>110</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>110.0</td>\n",
       "      <td>770.0</td>\n",
       "      <td>220.0</td>\n",
       "      <td>...</td>\n",
       "      <td>110.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>110.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>110.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>110.0</td>\n",
       "      <td>220.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>110.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 45 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        chrom   start     end name  score strand  \\\n",
       "0  GL000008.2  118900  118901    .      2      -   \n",
       "1  GL000008.2  131468  131481    .     40      +   \n",
       "2  GL000008.2  132030  132032    .      5      +   \n",
       "3  GL000008.2  154464  154475    .    113      +   \n",
       "4  GL000008.2  155093  155130    .    110      +   \n",
       "\n",
       "   tardbp-316del346-egfp-hek293-4-20181001-ju  \\\n",
       "0                                         NaN   \n",
       "1                                        40.0   \n",
       "2                                         NaN   \n",
       "3                                       339.0   \n",
       "4                                         NaN   \n",
       "\n",
       "   tardbp-316del346-egfp-hek293-3-20181001-ju  \\\n",
       "0                                         NaN   \n",
       "1                                         NaN   \n",
       "2                                         5.0   \n",
       "3                                       113.0   \n",
       "4                                       110.0   \n",
       "\n",
       "   tardbp-egfp-hek293-hd2-5-4-20180829-ju  \\\n",
       "0                                     NaN   \n",
       "1                                    80.0   \n",
       "2                                     NaN   \n",
       "3                                   339.0   \n",
       "4                                   770.0   \n",
       "\n",
       "   tardbp-a326p-egfp-hek293-4-20181001-ju  ...  \\\n",
       "0                                     NaN  ...   \n",
       "1                                   160.0  ...   \n",
       "2                                     NaN  ...   \n",
       "3                                   226.0  ...   \n",
       "4                                   220.0  ...   \n",
       "\n",
       "   tardbp-egfp-hek293-4-20181001-ju  tardbp-m337p-egfp-hek293-4-20181001-ju  \\\n",
       "0                               NaN                                     NaN   \n",
       "1                               NaN                                    40.0   \n",
       "2                               NaN                                     5.0   \n",
       "3                               NaN                                     NaN   \n",
       "4                             110.0                                     NaN   \n",
       "\n",
       "   tardbp-egfp-293flp-m2-20171010-ju  \\\n",
       "0                                NaN   \n",
       "1                               40.0   \n",
       "2                                NaN   \n",
       "3                              226.0   \n",
       "4                              110.0   \n",
       "\n",
       "   tardbp-316del346-egfp-293flp-l1-20171010-ju  \\\n",
       "0                                          NaN   \n",
       "1                                          NaN   \n",
       "2                                          NaN   \n",
       "3                                          NaN   \n",
       "4                                          NaN   \n",
       "\n",
       "   tardbp-316del346-egfp-293flp-l2-20171010-ju  \\\n",
       "0                                          NaN   \n",
       "1                                          NaN   \n",
       "2                                          NaN   \n",
       "3                                          NaN   \n",
       "4                                        110.0   \n",
       "\n",
       "   tardbp-egfp-293flp-l2-20171010-ju  \\\n",
       "0                                NaN   \n",
       "1                                NaN   \n",
       "2                                NaN   \n",
       "3                              113.0   \n",
       "4                                NaN   \n",
       "\n",
       "   tardbp-316del346-egfp-293flp-m1-20171010-ju  \\\n",
       "0                                          NaN   \n",
       "1                                          NaN   \n",
       "2                                          NaN   \n",
       "3                                          NaN   \n",
       "4                                        110.0   \n",
       "\n",
       "   tardbp-egfp-293flp-l1-20171010-ju  \\\n",
       "0                                NaN   \n",
       "1                                NaN   \n",
       "2                                NaN   \n",
       "3                                NaN   \n",
       "4                              220.0   \n",
       "\n",
       "   tardbp-316del346-egfp-293flp-m2-20171010-ju  \\\n",
       "0                                          NaN   \n",
       "1                                          NaN   \n",
       "2                                          NaN   \n",
       "3                                          NaN   \n",
       "4                                          NaN   \n",
       "\n",
       "   tardbp-egfp-293flp-m1-20171010-ju  \n",
       "0                                NaN  \n",
       "1                                NaN  \n",
       "2                                NaN  \n",
       "3                              226.0  \n",
       "4                              110.0  \n",
       "\n",
       "[5 rows x 45 columns]"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_out.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_out_nonan = df_out.fillna(0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "region_file = '/home/aram/genomes/regions_GENCODE_v30.gtf.gz'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "unfiltered_with_region = pbt.BedTool(\n",
    "        bed_out_all_unfiltered).intersect(\n",
    "            region_file,\n",
    "            s=True,\n",
    "            wb=True,\n",
    "            nonamecheck=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_unfiltered_with_region = unfiltered_with_region.to_dataframe(\n",
    "    names=['chrom', 'start', 'end', 'name', 'score', 'strand', 'chrom1', 'name1', 'region', 'start1', 'end1', \n",
    "          'x1', 'strand1', 'x2', 'info'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "del df_unfiltered_with_region['chrom1']\n",
    "del df_unfiltered_with_region['name1']\n",
    "del df_unfiltered_with_region['start1']\n",
    "del df_unfiltered_with_region['end1']\n",
    "del df_unfiltered_with_region['x1']\n",
    "del df_unfiltered_with_region['strand1']\n",
    "del df_unfiltered_with_region['x2']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>name</th>\n",
       "      <th>score</th>\n",
       "      <th>strand</th>\n",
       "      <th>region</th>\n",
       "      <th>info</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GL000008.2</td>\n",
       "      <td>118900</td>\n",
       "      <td>118901</td>\n",
       "      <td>.</td>\n",
       "      <td>2</td>\n",
       "      <td>-</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>biotype \"\";gene_id \".\";gene_name \"None\";</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GL000008.2</td>\n",
       "      <td>131468</td>\n",
       "      <td>131481</td>\n",
       "      <td>.</td>\n",
       "      <td>40</td>\n",
       "      <td>+</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>biotype \"\";gene_id \".\";gene_name \"None\";</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GL000008.2</td>\n",
       "      <td>132030</td>\n",
       "      <td>132032</td>\n",
       "      <td>.</td>\n",
       "      <td>5</td>\n",
       "      <td>+</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>biotype \"\";gene_id \".\";gene_name \"None\";</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GL000008.2</td>\n",
       "      <td>154464</td>\n",
       "      <td>154475</td>\n",
       "      <td>.</td>\n",
       "      <td>113</td>\n",
       "      <td>+</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>biotype \"\";gene_id \".\";gene_name \"None\";</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GL000008.2</td>\n",
       "      <td>155093</td>\n",
       "      <td>155130</td>\n",
       "      <td>.</td>\n",
       "      <td>110</td>\n",
       "      <td>+</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>biotype \"\";gene_id \".\";gene_name \"None\";</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        chrom   start     end name  score strand      region  \\\n",
       "0  GL000008.2  118900  118901    .      2      -  intergenic   \n",
       "1  GL000008.2  131468  131481    .     40      +  intergenic   \n",
       "2  GL000008.2  132030  132032    .      5      +  intergenic   \n",
       "3  GL000008.2  154464  154475    .    113      +  intergenic   \n",
       "4  GL000008.2  155093  155130    .    110      +  intergenic   \n",
       "\n",
       "                                       info  \n",
       "0  biotype \"\";gene_id \".\";gene_name \"None\";  \n",
       "1  biotype \"\";gene_id \".\";gene_name \"None\";  \n",
       "2  biotype \"\";gene_id \".\";gene_name \"None\";  \n",
       "3  biotype \"\";gene_id \".\";gene_name \"None\";  \n",
       "4  biotype \"\";gene_id \".\";gene_name \"None\";  "
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_unfiltered_with_region.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_unfiltered_with_region['biotype'] = df_unfiltered_with_region['info'].apply(\n",
    "    lambda x: x.split(';')[0].replace('biotype ', ''))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_unfiltered_with_region['gene_id'] = df_unfiltered_with_region['info'].apply(\n",
    "    lambda x: x.split(';')[1].replace('gene_id ', ''))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_unfiltered_with_region['gene_name'] = df_unfiltered_with_region['info'].apply(\n",
    "    lambda x: x.split(';')[2].replace('gene_name ', ''))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [],
   "source": [
    "del df_unfiltered_with_region['info']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "981033"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(df_unfiltered_with_region)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_genes = df_unfiltered_with_region[df_unfiltered_with_region.region != 'intergenic']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_genes_100 = df_genes[df_genes.score > 100]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "54503"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(df_genes_100)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/aram/anaconda3/envs/pandas024/lib/python3.7/site-packages/ipykernel_launcher.py:1: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \"\"\"Entry point for launching an IPython kernel.\n"
     ]
    }
   ],
   "source": [
    "df_genes['max'] = df_genes.groupby('gene_id')['score'].transform(max)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/aram/anaconda3/envs/pandas024/lib/python3.7/site-packages/ipykernel_launcher.py:1: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \"\"\"Entry point for launching an IPython kernel.\n"
     ]
    }
   ],
   "source": [
    "df_genes['ratio'] = df_genes['score'] / df_genes['max']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_genes_filtered = df_genes[df_genes.ratio > 0.1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_genes_filtered = df_genes_filtered[df_genes_filtered.score > 10]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_intergenic = df_unfiltered_with_region[df_unfiltered_with_region.region == 'intergenic']\n",
    "df_intergenic_filtered = df_unfiltered_with_region[df_unfiltered_with_region.score > 100]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(109871, 57022)"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(df_intergenic), len(df_intergenic_filtered)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>name</th>\n",
       "      <th>score</th>\n",
       "      <th>strand</th>\n",
       "      <th>region</th>\n",
       "      <th>biotype</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>max</th>\n",
       "      <th>ratio</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>71</th>\n",
       "      <td>GL000194.1</td>\n",
       "      <td>75522</td>\n",
       "      <td>75545</td>\n",
       "      <td>.</td>\n",
       "      <td>22</td>\n",
       "      <td>-</td>\n",
       "      <td>intron</td>\n",
       "      <td>\"pre-mRNA\"</td>\n",
       "      <td>\"ENSG00000274847.1\"</td>\n",
       "      <td>\"MAFIP\"</td>\n",
       "      <td>68</td>\n",
       "      <td>0.323529</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>74</th>\n",
       "      <td>GL000194.1</td>\n",
       "      <td>81461</td>\n",
       "      <td>81479</td>\n",
       "      <td>.</td>\n",
       "      <td>13</td>\n",
       "      <td>-</td>\n",
       "      <td>intron</td>\n",
       "      <td>\"pre-mRNA\"</td>\n",
       "      <td>\"ENSG00000274847.1\"</td>\n",
       "      <td>\"MAFIP\"</td>\n",
       "      <td>68</td>\n",
       "      <td>0.191176</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79</th>\n",
       "      <td>GL000194.1</td>\n",
       "      <td>92228</td>\n",
       "      <td>92285</td>\n",
       "      <td>.</td>\n",
       "      <td>68</td>\n",
       "      <td>-</td>\n",
       "      <td>intron</td>\n",
       "      <td>\"pre-mRNA\"</td>\n",
       "      <td>\"ENSG00000274847.1\"</td>\n",
       "      <td>\"MAFIP\"</td>\n",
       "      <td>68</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>88</th>\n",
       "      <td>GL000194.1</td>\n",
       "      <td>99578</td>\n",
       "      <td>99599</td>\n",
       "      <td>.</td>\n",
       "      <td>23</td>\n",
       "      <td>-</td>\n",
       "      <td>intron</td>\n",
       "      <td>\"pre-mRNA\"</td>\n",
       "      <td>\"ENSG00000274847.1\"</td>\n",
       "      <td>\"MAFIP\"</td>\n",
       "      <td>68</td>\n",
       "      <td>0.338235</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>254</th>\n",
       "      <td>KI270442.1</td>\n",
       "      <td>217328</td>\n",
       "      <td>217337</td>\n",
       "      <td>.</td>\n",
       "      <td>19</td>\n",
       "      <td>-</td>\n",
       "      <td>ncRNA</td>\n",
       "      <td>\"rRNA\"</td>\n",
       "      <td>\"ENSG00000278294.1\"</td>\n",
       "      <td>\"RF00002\"</td>\n",
       "      <td>19</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          chrom   start     end name  score strand  region     biotype  \\\n",
       "71   GL000194.1   75522   75545    .     22      -  intron  \"pre-mRNA\"   \n",
       "74   GL000194.1   81461   81479    .     13      -  intron  \"pre-mRNA\"   \n",
       "79   GL000194.1   92228   92285    .     68      -  intron  \"pre-mRNA\"   \n",
       "88   GL000194.1   99578   99599    .     23      -  intron  \"pre-mRNA\"   \n",
       "254  KI270442.1  217328  217337    .     19      -   ncRNA      \"rRNA\"   \n",
       "\n",
       "                 gene_id  gene_name  max     ratio  \n",
       "71   \"ENSG00000274847.1\"    \"MAFIP\"   68  0.323529  \n",
       "74   \"ENSG00000274847.1\"    \"MAFIP\"   68  0.191176  \n",
       "79   \"ENSG00000274847.1\"    \"MAFIP\"   68  1.000000  \n",
       "88   \"ENSG00000274847.1\"    \"MAFIP\"   68  0.338235  \n",
       "254  \"ENSG00000278294.1\"  \"RF00002\"   19  1.000000  "
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_genes_filtered.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [],
   "source": [
    "del df_genes_filtered['max']\n",
    "del df_genes_filtered['ratio']\n",
    "#del df_genes_100['max']\n",
    "#del df_genes_100['ratio']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_genes_filtered = df_genes_filtered.merge(df_genes_100, how='outer', \n",
    "    on=['chrom', 'start', 'end', 'name', 'score', 'strand', 'region', 'biotype', 'gene_id', 'gene_name'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_filtered = pd.concat([df_intergenic_filtered, df_genes_filtered], axis=0, join='outer')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_filtered = df_filtered.sort_index()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_filtered =  df_filtered.drop_duplicates()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "120159"
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(df_filtered)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>name</th>\n",
       "      <th>score</th>\n",
       "      <th>strand</th>\n",
       "      <th>region</th>\n",
       "      <th>biotype</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GL000194.1</td>\n",
       "      <td>75522</td>\n",
       "      <td>75545</td>\n",
       "      <td>.</td>\n",
       "      <td>22</td>\n",
       "      <td>-</td>\n",
       "      <td>intron</td>\n",
       "      <td>\"pre-mRNA\"</td>\n",
       "      <td>\"ENSG00000274847.1\"</td>\n",
       "      <td>\"MAFIP\"</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GL000194.1</td>\n",
       "      <td>81461</td>\n",
       "      <td>81479</td>\n",
       "      <td>.</td>\n",
       "      <td>13</td>\n",
       "      <td>-</td>\n",
       "      <td>intron</td>\n",
       "      <td>\"pre-mRNA\"</td>\n",
       "      <td>\"ENSG00000274847.1\"</td>\n",
       "      <td>\"MAFIP\"</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GL000194.1</td>\n",
       "      <td>92228</td>\n",
       "      <td>92285</td>\n",
       "      <td>.</td>\n",
       "      <td>68</td>\n",
       "      <td>-</td>\n",
       "      <td>intron</td>\n",
       "      <td>\"pre-mRNA\"</td>\n",
       "      <td>\"ENSG00000274847.1\"</td>\n",
       "      <td>\"MAFIP\"</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GL000194.1</td>\n",
       "      <td>99578</td>\n",
       "      <td>99599</td>\n",
       "      <td>.</td>\n",
       "      <td>23</td>\n",
       "      <td>-</td>\n",
       "      <td>intron</td>\n",
       "      <td>\"pre-mRNA\"</td>\n",
       "      <td>\"ENSG00000274847.1\"</td>\n",
       "      <td>\"MAFIP\"</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GL000008.2</td>\n",
       "      <td>154464</td>\n",
       "      <td>154475</td>\n",
       "      <td>.</td>\n",
       "      <td>113</td>\n",
       "      <td>+</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>\"\"</td>\n",
       "      <td>\".\"</td>\n",
       "      <td>\"None\"</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        chrom   start     end name  score strand      region     biotype  \\\n",
       "0  GL000194.1   75522   75545    .     22      -      intron  \"pre-mRNA\"   \n",
       "1  GL000194.1   81461   81479    .     13      -      intron  \"pre-mRNA\"   \n",
       "2  GL000194.1   92228   92285    .     68      -      intron  \"pre-mRNA\"   \n",
       "3  GL000194.1   99578   99599    .     23      -      intron  \"pre-mRNA\"   \n",
       "3  GL000008.2  154464  154475    .    113      +  intergenic          \"\"   \n",
       "\n",
       "               gene_id gene_name  \n",
       "0  \"ENSG00000274847.1\"   \"MAFIP\"  \n",
       "1  \"ENSG00000274847.1\"   \"MAFIP\"  \n",
       "2  \"ENSG00000274847.1\"   \"MAFIP\"  \n",
       "3  \"ENSG00000274847.1\"   \"MAFIP\"  \n",
       "3                  \".\"    \"None\"  "
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_filtered.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [],
   "source": [
    "rfiltered_merged30 = pbt.BedTool.from_dataframe(\n",
    "    df_filtered[['chrom', 'start', 'end', 'name', 'score', 'strand']])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [],
   "source": [
    "rfiltered_merged30_table = pbt.BedTool(rfiltered_merged30).intersect(\n",
    "        pbt.BedTool('/home/aram/genomes/regions.bed.gz'),\n",
    "        s=True,\n",
    "        wb=True,\n",
    "        nonamecheck=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_rfiltered_merged30_table = rfiltered_merged30_table.to_dataframe(\n",
    "    names=['chrom', 'start', 'end', 'name', 'score', 'strand', 'chrom1', 'name1', 'region', 'start1', 'end1', \n",
    "          'x1', 'strand1', 'x2', 'info'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [],
   "source": [
    "del df_rfiltered_merged30_table['chrom1']\n",
    "del df_rfiltered_merged30_table['name1']\n",
    "del df_rfiltered_merged30_table['start1']\n",
    "del df_rfiltered_merged30_table['end1']\n",
    "del df_rfiltered_merged30_table['x1']\n",
    "del df_rfiltered_merged30_table['strand1']\n",
    "del df_rfiltered_merged30_table['x2']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_rfiltered_merged30_table['biotype'] = df_rfiltered_merged30_table['info'].apply(\n",
    "    lambda x: x.split(';')[2].replace('biotype ', ''))\n",
    "df_rfiltered_merged30_table['gene_id'] = df_rfiltered_merged30_table['info'].apply(\n",
    "    lambda x: x.split(';')[0].replace('gene_id ', ''))\n",
    "df_rfiltered_merged30_table['gene_name'] = df_rfiltered_merged30_table['info'].apply(\n",
    "    lambda x: x.split(';')[1].replace('gene_name ', ''))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>name</th>\n",
       "      <th>score</th>\n",
       "      <th>strand</th>\n",
       "      <th>region</th>\n",
       "      <th>info</th>\n",
       "      <th>biotype</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GL000194.1</td>\n",
       "      <td>75522</td>\n",
       "      <td>75545</td>\n",
       "      <td>.</td>\n",
       "      <td>22</td>\n",
       "      <td>-</td>\n",
       "      <td>intron</td>\n",
       "      <td>gene_id \"ENSG00000274847.1\";gene_name \"AC14521...</td>\n",
       "      <td>\"pre-mRNA\"</td>\n",
       "      <td>\"ENSG00000274847.1\"</td>\n",
       "      <td>\"AC145212.1\"</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GL000194.1</td>\n",
       "      <td>81461</td>\n",
       "      <td>81479</td>\n",
       "      <td>.</td>\n",
       "      <td>13</td>\n",
       "      <td>-</td>\n",
       "      <td>intron</td>\n",
       "      <td>gene_id \"ENSG00000274847.1\";gene_name \"AC14521...</td>\n",
       "      <td>\"pre-mRNA\"</td>\n",
       "      <td>\"ENSG00000274847.1\"</td>\n",
       "      <td>\"AC145212.1\"</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GL000194.1</td>\n",
       "      <td>92228</td>\n",
       "      <td>92285</td>\n",
       "      <td>.</td>\n",
       "      <td>68</td>\n",
       "      <td>-</td>\n",
       "      <td>intron</td>\n",
       "      <td>gene_id \"ENSG00000274847.1\";gene_name \"AC14521...</td>\n",
       "      <td>\"pre-mRNA\"</td>\n",
       "      <td>\"ENSG00000274847.1\"</td>\n",
       "      <td>\"AC145212.1\"</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GL000194.1</td>\n",
       "      <td>99578</td>\n",
       "      <td>99599</td>\n",
       "      <td>.</td>\n",
       "      <td>23</td>\n",
       "      <td>-</td>\n",
       "      <td>intron</td>\n",
       "      <td>gene_id \"ENSG00000274847.1\";gene_name \"AC14521...</td>\n",
       "      <td>\"pre-mRNA\"</td>\n",
       "      <td>\"ENSG00000274847.1\"</td>\n",
       "      <td>\"AC145212.1\"</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GL000008.2</td>\n",
       "      <td>154464</td>\n",
       "      <td>154475</td>\n",
       "      <td>.</td>\n",
       "      <td>113</td>\n",
       "      <td>+</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>gene_id \".\";gene_name \"None\";biotype \"\";</td>\n",
       "      <td>\"\"</td>\n",
       "      <td>\".\"</td>\n",
       "      <td>\"None\"</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        chrom   start     end name  score strand      region  \\\n",
       "0  GL000194.1   75522   75545    .     22      -      intron   \n",
       "1  GL000194.1   81461   81479    .     13      -      intron   \n",
       "2  GL000194.1   92228   92285    .     68      -      intron   \n",
       "3  GL000194.1   99578   99599    .     23      -      intron   \n",
       "4  GL000008.2  154464  154475    .    113      +  intergenic   \n",
       "\n",
       "                                                info     biotype  \\\n",
       "0  gene_id \"ENSG00000274847.1\";gene_name \"AC14521...  \"pre-mRNA\"   \n",
       "1  gene_id \"ENSG00000274847.1\";gene_name \"AC14521...  \"pre-mRNA\"   \n",
       "2  gene_id \"ENSG00000274847.1\";gene_name \"AC14521...  \"pre-mRNA\"   \n",
       "3  gene_id \"ENSG00000274847.1\";gene_name \"AC14521...  \"pre-mRNA\"   \n",
       "4           gene_id \".\";gene_name \"None\";biotype \"\";          \"\"   \n",
       "\n",
       "               gene_id     gene_name  \n",
       "0  \"ENSG00000274847.1\"  \"AC145212.1\"  \n",
       "1  \"ENSG00000274847.1\"  \"AC145212.1\"  \n",
       "2  \"ENSG00000274847.1\"  \"AC145212.1\"  \n",
       "3  \"ENSG00000274847.1\"  \"AC145212.1\"  \n",
       "4                  \".\"        \"None\"  "
      ]
     },
     "execution_count": 52,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_rfiltered_merged30_table.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [],
   "source": [
    "del df_rfiltered_merged30_table['info']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_out_rfiltered_merged30 = df_rfiltered_merged30_table.copy()\n",
    "for file in files:\n",
    "    temp_name = file.split('/')[-1].replace('_trimmed_single.bed.gz', '')\n",
    "    temp = rfiltered_merged30.intersect(\n",
    "        pbt.BedTool(file),\n",
    "        s=True,\n",
    "        wa=True,\n",
    "        wb=True,\n",
    "        loj=True,\n",
    "        nonamecheck=True)\n",
    "    df_temp_unmerged = temp.to_dataframe()\n",
    "    df_temp = pd.DataFrame()\n",
    "    df_temp['chrome'] = df_temp_unmerged.chrom\n",
    "    df_temp['start'] = df_temp_unmerged.start\n",
    "    df_temp['end'] = df_temp_unmerged.end\n",
    "    df_temp['name'] = df_temp_unmerged.name\n",
    "    df_temp['score'] = df_temp_unmerged.blockSizes\n",
    "    df_temp['strand'] = df_temp_unmerged.strand\n",
    "    temp_merged = merge_peaks(pbt.BedTool.from_dataframe(df_temp), 0)\n",
    "    df_temp_merged = temp_merged.to_dataframe()\n",
    "    df_temp_merged = df_temp_merged.rename(columns={'score': f'{temp_name}'})\n",
    "    df_out_rfiltered_merged30 = pd.merge(df_out_rfiltered_merged30, df_temp_merged, on=['chrom', 'start', 'end','name', 'strand'], how='outer')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_out_rfiltered_merged30 = df_out_rfiltered_merged30.fillna(0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_out_rfiltered_merged30 = df_out_rfiltered_merged30.replace(-1.0, 0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>name</th>\n",
       "      <th>score</th>\n",
       "      <th>strand</th>\n",
       "      <th>region</th>\n",
       "      <th>biotype</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>...</th>\n",
       "      <th>tardbp-egfp-hek293-hd2-5-4-20180829-ju</th>\n",
       "      <th>tardbp-g294a-egfp-hek293-4-20181001-ju</th>\n",
       "      <th>tardbp-egfp-293flp-l2-20171010-ju</th>\n",
       "      <th>tardbp-egfp-293flp-l1-20171010-ju</th>\n",
       "      <th>tardbp-316del346-egfp-293flp-l1-20171010-ju</th>\n",
       "      <th>tardbp-316del346-egfp-293flp-l2-20171010-ju</th>\n",
       "      <th>tardbp-egfp-293flp-m1-20171010-ju</th>\n",
       "      <th>tardbp-316del346-egfp-293flp-m2-20171010-ju</th>\n",
       "      <th>tardbp-316del346-egfp-293flp-m1-20171010-ju</th>\n",
       "      <th>tardbp-egfp-293flp-m2-20171010-ju</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GL000194.1</td>\n",
       "      <td>75522</td>\n",
       "      <td>75545</td>\n",
       "      <td>.</td>\n",
       "      <td>22.0</td>\n",
       "      <td>-</td>\n",
       "      <td>intron</td>\n",
       "      <td>\"pre-mRNA\"</td>\n",
       "      <td>\"ENSG00000274847.1\"</td>\n",
       "      <td>\"AC145212.1\"</td>\n",
       "      <td>...</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GL000194.1</td>\n",
       "      <td>81461</td>\n",
       "      <td>81479</td>\n",
       "      <td>.</td>\n",
       "      <td>13.0</td>\n",
       "      <td>-</td>\n",
       "      <td>intron</td>\n",
       "      <td>\"pre-mRNA\"</td>\n",
       "      <td>\"ENSG00000274847.1\"</td>\n",
       "      <td>\"AC145212.1\"</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GL000194.1</td>\n",
       "      <td>92228</td>\n",
       "      <td>92285</td>\n",
       "      <td>.</td>\n",
       "      <td>68.0</td>\n",
       "      <td>-</td>\n",
       "      <td>intron</td>\n",
       "      <td>\"pre-mRNA\"</td>\n",
       "      <td>\"ENSG00000274847.1\"</td>\n",
       "      <td>\"AC145212.1\"</td>\n",
       "      <td>...</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GL000194.1</td>\n",
       "      <td>99578</td>\n",
       "      <td>99599</td>\n",
       "      <td>.</td>\n",
       "      <td>23.0</td>\n",
       "      <td>-</td>\n",
       "      <td>intron</td>\n",
       "      <td>\"pre-mRNA\"</td>\n",
       "      <td>\"ENSG00000274847.1\"</td>\n",
       "      <td>\"AC145212.1\"</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GL000008.2</td>\n",
       "      <td>154464</td>\n",
       "      <td>154475</td>\n",
       "      <td>.</td>\n",
       "      <td>113.0</td>\n",
       "      <td>+</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>\"\"</td>\n",
       "      <td>\".\"</td>\n",
       "      <td>\"None\"</td>\n",
       "      <td>...</td>\n",
       "      <td>6.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>GL000008.2</td>\n",
       "      <td>155093</td>\n",
       "      <td>155130</td>\n",
       "      <td>.</td>\n",
       "      <td>110.0</td>\n",
       "      <td>+</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>\"\"</td>\n",
       "      <td>\".\"</td>\n",
       "      <td>\"None\"</td>\n",
       "      <td>...</td>\n",
       "      <td>7.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>KI270442.1</td>\n",
       "      <td>217328</td>\n",
       "      <td>217337</td>\n",
       "      <td>.</td>\n",
       "      <td>19.0</td>\n",
       "      <td>-</td>\n",
       "      <td>ncRNA</td>\n",
       "      <td>\"rRNA\"</td>\n",
       "      <td>\"ENSG00000278294.1\"</td>\n",
       "      <td>\"5_8S_rRNA\"</td>\n",
       "      <td>...</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>KI270711.1</td>\n",
       "      <td>8179</td>\n",
       "      <td>8241</td>\n",
       "      <td>.</td>\n",
       "      <td>520.0</td>\n",
       "      <td>-</td>\n",
       "      <td>intron</td>\n",
       "      <td>\"pre-mRNA\"</td>\n",
       "      <td>\"ENSG00000271254.6\"</td>\n",
       "      <td>\"AC240274.1\"</td>\n",
       "      <td>...</td>\n",
       "      <td>19.0</td>\n",
       "      <td>14.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>8.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>KI270711.1</td>\n",
       "      <td>11269</td>\n",
       "      <td>11343</td>\n",
       "      <td>.</td>\n",
       "      <td>123.0</td>\n",
       "      <td>-</td>\n",
       "      <td>intron</td>\n",
       "      <td>\"pre-mRNA\"</td>\n",
       "      <td>\"ENSG00000271254.6\"</td>\n",
       "      <td>\"AC240274.1\"</td>\n",
       "      <td>...</td>\n",
       "      <td>8.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>KI270713.1</td>\n",
       "      <td>31706</td>\n",
       "      <td>31707</td>\n",
       "      <td>.</td>\n",
       "      <td>12.0</td>\n",
       "      <td>-</td>\n",
       "      <td>UTR3</td>\n",
       "      <td>\"mRNA\"</td>\n",
       "      <td>\"ENSG00000277475.1\"</td>\n",
       "      <td>\"AC213203.1\"</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>KI270728.1</td>\n",
       "      <td>1140224</td>\n",
       "      <td>1140264</td>\n",
       "      <td>.</td>\n",
       "      <td>157.0</td>\n",
       "      <td>-</td>\n",
       "      <td>UTR3</td>\n",
       "      <td>\"mRNA\"</td>\n",
       "      <td>\"ENSG00000273554.4\"</td>\n",
       "      <td>\"AC136616.1\"</td>\n",
       "      <td>...</td>\n",
       "      <td>20.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>KI270728.1</td>\n",
       "      <td>1236970</td>\n",
       "      <td>1236995</td>\n",
       "      <td>.</td>\n",
       "      <td>13.0</td>\n",
       "      <td>+</td>\n",
       "      <td>intron</td>\n",
       "      <td>\"pre-mRNA\"</td>\n",
       "      <td>\"ENSG00000277761.1\"</td>\n",
       "      <td>\"AC136616.2\"</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>KI270733.1</td>\n",
       "      <td>128915</td>\n",
       "      <td>128948</td>\n",
       "      <td>.</td>\n",
       "      <td>15.0</td>\n",
       "      <td>+</td>\n",
       "      <td>ncRNA</td>\n",
       "      <td>\"rRNA\"</td>\n",
       "      <td>\"ENSG00000276700.1\"</td>\n",
       "      <td>\"RNA5-8S5\"</td>\n",
       "      <td>...</td>\n",
       "      <td>3.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>chr1</td>\n",
       "      <td>280750</td>\n",
       "      <td>280772</td>\n",
       "      <td>.</td>\n",
       "      <td>43.0</td>\n",
       "      <td>-</td>\n",
       "      <td>intron</td>\n",
       "      <td>\"lncRNA\"</td>\n",
       "      <td>\"ENSG00000228463.10\"</td>\n",
       "      <td>\"AP006222.1\"</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>chr1</td>\n",
       "      <td>286726</td>\n",
       "      <td>286748</td>\n",
       "      <td>.</td>\n",
       "      <td>28.0</td>\n",
       "      <td>-</td>\n",
       "      <td>intron</td>\n",
       "      <td>\"lncRNA\"</td>\n",
       "      <td>\"ENSG00000228463.10\"</td>\n",
       "      <td>\"AP006222.1\"</td>\n",
       "      <td>...</td>\n",
       "      <td>3.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>chr1</td>\n",
       "      <td>517632</td>\n",
       "      <td>517641</td>\n",
       "      <td>.</td>\n",
       "      <td>12.0</td>\n",
       "      <td>-</td>\n",
       "      <td>intron</td>\n",
       "      <td>\"lncRNA\"</td>\n",
       "      <td>\"ENSG00000237094.12\"</td>\n",
       "      <td>\"AL732372.2\"</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>chr1</td>\n",
       "      <td>783745</td>\n",
       "      <td>783758</td>\n",
       "      <td>.</td>\n",
       "      <td>42.0</td>\n",
       "      <td>+</td>\n",
       "      <td>intron</td>\n",
       "      <td>\"lncRNA\"</td>\n",
       "      <td>\"ENSG00000237491.8\"</td>\n",
       "      <td>\"AL669831.5\"</td>\n",
       "      <td>...</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>chr1</td>\n",
       "      <td>783876</td>\n",
       "      <td>783882</td>\n",
       "      <td>.</td>\n",
       "      <td>22.0</td>\n",
       "      <td>+</td>\n",
       "      <td>intron</td>\n",
       "      <td>\"lncRNA\"</td>\n",
       "      <td>\"ENSG00000237491.8\"</td>\n",
       "      <td>\"AL669831.5\"</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>chr1</td>\n",
       "      <td>787103</td>\n",
       "      <td>787134</td>\n",
       "      <td>.</td>\n",
       "      <td>12.0</td>\n",
       "      <td>+</td>\n",
       "      <td>ncRNA</td>\n",
       "      <td>\"lncRNA\"</td>\n",
       "      <td>\"ENSG00000237491.8\"</td>\n",
       "      <td>\"AL669831.5\"</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>chr1</td>\n",
       "      <td>788818</td>\n",
       "      <td>788940</td>\n",
       "      <td>.</td>\n",
       "      <td>67.0</td>\n",
       "      <td>+</td>\n",
       "      <td>intron</td>\n",
       "      <td>\"lncRNA\"</td>\n",
       "      <td>\"ENSG00000237491.8\"</td>\n",
       "      <td>\"AL669831.5\"</td>\n",
       "      <td>...</td>\n",
       "      <td>3.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>chr1</td>\n",
       "      <td>788985</td>\n",
       "      <td>789003</td>\n",
       "      <td>.</td>\n",
       "      <td>18.0</td>\n",
       "      <td>+</td>\n",
       "      <td>intron</td>\n",
       "      <td>\"lncRNA\"</td>\n",
       "      <td>\"ENSG00000237491.8\"</td>\n",
       "      <td>\"AL669831.5\"</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>chr1</td>\n",
       "      <td>789057</td>\n",
       "      <td>789165</td>\n",
       "      <td>.</td>\n",
       "      <td>33.0</td>\n",
       "      <td>+</td>\n",
       "      <td>intron</td>\n",
       "      <td>\"lncRNA\"</td>\n",
       "      <td>\"ENSG00000237491.8\"</td>\n",
       "      <td>\"AL669831.5\"</td>\n",
       "      <td>...</td>\n",
       "      <td>2.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>chr1</td>\n",
       "      <td>789308</td>\n",
       "      <td>789319</td>\n",
       "      <td>.</td>\n",
       "      <td>11.0</td>\n",
       "      <td>+</td>\n",
       "      <td>intron</td>\n",
       "      <td>\"lncRNA\"</td>\n",
       "      <td>\"ENSG00000237491.8\"</td>\n",
       "      <td>\"AL669831.5\"</td>\n",
       "      <td>...</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>chr1</td>\n",
       "      <td>789979</td>\n",
       "      <td>790008</td>\n",
       "      <td>.</td>\n",
       "      <td>14.0</td>\n",
       "      <td>+</td>\n",
       "      <td>intron</td>\n",
       "      <td>\"lncRNA\"</td>\n",
       "      <td>\"ENSG00000237491.8\"</td>\n",
       "      <td>\"AL669831.5\"</td>\n",
       "      <td>...</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>chr1</td>\n",
       "      <td>790126</td>\n",
       "      <td>790175</td>\n",
       "      <td>.</td>\n",
       "      <td>33.0</td>\n",
       "      <td>+</td>\n",
       "      <td>intron</td>\n",
       "      <td>\"lncRNA\"</td>\n",
       "      <td>\"ENSG00000237491.8\"</td>\n",
       "      <td>\"AL669831.5\"</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>chr1</td>\n",
       "      <td>790293</td>\n",
       "      <td>790314</td>\n",
       "      <td>.</td>\n",
       "      <td>25.0</td>\n",
       "      <td>+</td>\n",
       "      <td>intron</td>\n",
       "      <td>\"lncRNA\"</td>\n",
       "      <td>\"ENSG00000237491.8\"</td>\n",
       "      <td>\"AL669831.5\"</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>chr1</td>\n",
       "      <td>791267</td>\n",
       "      <td>791302</td>\n",
       "      <td>.</td>\n",
       "      <td>21.0</td>\n",
       "      <td>+</td>\n",
       "      <td>intron</td>\n",
       "      <td>\"lncRNA\"</td>\n",
       "      <td>\"ENSG00000237491.8\"</td>\n",
       "      <td>\"AL669831.5\"</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>chr1</td>\n",
       "      <td>808217</td>\n",
       "      <td>808335</td>\n",
       "      <td>.</td>\n",
       "      <td>391.0</td>\n",
       "      <td>-</td>\n",
       "      <td>intron</td>\n",
       "      <td>\"lncRNA\"</td>\n",
       "      <td>\"ENSG00000230021.9\"</td>\n",
       "      <td>\"AL669831.3\"</td>\n",
       "      <td>...</td>\n",
       "      <td>20.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>chr1</td>\n",
       "      <td>819773</td>\n",
       "      <td>819849</td>\n",
       "      <td>.</td>\n",
       "      <td>46.0</td>\n",
       "      <td>-</td>\n",
       "      <td>intron</td>\n",
       "      <td>\"lncRNA\"</td>\n",
       "      <td>\"ENSG00000230021.9\"</td>\n",
       "      <td>\"AL669831.3\"</td>\n",
       "      <td>...</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>chr1</td>\n",
       "      <td>855188</td>\n",
       "      <td>855562</td>\n",
       "      <td>.</td>\n",
       "      <td>970.0</td>\n",
       "      <td>+</td>\n",
       "      <td>ncRNA</td>\n",
       "      <td>\"lncRNA\"</td>\n",
       "      <td>\"ENSG00000228794.8\"</td>\n",
       "      <td>\"LINC01128\"</td>\n",
       "      <td>...</td>\n",
       "      <td>102.0</td>\n",
       "      <td>22.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>14.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>13.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>chr1</td>\n",
       "      <td>907575</td>\n",
       "      <td>907585</td>\n",
       "      <td>.</td>\n",
       "      <td>18.0</td>\n",
       "      <td>+</td>\n",
       "      <td>intron</td>\n",
       "      <td>\"lncRNA\"</td>\n",
       "      <td>\"ENSG00000272438.1\"</td>\n",
       "      <td>\"AL645608.7\"</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>chr1</td>\n",
       "      <td>911477</td>\n",
       "      <td>911536</td>\n",
       "      <td>.</td>\n",
       "      <td>47.0</td>\n",
       "      <td>+</td>\n",
       "      <td>intron</td>\n",
       "      <td>\"lncRNA\"</td>\n",
       "      <td>\"ENSG00000272438.1\"</td>\n",
       "      <td>\"AL645608.7\"</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>chr1</td>\n",
       "      <td>911581</td>\n",
       "      <td>911589</td>\n",
       "      <td>.</td>\n",
       "      <td>29.0</td>\n",
       "      <td>+</td>\n",
       "      <td>intron</td>\n",
       "      <td>\"lncRNA\"</td>\n",
       "      <td>\"ENSG00000272438.1\"</td>\n",
       "      <td>\"AL645608.7\"</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>chr1</td>\n",
       "      <td>917870</td>\n",
       "      <td>917927</td>\n",
       "      <td>.</td>\n",
       "      <td>101.0</td>\n",
       "      <td>-</td>\n",
       "      <td>ncRNA</td>\n",
       "      <td>\"lncRNA\"</td>\n",
       "      <td>\"ENSG00000223764.2\"</td>\n",
       "      <td>\"AL645608.1\"</td>\n",
       "      <td>...</td>\n",
       "      <td>2.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>chr1</td>\n",
       "      <td>917980</td>\n",
       "      <td>918021</td>\n",
       "      <td>.</td>\n",
       "      <td>23.0</td>\n",
       "      <td>-</td>\n",
       "      <td>ncRNA</td>\n",
       "      <td>\"lncRNA\"</td>\n",
       "      <td>\"ENSG00000223764.2\"</td>\n",
       "      <td>\"AL645608.1\"</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>chr1</td>\n",
       "      <td>926449</td>\n",
       "      <td>926469</td>\n",
       "      <td>.</td>\n",
       "      <td>134.0</td>\n",
       "      <td>+</td>\n",
       "      <td>intron</td>\n",
       "      <td>\"pre-mRNA\"</td>\n",
       "      <td>\"ENSG00000187634.11\"</td>\n",
       "      <td>\"SAMD11\"</td>\n",
       "      <td>...</td>\n",
       "      <td>11.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>chr1</td>\n",
       "      <td>931389</td>\n",
       "      <td>931422</td>\n",
       "      <td>.</td>\n",
       "      <td>163.0</td>\n",
       "      <td>+</td>\n",
       "      <td>intron</td>\n",
       "      <td>\"pre-mRNA\"</td>\n",
       "      <td>\"ENSG00000187634.11\"</td>\n",
       "      <td>\"SAMD11\"</td>\n",
       "      <td>...</td>\n",
       "      <td>14.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>chr1</td>\n",
       "      <td>931465</td>\n",
       "      <td>931623</td>\n",
       "      <td>.</td>\n",
       "      <td>902.0</td>\n",
       "      <td>+</td>\n",
       "      <td>intron</td>\n",
       "      <td>\"pre-mRNA\"</td>\n",
       "      <td>\"ENSG00000187634.11\"</td>\n",
       "      <td>\"SAMD11\"</td>\n",
       "      <td>...</td>\n",
       "      <td>48.0</td>\n",
       "      <td>24.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>12.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>11.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>chr1</td>\n",
       "      <td>950069</td>\n",
       "      <td>950784</td>\n",
       "      <td>.</td>\n",
       "      <td>5158.0</td>\n",
       "      <td>-</td>\n",
       "      <td>intron</td>\n",
       "      <td>\"pre-mRNA\"</td>\n",
       "      <td>\"ENSG00000188976.10\"</td>\n",
       "      <td>\"NOC2L\"</td>\n",
       "      <td>...</td>\n",
       "      <td>310.0</td>\n",
       "      <td>92.0</td>\n",
       "      <td>68.0</td>\n",
       "      <td>78.0</td>\n",
       "      <td>36.0</td>\n",
       "      <td>50.0</td>\n",
       "      <td>53.0</td>\n",
       "      <td>35.0</td>\n",
       "      <td>35.0</td>\n",
       "      <td>84.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>chr1</td>\n",
       "      <td>962159</td>\n",
       "      <td>962172</td>\n",
       "      <td>.</td>\n",
       "      <td>28.0</td>\n",
       "      <td>+</td>\n",
       "      <td>UTR5</td>\n",
       "      <td>\"mRNA\"</td>\n",
       "      <td>\"ENSG00000187961.13\"</td>\n",
       "      <td>\"KLHL17\"</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>chr1</td>\n",
       "      <td>962211</td>\n",
       "      <td>962251</td>\n",
       "      <td>.</td>\n",
       "      <td>33.0</td>\n",
       "      <td>+</td>\n",
       "      <td>UTR5</td>\n",
       "      <td>\"mRNA\"</td>\n",
       "      <td>\"ENSG00000187961.13\"</td>\n",
       "      <td>\"KLHL17\"</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>chr1</td>\n",
       "      <td>969305</td>\n",
       "      <td>969434</td>\n",
       "      <td>.</td>\n",
       "      <td>42.0</td>\n",
       "      <td>+</td>\n",
       "      <td>intron</td>\n",
       "      <td>\"pre-mRNA\"</td>\n",
       "      <td>\"ENSG00000187583.10\"</td>\n",
       "      <td>\"PLEKHN1\"</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>chr1</td>\n",
       "      <td>969497</td>\n",
       "      <td>969817</td>\n",
       "      <td>.</td>\n",
       "      <td>119.0</td>\n",
       "      <td>+</td>\n",
       "      <td>intron</td>\n",
       "      <td>\"pre-mRNA\"</td>\n",
       "      <td>\"ENSG00000187583.10\"</td>\n",
       "      <td>\"PLEKHN1\"</td>\n",
       "      <td>...</td>\n",
       "      <td>11.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>chr1</td>\n",
       "      <td>969856</td>\n",
       "      <td>969914</td>\n",
       "      <td>.</td>\n",
       "      <td>25.0</td>\n",
       "      <td>+</td>\n",
       "      <td>intron</td>\n",
       "      <td>\"pre-mRNA\"</td>\n",
       "      <td>\"ENSG00000187583.10\"</td>\n",
       "      <td>\"PLEKHN1\"</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>chr1</td>\n",
       "      <td>970839</td>\n",
       "      <td>970855</td>\n",
       "      <td>.</td>\n",
       "      <td>20.0</td>\n",
       "      <td>+</td>\n",
       "      <td>CDS</td>\n",
       "      <td>\"mRNA\"</td>\n",
       "      <td>\"ENSG00000187583.10\"</td>\n",
       "      <td>\"PLEKHN1\"</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>chr1</td>\n",
       "      <td>995965</td>\n",
       "      <td>996021</td>\n",
       "      <td>.</td>\n",
       "      <td>48.0</td>\n",
       "      <td>-</td>\n",
       "      <td>ncRNA</td>\n",
       "      <td>\"lncRNA\"</td>\n",
       "      <td>\"ENSG00000272512.1\"</td>\n",
       "      <td>\"AL645608.8\"</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>chr1</td>\n",
       "      <td>996088</td>\n",
       "      <td>996128</td>\n",
       "      <td>.</td>\n",
       "      <td>30.0</td>\n",
       "      <td>-</td>\n",
       "      <td>ncRNA</td>\n",
       "      <td>\"lncRNA\"</td>\n",
       "      <td>\"ENSG00000272512.1\"</td>\n",
       "      <td>\"AL645608.8\"</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>chr1</td>\n",
       "      <td>999284</td>\n",
       "      <td>999323</td>\n",
       "      <td>.</td>\n",
       "      <td>52.0</td>\n",
       "      <td>-</td>\n",
       "      <td>CDS</td>\n",
       "      <td>\"mRNA\"</td>\n",
       "      <td>\"ENSG00000188290.10\"</td>\n",
       "      <td>\"HES4\"</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>chr1</td>\n",
       "      <td>999371</td>\n",
       "      <td>999379</td>\n",
       "      <td>.</td>\n",
       "      <td>55.0</td>\n",
       "      <td>-</td>\n",
       "      <td>CDS</td>\n",
       "      <td>\"mRNA\"</td>\n",
       "      <td>\"ENSG00000188290.10\"</td>\n",
       "      <td>\"HES4\"</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1000161</td>\n",
       "      <td>1000172</td>\n",
       "      <td>.</td>\n",
       "      <td>260.0</td>\n",
       "      <td>-</td>\n",
       "      <td>UTR5</td>\n",
       "      <td>\"mRNA\"</td>\n",
       "      <td>\"ENSG00000188290.10\"</td>\n",
       "      <td>\"HES4\"</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>50 rows × 49 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         chrom    start      end name   score strand      region     biotype  \\\n",
       "0   GL000194.1    75522    75545    .    22.0      -      intron  \"pre-mRNA\"   \n",
       "1   GL000194.1    81461    81479    .    13.0      -      intron  \"pre-mRNA\"   \n",
       "2   GL000194.1    92228    92285    .    68.0      -      intron  \"pre-mRNA\"   \n",
       "3   GL000194.1    99578    99599    .    23.0      -      intron  \"pre-mRNA\"   \n",
       "4   GL000008.2   154464   154475    .   113.0      +  intergenic          \"\"   \n",
       "5   GL000008.2   155093   155130    .   110.0      +  intergenic          \"\"   \n",
       "6   KI270442.1   217328   217337    .    19.0      -       ncRNA      \"rRNA\"   \n",
       "7   KI270711.1     8179     8241    .   520.0      -      intron  \"pre-mRNA\"   \n",
       "8   KI270711.1    11269    11343    .   123.0      -      intron  \"pre-mRNA\"   \n",
       "9   KI270713.1    31706    31707    .    12.0      -        UTR3      \"mRNA\"   \n",
       "10  KI270728.1  1140224  1140264    .   157.0      -        UTR3      \"mRNA\"   \n",
       "11  KI270728.1  1236970  1236995    .    13.0      +      intron  \"pre-mRNA\"   \n",
       "12  KI270733.1   128915   128948    .    15.0      +       ncRNA      \"rRNA\"   \n",
       "13        chr1   280750   280772    .    43.0      -      intron    \"lncRNA\"   \n",
       "14        chr1   286726   286748    .    28.0      -      intron    \"lncRNA\"   \n",
       "15        chr1   517632   517641    .    12.0      -      intron    \"lncRNA\"   \n",
       "16        chr1   783745   783758    .    42.0      +      intron    \"lncRNA\"   \n",
       "17        chr1   783876   783882    .    22.0      +      intron    \"lncRNA\"   \n",
       "18        chr1   787103   787134    .    12.0      +       ncRNA    \"lncRNA\"   \n",
       "19        chr1   788818   788940    .    67.0      +      intron    \"lncRNA\"   \n",
       "20        chr1   788985   789003    .    18.0      +      intron    \"lncRNA\"   \n",
       "21        chr1   789057   789165    .    33.0      +      intron    \"lncRNA\"   \n",
       "22        chr1   789308   789319    .    11.0      +      intron    \"lncRNA\"   \n",
       "23        chr1   789979   790008    .    14.0      +      intron    \"lncRNA\"   \n",
       "24        chr1   790126   790175    .    33.0      +      intron    \"lncRNA\"   \n",
       "25        chr1   790293   790314    .    25.0      +      intron    \"lncRNA\"   \n",
       "26        chr1   791267   791302    .    21.0      +      intron    \"lncRNA\"   \n",
       "27        chr1   808217   808335    .   391.0      -      intron    \"lncRNA\"   \n",
       "28        chr1   819773   819849    .    46.0      -      intron    \"lncRNA\"   \n",
       "29        chr1   855188   855562    .   970.0      +       ncRNA    \"lncRNA\"   \n",
       "30        chr1   907575   907585    .    18.0      +      intron    \"lncRNA\"   \n",
       "31        chr1   911477   911536    .    47.0      +      intron    \"lncRNA\"   \n",
       "32        chr1   911581   911589    .    29.0      +      intron    \"lncRNA\"   \n",
       "33        chr1   917870   917927    .   101.0      -       ncRNA    \"lncRNA\"   \n",
       "34        chr1   917980   918021    .    23.0      -       ncRNA    \"lncRNA\"   \n",
       "35        chr1   926449   926469    .   134.0      +      intron  \"pre-mRNA\"   \n",
       "36        chr1   931389   931422    .   163.0      +      intron  \"pre-mRNA\"   \n",
       "37        chr1   931465   931623    .   902.0      +      intron  \"pre-mRNA\"   \n",
       "38        chr1   950069   950784    .  5158.0      -      intron  \"pre-mRNA\"   \n",
       "39        chr1   962159   962172    .    28.0      +        UTR5      \"mRNA\"   \n",
       "40        chr1   962211   962251    .    33.0      +        UTR5      \"mRNA\"   \n",
       "41        chr1   969305   969434    .    42.0      +      intron  \"pre-mRNA\"   \n",
       "42        chr1   969497   969817    .   119.0      +      intron  \"pre-mRNA\"   \n",
       "43        chr1   969856   969914    .    25.0      +      intron  \"pre-mRNA\"   \n",
       "44        chr1   970839   970855    .    20.0      +         CDS      \"mRNA\"   \n",
       "45        chr1   995965   996021    .    48.0      -       ncRNA    \"lncRNA\"   \n",
       "46        chr1   996088   996128    .    30.0      -       ncRNA    \"lncRNA\"   \n",
       "47        chr1   999284   999323    .    52.0      -         CDS      \"mRNA\"   \n",
       "48        chr1   999371   999379    .    55.0      -         CDS      \"mRNA\"   \n",
       "49        chr1  1000161  1000172    .   260.0      -        UTR5      \"mRNA\"   \n",
       "\n",
       "                 gene_id     gene_name  ...  \\\n",
       "0    \"ENSG00000274847.1\"  \"AC145212.1\"  ...   \n",
       "1    \"ENSG00000274847.1\"  \"AC145212.1\"  ...   \n",
       "2    \"ENSG00000274847.1\"  \"AC145212.1\"  ...   \n",
       "3    \"ENSG00000274847.1\"  \"AC145212.1\"  ...   \n",
       "4                    \".\"        \"None\"  ...   \n",
       "5                    \".\"        \"None\"  ...   \n",
       "6    \"ENSG00000278294.1\"   \"5_8S_rRNA\"  ...   \n",
       "7    \"ENSG00000271254.6\"  \"AC240274.1\"  ...   \n",
       "8    \"ENSG00000271254.6\"  \"AC240274.1\"  ...   \n",
       "9    \"ENSG00000277475.1\"  \"AC213203.1\"  ...   \n",
       "10   \"ENSG00000273554.4\"  \"AC136616.1\"  ...   \n",
       "11   \"ENSG00000277761.1\"  \"AC136616.2\"  ...   \n",
       "12   \"ENSG00000276700.1\"    \"RNA5-8S5\"  ...   \n",
       "13  \"ENSG00000228463.10\"  \"AP006222.1\"  ...   \n",
       "14  \"ENSG00000228463.10\"  \"AP006222.1\"  ...   \n",
       "15  \"ENSG00000237094.12\"  \"AL732372.2\"  ...   \n",
       "16   \"ENSG00000237491.8\"  \"AL669831.5\"  ...   \n",
       "17   \"ENSG00000237491.8\"  \"AL669831.5\"  ...   \n",
       "18   \"ENSG00000237491.8\"  \"AL669831.5\"  ...   \n",
       "19   \"ENSG00000237491.8\"  \"AL669831.5\"  ...   \n",
       "20   \"ENSG00000237491.8\"  \"AL669831.5\"  ...   \n",
       "21   \"ENSG00000237491.8\"  \"AL669831.5\"  ...   \n",
       "22   \"ENSG00000237491.8\"  \"AL669831.5\"  ...   \n",
       "23   \"ENSG00000237491.8\"  \"AL669831.5\"  ...   \n",
       "24   \"ENSG00000237491.8\"  \"AL669831.5\"  ...   \n",
       "25   \"ENSG00000237491.8\"  \"AL669831.5\"  ...   \n",
       "26   \"ENSG00000237491.8\"  \"AL669831.5\"  ...   \n",
       "27   \"ENSG00000230021.9\"  \"AL669831.3\"  ...   \n",
       "28   \"ENSG00000230021.9\"  \"AL669831.3\"  ...   \n",
       "29   \"ENSG00000228794.8\"   \"LINC01128\"  ...   \n",
       "30   \"ENSG00000272438.1\"  \"AL645608.7\"  ...   \n",
       "31   \"ENSG00000272438.1\"  \"AL645608.7\"  ...   \n",
       "32   \"ENSG00000272438.1\"  \"AL645608.7\"  ...   \n",
       "33   \"ENSG00000223764.2\"  \"AL645608.1\"  ...   \n",
       "34   \"ENSG00000223764.2\"  \"AL645608.1\"  ...   \n",
       "35  \"ENSG00000187634.11\"      \"SAMD11\"  ...   \n",
       "36  \"ENSG00000187634.11\"      \"SAMD11\"  ...   \n",
       "37  \"ENSG00000187634.11\"      \"SAMD11\"  ...   \n",
       "38  \"ENSG00000188976.10\"       \"NOC2L\"  ...   \n",
       "39  \"ENSG00000187961.13\"      \"KLHL17\"  ...   \n",
       "40  \"ENSG00000187961.13\"      \"KLHL17\"  ...   \n",
       "41  \"ENSG00000187583.10\"     \"PLEKHN1\"  ...   \n",
       "42  \"ENSG00000187583.10\"     \"PLEKHN1\"  ...   \n",
       "43  \"ENSG00000187583.10\"     \"PLEKHN1\"  ...   \n",
       "44  \"ENSG00000187583.10\"     \"PLEKHN1\"  ...   \n",
       "45   \"ENSG00000272512.1\"  \"AL645608.8\"  ...   \n",
       "46   \"ENSG00000272512.1\"  \"AL645608.8\"  ...   \n",
       "47  \"ENSG00000188290.10\"        \"HES4\"  ...   \n",
       "48  \"ENSG00000188290.10\"        \"HES4\"  ...   \n",
       "49  \"ENSG00000188290.10\"        \"HES4\"  ...   \n",
       "\n",
       "    tardbp-egfp-hek293-hd2-5-4-20180829-ju  \\\n",
       "0                                      2.0   \n",
       "1                                      0.0   \n",
       "2                                      4.0   \n",
       "3                                      0.0   \n",
       "4                                      6.0   \n",
       "5                                      7.0   \n",
       "6                                      2.0   \n",
       "7                                     19.0   \n",
       "8                                      8.0   \n",
       "9                                      0.0   \n",
       "10                                    20.0   \n",
       "11                                     1.0   \n",
       "12                                     3.0   \n",
       "13                                     1.0   \n",
       "14                                     3.0   \n",
       "15                                     0.0   \n",
       "16                                     2.0   \n",
       "17                                     1.0   \n",
       "18                                     0.0   \n",
       "19                                     3.0   \n",
       "20                                     0.0   \n",
       "21                                     2.0   \n",
       "22                                     2.0   \n",
       "23                                     2.0   \n",
       "24                                     1.0   \n",
       "25                                     0.0   \n",
       "26                                     0.0   \n",
       "27                                    20.0   \n",
       "28                                     3.0   \n",
       "29                                   102.0   \n",
       "30                                     0.0   \n",
       "31                                     0.0   \n",
       "32                                     1.0   \n",
       "33                                     2.0   \n",
       "34                                     0.0   \n",
       "35                                    11.0   \n",
       "36                                    14.0   \n",
       "37                                    48.0   \n",
       "38                                   310.0   \n",
       "39                                     0.0   \n",
       "40                                     0.0   \n",
       "41                                     0.0   \n",
       "42                                    11.0   \n",
       "43                                     0.0   \n",
       "44                                     0.0   \n",
       "45                                     0.0   \n",
       "46                                     0.0   \n",
       "47                                     0.0   \n",
       "48                                     0.0   \n",
       "49                                     0.0   \n",
       "\n",
       "    tardbp-g294a-egfp-hek293-4-20181001-ju  tardbp-egfp-293flp-l2-20171010-ju  \\\n",
       "0                                      0.0                                0.0   \n",
       "1                                      0.0                                0.0   \n",
       "2                                      0.0                                0.0   \n",
       "3                                      0.0                                1.0   \n",
       "4                                      2.0                                1.0   \n",
       "5                                      1.0                                0.0   \n",
       "6                                      0.0                                2.0   \n",
       "7                                     14.0                                3.0   \n",
       "8                                      0.0                                0.0   \n",
       "9                                      0.0                                0.0   \n",
       "10                                     0.0                                0.0   \n",
       "11                                     0.0                                0.0   \n",
       "12                                     1.0                                0.0   \n",
       "13                                     1.0                                1.0   \n",
       "14                                     1.0                                0.0   \n",
       "15                                     0.0                                0.0   \n",
       "16                                     0.0                                1.0   \n",
       "17                                     0.0                                0.0   \n",
       "18                                     0.0                                0.0   \n",
       "19                                     1.0                                0.0   \n",
       "20                                     0.0                                0.0   \n",
       "21                                     4.0                                0.0   \n",
       "22                                     0.0                                0.0   \n",
       "23                                     1.0                                1.0   \n",
       "24                                     0.0                                0.0   \n",
       "25                                     1.0                                0.0   \n",
       "26                                     1.0                                0.0   \n",
       "27                                    10.0                                6.0   \n",
       "28                                     0.0                                0.0   \n",
       "29                                    22.0                                7.0   \n",
       "30                                     0.0                                0.0   \n",
       "31                                     2.0                                1.0   \n",
       "32                                     0.0                                1.0   \n",
       "33                                     5.0                                0.0   \n",
       "34                                     1.0                                0.0   \n",
       "35                                     4.0                                2.0   \n",
       "36                                     1.0                                0.0   \n",
       "37                                    24.0                               11.0   \n",
       "38                                    92.0                               68.0   \n",
       "39                                     0.0                                0.0   \n",
       "40                                     0.0                                0.0   \n",
       "41                                     1.0                                0.0   \n",
       "42                                     3.0                                4.0   \n",
       "43                                     0.0                                0.0   \n",
       "44                                     0.0                                0.0   \n",
       "45                                     3.0                                0.0   \n",
       "46                                     0.0                                0.0   \n",
       "47                                     2.0                                1.0   \n",
       "48                                     0.0                                0.0   \n",
       "49                                     0.0                                0.0   \n",
       "\n",
       "    tardbp-egfp-293flp-l1-20171010-ju  \\\n",
       "0                                 0.0   \n",
       "1                                 0.0   \n",
       "2                                 0.0   \n",
       "3                                 2.0   \n",
       "4                                 0.0   \n",
       "5                                 2.0   \n",
       "6                                 1.0   \n",
       "7                                 4.0   \n",
       "8                                 0.0   \n",
       "9                                 0.0   \n",
       "10                                0.0   \n",
       "11                                0.0   \n",
       "12                                0.0   \n",
       "13                                0.0   \n",
       "14                                0.0   \n",
       "15                                0.0   \n",
       "16                                0.0   \n",
       "17                                0.0   \n",
       "18                                0.0   \n",
       "19                                0.0   \n",
       "20                                0.0   \n",
       "21                                0.0   \n",
       "22                                0.0   \n",
       "23                                0.0   \n",
       "24                                0.0   \n",
       "25                                0.0   \n",
       "26                                0.0   \n",
       "27                                2.0   \n",
       "28                                0.0   \n",
       "29                                8.0   \n",
       "30                                0.0   \n",
       "31                                4.0   \n",
       "32                                0.0   \n",
       "33                                1.0   \n",
       "34                                0.0   \n",
       "35                                1.0   \n",
       "36                                0.0   \n",
       "37                               12.0   \n",
       "38                               78.0   \n",
       "39                                1.0   \n",
       "40                                1.0   \n",
       "41                                0.0   \n",
       "42                                2.0   \n",
       "43                                0.0   \n",
       "44                                0.0   \n",
       "45                                1.0   \n",
       "46                                1.0   \n",
       "47                                0.0   \n",
       "48                                0.0   \n",
       "49                                0.0   \n",
       "\n",
       "    tardbp-316del346-egfp-293flp-l1-20171010-ju  \\\n",
       "0                                           0.0   \n",
       "1                                           0.0   \n",
       "2                                           0.0   \n",
       "3                                           0.0   \n",
       "4                                           0.0   \n",
       "5                                           0.0   \n",
       "6                                           0.0   \n",
       "7                                           1.0   \n",
       "8                                           0.0   \n",
       "9                                           0.0   \n",
       "10                                          0.0   \n",
       "11                                          0.0   \n",
       "12                                          0.0   \n",
       "13                                          0.0   \n",
       "14                                          0.0   \n",
       "15                                          0.0   \n",
       "16                                          2.0   \n",
       "17                                          0.0   \n",
       "18                                          0.0   \n",
       "19                                          0.0   \n",
       "20                                          0.0   \n",
       "21                                          0.0   \n",
       "22                                          0.0   \n",
       "23                                          0.0   \n",
       "24                                          0.0   \n",
       "25                                          0.0   \n",
       "26                                          0.0   \n",
       "27                                          1.0   \n",
       "28                                          1.0   \n",
       "29                                          3.0   \n",
       "30                                          0.0   \n",
       "31                                          0.0   \n",
       "32                                          0.0   \n",
       "33                                          1.0   \n",
       "34                                          0.0   \n",
       "35                                          0.0   \n",
       "36                                          2.0   \n",
       "37                                          0.0   \n",
       "38                                         36.0   \n",
       "39                                          0.0   \n",
       "40                                          0.0   \n",
       "41                                          0.0   \n",
       "42                                          0.0   \n",
       "43                                          0.0   \n",
       "44                                          0.0   \n",
       "45                                          1.0   \n",
       "46                                          0.0   \n",
       "47                                          0.0   \n",
       "48                                          1.0   \n",
       "49                                          0.0   \n",
       "\n",
       "    tardbp-316del346-egfp-293flp-l2-20171010-ju  \\\n",
       "0                                           0.0   \n",
       "1                                           0.0   \n",
       "2                                           0.0   \n",
       "3                                           3.0   \n",
       "4                                           0.0   \n",
       "5                                           1.0   \n",
       "6                                           0.0   \n",
       "7                                           1.0   \n",
       "8                                           0.0   \n",
       "9                                           0.0   \n",
       "10                                          1.0   \n",
       "11                                          0.0   \n",
       "12                                          0.0   \n",
       "13                                          0.0   \n",
       "14                                          0.0   \n",
       "15                                          0.0   \n",
       "16                                          0.0   \n",
       "17                                          0.0   \n",
       "18                                          0.0   \n",
       "19                                          1.0   \n",
       "20                                          0.0   \n",
       "21                                          0.0   \n",
       "22                                          0.0   \n",
       "23                                          0.0   \n",
       "24                                          1.0   \n",
       "25                                          0.0   \n",
       "26                                          0.0   \n",
       "27                                          2.0   \n",
       "28                                          1.0   \n",
       "29                                          1.0   \n",
       "30                                          0.0   \n",
       "31                                          0.0   \n",
       "32                                          0.0   \n",
       "33                                          2.0   \n",
       "34                                          0.0   \n",
       "35                                          3.0   \n",
       "36                                          4.0   \n",
       "37                                          5.0   \n",
       "38                                         50.0   \n",
       "39                                          0.0   \n",
       "40                                          0.0   \n",
       "41                                          0.0   \n",
       "42                                          0.0   \n",
       "43                                          0.0   \n",
       "44                                          0.0   \n",
       "45                                          0.0   \n",
       "46                                          0.0   \n",
       "47                                          1.0   \n",
       "48                                          1.0   \n",
       "49                                          0.0   \n",
       "\n",
       "    tardbp-egfp-293flp-m1-20171010-ju  \\\n",
       "0                                 0.0   \n",
       "1                                 0.0   \n",
       "2                                 1.0   \n",
       "3                                 1.0   \n",
       "4                                 2.0   \n",
       "5                                 1.0   \n",
       "6                                 0.0   \n",
       "7                                 4.0   \n",
       "8                                 0.0   \n",
       "9                                 0.0   \n",
       "10                                0.0   \n",
       "11                                0.0   \n",
       "12                                0.0   \n",
       "13                                0.0   \n",
       "14                                0.0   \n",
       "15                                0.0   \n",
       "16                                2.0   \n",
       "17                                0.0   \n",
       "18                                0.0   \n",
       "19                                3.0   \n",
       "20                                0.0   \n",
       "21                                0.0   \n",
       "22                                0.0   \n",
       "23                                0.0   \n",
       "24                                0.0   \n",
       "25                                0.0   \n",
       "26                                0.0   \n",
       "27                                4.0   \n",
       "28                                0.0   \n",
       "29                               14.0   \n",
       "30                                0.0   \n",
       "31                                1.0   \n",
       "32                                0.0   \n",
       "33                                2.0   \n",
       "34                                1.0   \n",
       "35                                2.0   \n",
       "36                                1.0   \n",
       "37                               10.0   \n",
       "38                               53.0   \n",
       "39                                0.0   \n",
       "40                                0.0   \n",
       "41                                0.0   \n",
       "42                                2.0   \n",
       "43                                1.0   \n",
       "44                                0.0   \n",
       "45                                2.0   \n",
       "46                                1.0   \n",
       "47                                1.0   \n",
       "48                                1.0   \n",
       "49                                0.0   \n",
       "\n",
       "    tardbp-316del346-egfp-293flp-m2-20171010-ju  \\\n",
       "0                                           0.0   \n",
       "1                                           0.0   \n",
       "2                                           0.0   \n",
       "3                                           1.0   \n",
       "4                                           0.0   \n",
       "5                                           0.0   \n",
       "6                                           2.0   \n",
       "7                                           0.0   \n",
       "8                                           0.0   \n",
       "9                                           0.0   \n",
       "10                                          0.0   \n",
       "11                                          0.0   \n",
       "12                                          1.0   \n",
       "13                                          0.0   \n",
       "14                                          1.0   \n",
       "15                                          0.0   \n",
       "16                                          2.0   \n",
       "17                                          0.0   \n",
       "18                                          0.0   \n",
       "19                                          0.0   \n",
       "20                                          0.0   \n",
       "21                                          0.0   \n",
       "22                                          0.0   \n",
       "23                                          0.0   \n",
       "24                                          0.0   \n",
       "25                                          0.0   \n",
       "26                                          0.0   \n",
       "27                                          0.0   \n",
       "28                                          0.0   \n",
       "29                                          1.0   \n",
       "30                                          0.0   \n",
       "31                                          0.0   \n",
       "32                                          1.0   \n",
       "33                                          1.0   \n",
       "34                                          0.0   \n",
       "35                                          4.0   \n",
       "36                                          0.0   \n",
       "37                                          0.0   \n",
       "38                                         35.0   \n",
       "39                                          0.0   \n",
       "40                                          0.0   \n",
       "41                                          0.0   \n",
       "42                                          0.0   \n",
       "43                                          1.0   \n",
       "44                                          0.0   \n",
       "45                                          0.0   \n",
       "46                                          0.0   \n",
       "47                                          0.0   \n",
       "48                                          2.0   \n",
       "49                                          0.0   \n",
       "\n",
       "    tardbp-316del346-egfp-293flp-m1-20171010-ju  \\\n",
       "0                                           0.0   \n",
       "1                                           0.0   \n",
       "2                                           0.0   \n",
       "3                                           1.0   \n",
       "4                                           0.0   \n",
       "5                                           1.0   \n",
       "6                                           0.0   \n",
       "7                                           2.0   \n",
       "8                                           0.0   \n",
       "9                                           0.0   \n",
       "10                                          0.0   \n",
       "11                                          0.0   \n",
       "12                                          0.0   \n",
       "13                                          0.0   \n",
       "14                                          0.0   \n",
       "15                                          0.0   \n",
       "16                                          0.0   \n",
       "17                                          1.0   \n",
       "18                                          1.0   \n",
       "19                                          0.0   \n",
       "20                                          0.0   \n",
       "21                                          0.0   \n",
       "22                                          0.0   \n",
       "23                                          0.0   \n",
       "24                                          0.0   \n",
       "25                                          0.0   \n",
       "26                                          0.0   \n",
       "27                                          1.0   \n",
       "28                                          0.0   \n",
       "29                                          0.0   \n",
       "30                                          0.0   \n",
       "31                                          0.0   \n",
       "32                                          0.0   \n",
       "33                                          0.0   \n",
       "34                                          0.0   \n",
       "35                                          0.0   \n",
       "36                                          1.0   \n",
       "37                                          3.0   \n",
       "38                                         35.0   \n",
       "39                                          0.0   \n",
       "40                                          0.0   \n",
       "41                                          0.0   \n",
       "42                                          0.0   \n",
       "43                                          0.0   \n",
       "44                                          0.0   \n",
       "45                                          0.0   \n",
       "46                                          1.0   \n",
       "47                                          0.0   \n",
       "48                                          0.0   \n",
       "49                                          0.0   \n",
       "\n",
       "    tardbp-egfp-293flp-m2-20171010-ju  \n",
       "0                                 1.0  \n",
       "1                                 0.0  \n",
       "2                                 0.0  \n",
       "3                                 0.0  \n",
       "4                                 2.0  \n",
       "5                                 1.0  \n",
       "6                                 3.0  \n",
       "7                                 8.0  \n",
       "8                                 1.0  \n",
       "9                                 0.0  \n",
       "10                                1.0  \n",
       "11                                0.0  \n",
       "12                                0.0  \n",
       "13                                0.0  \n",
       "14                                1.0  \n",
       "15                                0.0  \n",
       "16                                1.0  \n",
       "17                                0.0  \n",
       "18                                0.0  \n",
       "19                                0.0  \n",
       "20                                0.0  \n",
       "21                                0.0  \n",
       "22                                0.0  \n",
       "23                                0.0  \n",
       "24                                0.0  \n",
       "25                                0.0  \n",
       "26                                0.0  \n",
       "27                                5.0  \n",
       "28                                1.0  \n",
       "29                               13.0  \n",
       "30                                0.0  \n",
       "31                                1.0  \n",
       "32                                0.0  \n",
       "33                                1.0  \n",
       "34                                1.0  \n",
       "35                                4.0  \n",
       "36                                1.0  \n",
       "37                               11.0  \n",
       "38                               84.0  \n",
       "39                                1.0  \n",
       "40                                1.0  \n",
       "41                                1.0  \n",
       "42                                2.0  \n",
       "43                                0.0  \n",
       "44                                0.0  \n",
       "45                                0.0  \n",
       "46                                0.0  \n",
       "47                                0.0  \n",
       "48                                0.0  \n",
       "49                                0.0  \n",
       "\n",
       "[50 rows x 49 columns]"
      ]
     },
     "execution_count": 58,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_out_rfiltered_merged30.head(50)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['chrom', 'start', 'end', 'name', 'score', 'strand', 'region', 'biotype',\n",
       "       'gene_id', 'gene_name',\n",
       "       'tardbp-316del346-egfp-hek293-hd1-6-4-20180829-ju',\n",
       "       'tardbp-egfp-hek293-hd2-5-3-20180829-ju',\n",
       "       'tardbp-egfp-hek293-hd2-5-on-bead-rt-1-20180829-ju',\n",
       "       'tardbp-316del346-egfp-hek293-hd2-5-3-20180829-ju',\n",
       "       'tardbp-egfp-hek293-3-20181001-ju',\n",
       "       'tardbp-q331k-egfp-hek293-no-uv-1-20181001-ju',\n",
       "       'tardbp-320del414-egfp-hek293-hd1-6-on-bead-rt-1-20180829-ju',\n",
       "       'tardbp-316del346-egfp-hek293-4-20181001-ju',\n",
       "       'tardbp-egfp-hek293-no-uv-1-20180829-ju',\n",
       "       'tardbp-egfp-hek293-4-20181001-ju',\n",
       "       'tardbp-egfp-hek293-hd1-6-4-20180829-ju',\n",
       "       'tardbp-a326p-egfp-hek293-4-20181001-ju',\n",
       "       'tardbp-egfp-hek293-hd1-6-3-20180829-ju',\n",
       "       'tardbp-320del414-egfp-hek293-hd1-6-4-20180829-ju',\n",
       "       'tardbp-q331k-egfp-hek293-3-20181001-ju',\n",
       "       'tardbp-316del346-egfp-hek293-hd1-6-3-20180829-ju',\n",
       "       'tardbp-320del414-egfp-hek293-hd1-6-3-20180829-ju',\n",
       "       'tardbp-egfp-hek293-hd1-6-on-bead-rt-1-20180829-ju',\n",
       "       'tardbp-316del346-egfp-hek293-hd1-6-on-bead-rt-1-20180829-ju',\n",
       "       'tardbp-m337p-egfp-hek293-4-20181001-ju',\n",
       "       'tardbp-m337p-egfp-hek293-3-20181001-ju',\n",
       "       'tardbp-a326p-egfp-hek293-3-20181001-ju',\n",
       "       'tardbp-316del346-egfp-hek293-hd2-5-4-20180829-ju',\n",
       "       'tardbp-q331k-egfp-hek293-4-20181001-ju',\n",
       "       'tardbp-g335a-egfp-hek293-3-20181001-ju',\n",
       "       'tardbp-g335a-egfp-hek293-4-20181001-ju',\n",
       "       'tardbp-g294a-egfp-hek293-3-20181001-ju',\n",
       "       'tardbp-316del346-egfp-hek293-hd2-5-on-bead-rt-1-20180829-ju',\n",
       "       'tardbp-316del346-egfp-hek293-3-20181001-ju',\n",
       "       'tardbp-egfp-hek293-hd2-5-4-20180829-ju',\n",
       "       'tardbp-g294a-egfp-hek293-4-20181001-ju',\n",
       "       'tardbp-egfp-293flp-l2-20171010-ju',\n",
       "       'tardbp-egfp-293flp-l1-20171010-ju',\n",
       "       'tardbp-316del346-egfp-293flp-l1-20171010-ju',\n",
       "       'tardbp-316del346-egfp-293flp-l2-20171010-ju',\n",
       "       'tardbp-egfp-293flp-m1-20171010-ju',\n",
       "       'tardbp-316del346-egfp-293flp-m2-20171010-ju',\n",
       "       'tardbp-316del346-egfp-293flp-m1-20171010-ju',\n",
       "       'tardbp-egfp-293flp-m2-20171010-ju'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 59,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_out_rfiltered_merged30.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "122170"
      ]
     },
     "execution_count": 60,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(df_out_rfiltered_merged30)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [],
   "source": [
    "# df_out_rfiltered_merged30[['chrom', 'start', 'end', 'name', 'score', 'strand', 'region', 'biotype',\n",
    "#        'gene_id', 'gene_name']].to_csv('rfiltered_tdp_bs.tsv', sep='\\t', index=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_out_rfiltered_merged30['len'] = df_out_rfiltered_merged30.end - df_out_rfiltered_merged30.start"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [],
   "source": [
    "group1_raw = pbt.BedTool(\n",
    "    '/home/aram/Dropbox (The Francis Crick)/ucl/aram/martina_tdp43/binding_sites_set_1and2/genome_larger_motif_groups/set1_2_merged0_TGTGTG_GTGTGT_CGCGCG.bed')\n",
    "group2_raw = pbt.BedTool(\n",
    "    '/home/aram/Dropbox (The Francis Crick)/ucl/aram/martina_tdp43/binding_sites_set_1and2/genome_larger_motif_groups/set1_2_merged0_GTGCGC_CGTGTG_TGTGCG.bed')\n",
    "group3_raw = pbt.BedTool(\n",
    "    '/home/aram/Dropbox (The Francis Crick)/ucl/aram/martina_tdp43/binding_sites_set_1and2/genome_larger_motif_groups/set1_2_merged0_GTCTGT_TCTGTG_TGTGTC.bed')\n",
    "group4_raw = pbt.BedTool(\n",
    "    '/home/aram/Dropbox (The Francis Crick)/ucl/aram/martina_tdp43/binding_sites_set_1and2/genome_larger_motif_groups/set1_2_merged0_ATGTGA_GACTGT_TGTGAT.bed')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "***** WARNING: File /tmp/pybedtools.60okb10c.tmp has inconsistent naming convention for record:\n",
      "chr1\t280750\t280772\t.\t43\t-\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.60okb10c.tmp has inconsistent naming convention for record:\n",
      "chr1\t280750\t280772\t.\t43\t-\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.60okb10c.tmp has inconsistent naming convention for record:\n",
      "chr1\t280750\t280772\t.\t43\t-\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.60okb10c.tmp has inconsistent naming convention for record:\n",
      "chr1\t280750\t280772\t.\t43\t-\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.60okb10c.tmp has inconsistent naming convention for record:\n",
      "chr1\t280750\t280772\t.\t43\t-\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.60okb10c.tmp has inconsistent naming convention for record:\n",
      "chr1\t280750\t280772\t.\t43\t-\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.60okb10c.tmp has inconsistent naming convention for record:\n",
      "chr1\t280750\t280772\t.\t43\t-\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.60okb10c.tmp has inconsistent naming convention for record:\n",
      "chr1\t280750\t280772\t.\t43\t-\n",
      "\n"
     ]
    }
   ],
   "source": [
    "group1 = group1_raw.intersect(rfiltered_merged30, s=True)\n",
    "group2 = group2_raw.intersect(rfiltered_merged30, s=True)\n",
    "group3 = group3_raw.intersect(rfiltered_merged30, s=True)\n",
    "group4 = group4_raw.intersect(rfiltered_merged30, s=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "***** WARNING: File /tmp/pybedtools.dw0tp997.tmp has inconsistent naming convention for record:\n",
      "chr1\t280750\t280760\t.\t120\t-\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.dw0tp997.tmp has inconsistent naming convention for record:\n",
      "chr1\t280750\t280760\t.\t120\t-\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.dw0tp997.tmp has inconsistent naming convention for record:\n",
      "chr1\t280750\t280760\t.\t120\t-\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.dw0tp997.tmp has inconsistent naming convention for record:\n",
      "chr1\t280750\t280760\t.\t120\t-\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.1hwqkfy4.tmp has inconsistent naming convention for record:\n",
      "chr1\t280755\t280763\t.\t52\t-\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.1hwqkfy4.tmp has inconsistent naming convention for record:\n",
      "chr1\t280755\t280763\t.\t52\t-\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.j3y0zuml.tmp has inconsistent naming convention for record:\n",
      "chr1\t286745\t286748\t.\t6\t-\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.j3y0zuml.tmp has inconsistent naming convention for record:\n",
      "chr1\t286745\t286748\t.\t6\t-\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.dw0tp997.tmp has inconsistent naming convention for record:\n",
      "chr1\t280750\t280760\t.\t120\t-\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.dw0tp997.tmp has inconsistent naming convention for record:\n",
      "chr1\t280750\t280760\t.\t120\t-\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.1hwqkfy4.tmp has inconsistent naming convention for record:\n",
      "chr1\t280755\t280763\t.\t52\t-\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.1hwqkfy4.tmp has inconsistent naming convention for record:\n",
      "chr1\t280755\t280763\t.\t52\t-\n",
      "\n"
     ]
    }
   ],
   "source": [
    "group2_filtered = group2.intersect(group1, s=True, v=True)\n",
    "group3_filtered = group3.intersect(group1, s=True, v=True).intersect(group2, s=True, v=True).intersect(\n",
    "    group4, s=True, v=True)\n",
    "group4_filtered = group4.intersect(group1, s=True, v=True).intersect(group2, s=True, v=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "***** WARNING: File /tmp/pybedtools.dw0tp997.tmp has inconsistent naming convention for record:\n",
      "chr1\t280750\t280760\t.\t120\t-\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.dw0tp997.tmp has inconsistent naming convention for record:\n",
      "chr1\t280750\t280760\t.\t120\t-\n",
      "\n"
     ]
    }
   ],
   "source": [
    "intersect_1_unmerged = rfiltered_merged30.intersect(group1,\n",
    "    s=True,\n",
    "    #wa=True,\n",
    "    wao=True,\n",
    "    #loj=True\n",
    "    )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {},
   "outputs": [],
   "source": [
    "def merge_peaks_temp(peaks, n):\n",
    "    return peaks.sort().merge(d=n, s=True, c=[4, 5, 6, 13], o=['distinct', 'distinct', 'distinct', 'max'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {},
   "outputs": [],
   "source": [
    "intersect_1 = merge_peaks_temp(intersect_1_unmerged, 0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "***** WARNING: File /tmp/pybedtools.u_0zd1o4.tmp has inconsistent naming convention for record:\n",
      "chr1\t808226\t808232\t.\t35\t-\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.u_0zd1o4.tmp has inconsistent naming convention for record:\n",
      "chr1\t808226\t808232\t.\t35\t-\n",
      "\n"
     ]
    }
   ],
   "source": [
    "intersect_2_unmerged = rfiltered_merged30.intersect(group2_filtered,\n",
    "    s=True,\n",
    "    #wa=True,\n",
    "    wao=True,\n",
    "    #loj=True\n",
    "    )\n",
    "intersect_2 = merge_peaks_temp(intersect_2_unmerged, 0)\n",
    "df_intersect2 = intersect_2.to_dataframe(names=['chrom', 'start', 'end', 'name', 'score', 'strand', 'overlap_g2'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "***** WARNING: File /tmp/pybedtools.8yu9akq3.tmp has inconsistent naming convention for record:\n",
      "chr1\t280766\t280772\t.\t5\t-\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.8yu9akq3.tmp has inconsistent naming convention for record:\n",
      "chr1\t280766\t280772\t.\t5\t-\n",
      "\n"
     ]
    }
   ],
   "source": [
    "intersect_3_unmerged = rfiltered_merged30.intersect(group3_filtered,\n",
    "    s=True,\n",
    "    #wa=True,\n",
    "    wao=True,\n",
    "    #loj=True\n",
    "    )\n",
    "intersect_3 = merge_peaks_temp(intersect_3_unmerged, 0)\n",
    "df_intersect3 = intersect_3.to_dataframe(names=['chrom', 'start', 'end', 'name', 'score', 'strand', 'overlap_g3'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "***** WARNING: File /tmp/pybedtools.7k2cb7ec.tmp has inconsistent naming convention for record:\n",
      "chr1\t286745\t286748\t.\t6\t-\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.7k2cb7ec.tmp has inconsistent naming convention for record:\n",
      "chr1\t286745\t286748\t.\t6\t-\n",
      "\n"
     ]
    }
   ],
   "source": [
    "intersect_4_unmerged = rfiltered_merged30.intersect(group4_filtered,\n",
    "    s=True,\n",
    "    #wa=True,\n",
    "    wao=True,\n",
    "    #loj=True\n",
    "    )\n",
    "intersect_4 = merge_peaks_temp(intersect_4_unmerged, 0)\n",
    "df_intersect4 = intersect_4.to_dataframe(names=['chrom', 'start', 'end', 'name', 'score', 'strand', 'overlap_g4'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_intersect1 = intersect_1.to_dataframe(names=['chrom', 'start', 'end', 'name', 'score', 'strand', 'overlap_g1'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_out_rfiltered_merged30_inner = pd.merge(df_out_rfiltered_merged30, df_intersect1, on=[\n",
    "    'chrom', 'start', 'end', 'name', 'score', 'strand'], how='inner')\n",
    "df_out_rfiltered_merged30_inner = pd.merge(df_out_rfiltered_merged30_inner, df_intersect2, on=[\n",
    "    'chrom', 'start', 'end', 'name', 'score', 'strand'], how='inner')\n",
    "df_out_rfiltered_merged30_inner = pd.merge(df_out_rfiltered_merged30_inner, df_intersect3, on=[\n",
    "    'chrom', 'start', 'end', 'name', 'score', 'strand'], how='inner')\n",
    "df_out_rfiltered_merged30_inner = pd.merge(df_out_rfiltered_merged30_inner, df_intersect4, on=[\n",
    "    'chrom', 'start', 'end', 'name', 'score', 'strand'], how='inner')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_out_rfiltered_merged30_inner['len'] = df_out_rfiltered_merged30_inner.end - df_out_rfiltered_merged30_inner.start"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_out_rfiltered_merged30_inner['overlap_g1'] = df_out_rfiltered_merged30_inner['overlap_g1'] * 100 / df_out_rfiltered_merged30_inner.len\n",
    "df_out_rfiltered_merged30_inner['overlap_g2'] = df_out_rfiltered_merged30_inner['overlap_g2'] * 100 / df_out_rfiltered_merged30_inner.len\n",
    "df_out_rfiltered_merged30_inner['overlap_g3'] = df_out_rfiltered_merged30_inner['overlap_g3'] * 100 / df_out_rfiltered_merged30_inner.len\n",
    "df_out_rfiltered_merged30_inner['overlap_g4'] = df_out_rfiltered_merged30_inner['overlap_g4'] * 100 / df_out_rfiltered_merged30_inner.len"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['chrom', 'start', 'end', 'name', 'score', 'strand', 'region', 'biotype',\n",
       "       'gene_id', 'gene_name',\n",
       "       'tardbp-316del346-egfp-hek293-hd1-6-4-20180829-ju',\n",
       "       'tardbp-egfp-hek293-hd2-5-3-20180829-ju',\n",
       "       'tardbp-egfp-hek293-hd2-5-on-bead-rt-1-20180829-ju',\n",
       "       'tardbp-316del346-egfp-hek293-hd2-5-3-20180829-ju',\n",
       "       'tardbp-egfp-hek293-3-20181001-ju',\n",
       "       'tardbp-q331k-egfp-hek293-no-uv-1-20181001-ju',\n",
       "       'tardbp-320del414-egfp-hek293-hd1-6-on-bead-rt-1-20180829-ju',\n",
       "       'tardbp-316del346-egfp-hek293-4-20181001-ju',\n",
       "       'tardbp-egfp-hek293-no-uv-1-20180829-ju',\n",
       "       'tardbp-egfp-hek293-4-20181001-ju',\n",
       "       'tardbp-egfp-hek293-hd1-6-4-20180829-ju',\n",
       "       'tardbp-a326p-egfp-hek293-4-20181001-ju',\n",
       "       'tardbp-egfp-hek293-hd1-6-3-20180829-ju',\n",
       "       'tardbp-320del414-egfp-hek293-hd1-6-4-20180829-ju',\n",
       "       'tardbp-q331k-egfp-hek293-3-20181001-ju',\n",
       "       'tardbp-316del346-egfp-hek293-hd1-6-3-20180829-ju',\n",
       "       'tardbp-320del414-egfp-hek293-hd1-6-3-20180829-ju',\n",
       "       'tardbp-egfp-hek293-hd1-6-on-bead-rt-1-20180829-ju',\n",
       "       'tardbp-316del346-egfp-hek293-hd1-6-on-bead-rt-1-20180829-ju',\n",
       "       'tardbp-m337p-egfp-hek293-4-20181001-ju',\n",
       "       'tardbp-m337p-egfp-hek293-3-20181001-ju',\n",
       "       'tardbp-a326p-egfp-hek293-3-20181001-ju',\n",
       "       'tardbp-316del346-egfp-hek293-hd2-5-4-20180829-ju',\n",
       "       'tardbp-q331k-egfp-hek293-4-20181001-ju',\n",
       "       'tardbp-g335a-egfp-hek293-3-20181001-ju',\n",
       "       'tardbp-g335a-egfp-hek293-4-20181001-ju',\n",
       "       'tardbp-g294a-egfp-hek293-3-20181001-ju',\n",
       "       'tardbp-316del346-egfp-hek293-hd2-5-on-bead-rt-1-20180829-ju',\n",
       "       'tardbp-316del346-egfp-hek293-3-20181001-ju',\n",
       "       'tardbp-egfp-hek293-hd2-5-4-20180829-ju',\n",
       "       'tardbp-g294a-egfp-hek293-4-20181001-ju',\n",
       "       'tardbp-egfp-293flp-l2-20171010-ju',\n",
       "       'tardbp-egfp-293flp-l1-20171010-ju',\n",
       "       'tardbp-316del346-egfp-293flp-l1-20171010-ju',\n",
       "       'tardbp-316del346-egfp-293flp-l2-20171010-ju',\n",
       "       'tardbp-egfp-293flp-m1-20171010-ju',\n",
       "       'tardbp-316del346-egfp-293flp-m2-20171010-ju',\n",
       "       'tardbp-316del346-egfp-293flp-m1-20171010-ju',\n",
       "       'tardbp-egfp-293flp-m2-20171010-ju', 'len', 'overlap_g1', 'overlap_g2',\n",
       "       'overlap_g3', 'overlap_g4'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 76,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_out_rfiltered_merged30_inner.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>name</th>\n",
       "      <th>score</th>\n",
       "      <th>strand</th>\n",
       "      <th>region</th>\n",
       "      <th>biotype</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>...</th>\n",
       "      <th>tardbp-316del346-egfp-293flp-l2-20171010-ju</th>\n",
       "      <th>tardbp-egfp-293flp-m1-20171010-ju</th>\n",
       "      <th>tardbp-316del346-egfp-293flp-m2-20171010-ju</th>\n",
       "      <th>tardbp-316del346-egfp-293flp-m1-20171010-ju</th>\n",
       "      <th>tardbp-egfp-293flp-m2-20171010-ju</th>\n",
       "      <th>len</th>\n",
       "      <th>overlap_g1</th>\n",
       "      <th>overlap_g2</th>\n",
       "      <th>overlap_g3</th>\n",
       "      <th>overlap_g4</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GL000194.1</td>\n",
       "      <td>75522</td>\n",
       "      <td>75545</td>\n",
       "      <td>.</td>\n",
       "      <td>22.0</td>\n",
       "      <td>-</td>\n",
       "      <td>intron</td>\n",
       "      <td>\"pre-mRNA\"</td>\n",
       "      <td>\"ENSG00000274847.1\"</td>\n",
       "      <td>\"AC145212.1\"</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>23</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>100.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GL000194.1</td>\n",
       "      <td>81461</td>\n",
       "      <td>81479</td>\n",
       "      <td>.</td>\n",
       "      <td>13.0</td>\n",
       "      <td>-</td>\n",
       "      <td>intron</td>\n",
       "      <td>\"pre-mRNA\"</td>\n",
       "      <td>\"ENSG00000274847.1\"</td>\n",
       "      <td>\"AC145212.1\"</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>18</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>22.222222</td>\n",
       "      <td>33.333333</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GL000194.1</td>\n",
       "      <td>92228</td>\n",
       "      <td>92285</td>\n",
       "      <td>.</td>\n",
       "      <td>68.0</td>\n",
       "      <td>-</td>\n",
       "      <td>intron</td>\n",
       "      <td>\"pre-mRNA\"</td>\n",
       "      <td>\"ENSG00000274847.1\"</td>\n",
       "      <td>\"AC145212.1\"</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>57</td>\n",
       "      <td>98.245614</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GL000194.1</td>\n",
       "      <td>99578</td>\n",
       "      <td>99599</td>\n",
       "      <td>.</td>\n",
       "      <td>23.0</td>\n",
       "      <td>-</td>\n",
       "      <td>intron</td>\n",
       "      <td>\"pre-mRNA\"</td>\n",
       "      <td>\"ENSG00000274847.1\"</td>\n",
       "      <td>\"AC145212.1\"</td>\n",
       "      <td>...</td>\n",
       "      <td>3.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>21</td>\n",
       "      <td>47.619048</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GL000008.2</td>\n",
       "      <td>154464</td>\n",
       "      <td>154475</td>\n",
       "      <td>.</td>\n",
       "      <td>113.0</td>\n",
       "      <td>+</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>\"\"</td>\n",
       "      <td>\".\"</td>\n",
       "      <td>\"None\"</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>11</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>100.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 54 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        chrom   start     end name  score strand      region     biotype  \\\n",
       "0  GL000194.1   75522   75545    .   22.0      -      intron  \"pre-mRNA\"   \n",
       "1  GL000194.1   81461   81479    .   13.0      -      intron  \"pre-mRNA\"   \n",
       "2  GL000194.1   92228   92285    .   68.0      -      intron  \"pre-mRNA\"   \n",
       "3  GL000194.1   99578   99599    .   23.0      -      intron  \"pre-mRNA\"   \n",
       "4  GL000008.2  154464  154475    .  113.0      +  intergenic          \"\"   \n",
       "\n",
       "               gene_id     gene_name  ...  \\\n",
       "0  \"ENSG00000274847.1\"  \"AC145212.1\"  ...   \n",
       "1  \"ENSG00000274847.1\"  \"AC145212.1\"  ...   \n",
       "2  \"ENSG00000274847.1\"  \"AC145212.1\"  ...   \n",
       "3  \"ENSG00000274847.1\"  \"AC145212.1\"  ...   \n",
       "4                  \".\"        \"None\"  ...   \n",
       "\n",
       "   tardbp-316del346-egfp-293flp-l2-20171010-ju  \\\n",
       "0                                          0.0   \n",
       "1                                          0.0   \n",
       "2                                          0.0   \n",
       "3                                          3.0   \n",
       "4                                          0.0   \n",
       "\n",
       "   tardbp-egfp-293flp-m1-20171010-ju  \\\n",
       "0                                0.0   \n",
       "1                                0.0   \n",
       "2                                1.0   \n",
       "3                                1.0   \n",
       "4                                2.0   \n",
       "\n",
       "   tardbp-316del346-egfp-293flp-m2-20171010-ju  \\\n",
       "0                                          0.0   \n",
       "1                                          0.0   \n",
       "2                                          0.0   \n",
       "3                                          1.0   \n",
       "4                                          0.0   \n",
       "\n",
       "   tardbp-316del346-egfp-293flp-m1-20171010-ju  \\\n",
       "0                                          0.0   \n",
       "1                                          0.0   \n",
       "2                                          0.0   \n",
       "3                                          1.0   \n",
       "4                                          0.0   \n",
       "\n",
       "   tardbp-egfp-293flp-m2-20171010-ju  len  overlap_g1  overlap_g2  overlap_g3  \\\n",
       "0                                1.0   23    0.000000         0.0    0.000000   \n",
       "1                                0.0   18    0.000000         0.0   22.222222   \n",
       "2                                0.0   57   98.245614         0.0    0.000000   \n",
       "3                                0.0   21   47.619048         0.0    0.000000   \n",
       "4                                2.0   11    0.000000         0.0    0.000000   \n",
       "\n",
       "   overlap_g4  \n",
       "0  100.000000  \n",
       "1   33.333333  \n",
       "2    0.000000  \n",
       "3    0.000000  \n",
       "4  100.000000  \n",
       "\n",
       "[5 rows x 54 columns]"
      ]
     },
     "execution_count": 77,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_out_rfiltered_merged30_inner.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_out_rfiltered_merged30_inner['total_overlap'] = df_out_rfiltered_merged30_inner.overlap_g1 + df_out_rfiltered_merged30_inner.overlap_g2 + df_out_rfiltered_merged30_inner.overlap_g3 + df_out_rfiltered_merged30_inner.overlap_g4"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_out_rfiltered_merged30_inner['overlap_g1_pseudoc'] = df_out_rfiltered_merged30_inner.overlap_g1 + 1\n",
    "df_out_rfiltered_merged30_inner['overlap_g2_pseudoc'] = df_out_rfiltered_merged30_inner.overlap_g2 + 1 \n",
    "df_out_rfiltered_merged30_inner['overlap_g3_pseudoc'] = df_out_rfiltered_merged30_inner.overlap_g3 + 1\n",
    "df_out_rfiltered_merged30_inner['overlap_g4_pseudoc'] = df_out_rfiltered_merged30_inner.overlap_g4 + 1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_out_rfiltered_merged30_inner['length_groups'] = 'default'\n",
    "df_out_rfiltered_merged30_inner.loc[df_out_rfiltered_merged30_inner.len <= 30, 'length_groups'] = '3len30'\n",
    "df_out_rfiltered_merged30_inner.loc[(df_out_rfiltered_merged30_inner.len > 30) & (df_out_rfiltered_merged30_inner.len <= 60), 'length_groups'] = '2len30-60'\n",
    "df_out_rfiltered_merged30_inner.loc[(df_out_rfiltered_merged30_inner.len > 60) & (df_out_rfiltered_merged30_inner.len <= 100), 'length_groups'] = '1len60-100'\n",
    "df_out_rfiltered_merged30_inner.loc[df_out_rfiltered_merged30_inner.len > 100, 'length_groups'] = '0len100plus'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "count        116332\n",
       "unique            4\n",
       "top       2len30-60\n",
       "freq          41128\n",
       "Name: length_groups, dtype: object"
      ]
     },
     "execution_count": 81,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_out_rfiltered_merged30_inner.length_groups.describe()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_out_rfiltered_merged30_inner['total_overlap_bin'] = pd.qcut(df_out_rfiltered_merged30_inner['total_overlap'], 3, labels=['sparse', 'medium', 'dense'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_out_rfiltered_merged30_inner['g1plusg2byg4_ratio'] = (df_out_rfiltered_merged30_inner.overlap_g1_pseudoc + df_out_rfiltered_merged30_inner.overlap_g2_pseudoc) / df_out_rfiltered_merged30_inner.overlap_g4_pseudoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {},
   "outputs": [],
   "source": [
    "del df_out_rfiltered_merged30_inner['total_overlap_bin']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'0.24.2'"
      ]
     },
     "execution_count": 85,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.__version__"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "count    116332.000000\n",
       "mean         45.121566\n",
       "std          26.514989\n",
       "min           0.000000\n",
       "25%          24.675325\n",
       "50%          37.037037\n",
       "75%          60.000000\n",
       "max         100.000000\n",
       "Name: total_overlap, dtype: float64"
      ]
     },
     "execution_count": 86,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_out_rfiltered_merged30_inner.total_overlap.describe()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>count</th>\n",
       "      <th>mean</th>\n",
       "      <th>std</th>\n",
       "      <th>min</th>\n",
       "      <th>25%</th>\n",
       "      <th>50%</th>\n",
       "      <th>75%</th>\n",
       "      <th>max</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>length_groups</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0len100plus</th>\n",
       "      <td>11850.0</td>\n",
       "      <td>20.447389</td>\n",
       "      <td>10.386551</td>\n",
       "      <td>1.039169</td>\n",
       "      <td>13.709677</td>\n",
       "      <td>18.238994</td>\n",
       "      <td>24.236506</td>\n",
       "      <td>98.701299</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1len60-100</th>\n",
       "      <td>25056.0</td>\n",
       "      <td>30.597177</td>\n",
       "      <td>15.089638</td>\n",
       "      <td>6.060606</td>\n",
       "      <td>20.634921</td>\n",
       "      <td>26.562500</td>\n",
       "      <td>35.483871</td>\n",
       "      <td>100.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2len30-60</th>\n",
       "      <td>41128.0</td>\n",
       "      <td>40.080057</td>\n",
       "      <td>19.535335</td>\n",
       "      <td>3.030303</td>\n",
       "      <td>26.666667</td>\n",
       "      <td>35.294118</td>\n",
       "      <td>47.500000</td>\n",
       "      <td>100.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3len30</th>\n",
       "      <td>38298.0</td>\n",
       "      <td>67.672596</td>\n",
       "      <td>26.098560</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>45.833333</td>\n",
       "      <td>66.666667</td>\n",
       "      <td>100.000000</td>\n",
       "      <td>100.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                 count       mean        std       min        25%        50%  \\\n",
       "length_groups                                                                  \n",
       "0len100plus    11850.0  20.447389  10.386551  1.039169  13.709677  18.238994   \n",
       "1len60-100     25056.0  30.597177  15.089638  6.060606  20.634921  26.562500   \n",
       "2len30-60      41128.0  40.080057  19.535335  3.030303  26.666667  35.294118   \n",
       "3len30         38298.0  67.672596  26.098560  0.000000  45.833333  66.666667   \n",
       "\n",
       "                      75%         max  \n",
       "length_groups                          \n",
       "0len100plus     24.236506   98.701299  \n",
       "1len60-100      35.483871  100.000000  \n",
       "2len30-60       47.500000  100.000000  \n",
       "3len30         100.000000  100.000000  "
      ]
     },
     "execution_count": 87,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_out_rfiltered_merged30_inner.groupby(['length_groups'], observed=False).total_overlap.describe()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_out_rfiltered_merged30_inner['overlap_group'] = df_out_rfiltered_merged30_inner.groupby(['length_groups'], observed=False)['total_overlap'].transform(\n",
    "                     lambda x: pd.qcut(x, 3, labels=['sparse', 'medium', 'dense']))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "count     116332\n",
       "unique         3\n",
       "top       medium\n",
       "freq       38850\n",
       "Name: overlap_group, dtype: object"
      ]
     },
     "execution_count": 89,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_out_rfiltered_merged30_inner.overlap_group.describe()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_out_rfiltered_merged30_inner['g4byg1g2g3_ratio'] = df_out_rfiltered_merged30_inner.overlap_g4_pseudoc / (df_out_rfiltered_merged30_inner.overlap_g1_pseudoc + df_out_rfiltered_merged30_inner.overlap_g2_pseudoc + df_out_rfiltered_merged30_inner.overlap_g3_pseudoc)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>name</th>\n",
       "      <th>score</th>\n",
       "      <th>strand</th>\n",
       "      <th>region</th>\n",
       "      <th>biotype</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>...</th>\n",
       "      <th>overlap_g4</th>\n",
       "      <th>total_overlap</th>\n",
       "      <th>overlap_g1_pseudoc</th>\n",
       "      <th>overlap_g2_pseudoc</th>\n",
       "      <th>overlap_g3_pseudoc</th>\n",
       "      <th>overlap_g4_pseudoc</th>\n",
       "      <th>length_groups</th>\n",
       "      <th>g1plusg2byg4_ratio</th>\n",
       "      <th>overlap_group</th>\n",
       "      <th>g4byg1g2g3_ratio</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GL000194.1</td>\n",
       "      <td>75522</td>\n",
       "      <td>75545</td>\n",
       "      <td>.</td>\n",
       "      <td>22.0</td>\n",
       "      <td>-</td>\n",
       "      <td>intron</td>\n",
       "      <td>\"pre-mRNA\"</td>\n",
       "      <td>\"ENSG00000274847.1\"</td>\n",
       "      <td>\"AC145212.1\"</td>\n",
       "      <td>...</td>\n",
       "      <td>100.000000</td>\n",
       "      <td>100.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>101.000000</td>\n",
       "      <td>3len30</td>\n",
       "      <td>0.019802</td>\n",
       "      <td>dense</td>\n",
       "      <td>33.666667</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GL000194.1</td>\n",
       "      <td>81461</td>\n",
       "      <td>81479</td>\n",
       "      <td>.</td>\n",
       "      <td>13.0</td>\n",
       "      <td>-</td>\n",
       "      <td>intron</td>\n",
       "      <td>\"pre-mRNA\"</td>\n",
       "      <td>\"ENSG00000274847.1\"</td>\n",
       "      <td>\"AC145212.1\"</td>\n",
       "      <td>...</td>\n",
       "      <td>33.333333</td>\n",
       "      <td>55.555556</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.0</td>\n",
       "      <td>23.222222</td>\n",
       "      <td>34.333333</td>\n",
       "      <td>3len30</td>\n",
       "      <td>0.058252</td>\n",
       "      <td>medium</td>\n",
       "      <td>1.361233</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GL000194.1</td>\n",
       "      <td>92228</td>\n",
       "      <td>92285</td>\n",
       "      <td>.</td>\n",
       "      <td>68.0</td>\n",
       "      <td>-</td>\n",
       "      <td>intron</td>\n",
       "      <td>\"pre-mRNA\"</td>\n",
       "      <td>\"ENSG00000274847.1\"</td>\n",
       "      <td>\"AC145212.1\"</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>98.245614</td>\n",
       "      <td>99.245614</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>2len30-60</td>\n",
       "      <td>100.245614</td>\n",
       "      <td>dense</td>\n",
       "      <td>0.009877</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GL000194.1</td>\n",
       "      <td>99578</td>\n",
       "      <td>99599</td>\n",
       "      <td>.</td>\n",
       "      <td>23.0</td>\n",
       "      <td>-</td>\n",
       "      <td>intron</td>\n",
       "      <td>\"pre-mRNA\"</td>\n",
       "      <td>\"ENSG00000274847.1\"</td>\n",
       "      <td>\"AC145212.1\"</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>47.619048</td>\n",
       "      <td>48.619048</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>3len30</td>\n",
       "      <td>49.619048</td>\n",
       "      <td>sparse</td>\n",
       "      <td>0.019755</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GL000008.2</td>\n",
       "      <td>154464</td>\n",
       "      <td>154475</td>\n",
       "      <td>.</td>\n",
       "      <td>113.0</td>\n",
       "      <td>+</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>\"\"</td>\n",
       "      <td>\".\"</td>\n",
       "      <td>\"None\"</td>\n",
       "      <td>...</td>\n",
       "      <td>100.000000</td>\n",
       "      <td>100.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>101.000000</td>\n",
       "      <td>3len30</td>\n",
       "      <td>0.019802</td>\n",
       "      <td>dense</td>\n",
       "      <td>33.666667</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 63 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        chrom   start     end name  score strand      region     biotype  \\\n",
       "0  GL000194.1   75522   75545    .   22.0      -      intron  \"pre-mRNA\"   \n",
       "1  GL000194.1   81461   81479    .   13.0      -      intron  \"pre-mRNA\"   \n",
       "2  GL000194.1   92228   92285    .   68.0      -      intron  \"pre-mRNA\"   \n",
       "3  GL000194.1   99578   99599    .   23.0      -      intron  \"pre-mRNA\"   \n",
       "4  GL000008.2  154464  154475    .  113.0      +  intergenic          \"\"   \n",
       "\n",
       "               gene_id     gene_name  ...  overlap_g4  total_overlap  \\\n",
       "0  \"ENSG00000274847.1\"  \"AC145212.1\"  ...  100.000000     100.000000   \n",
       "1  \"ENSG00000274847.1\"  \"AC145212.1\"  ...   33.333333      55.555556   \n",
       "2  \"ENSG00000274847.1\"  \"AC145212.1\"  ...    0.000000      98.245614   \n",
       "3  \"ENSG00000274847.1\"  \"AC145212.1\"  ...    0.000000      47.619048   \n",
       "4                  \".\"        \"None\"  ...  100.000000     100.000000   \n",
       "\n",
       "   overlap_g1_pseudoc  overlap_g2_pseudoc  overlap_g3_pseudoc  \\\n",
       "0            1.000000                 1.0            1.000000   \n",
       "1            1.000000                 1.0           23.222222   \n",
       "2           99.245614                 1.0            1.000000   \n",
       "3           48.619048                 1.0            1.000000   \n",
       "4            1.000000                 1.0            1.000000   \n",
       "\n",
       "   overlap_g4_pseudoc  length_groups  g1plusg2byg4_ratio  overlap_group  \\\n",
       "0          101.000000         3len30            0.019802          dense   \n",
       "1           34.333333         3len30            0.058252         medium   \n",
       "2            1.000000      2len30-60          100.245614          dense   \n",
       "3            1.000000         3len30           49.619048         sparse   \n",
       "4          101.000000         3len30            0.019802          dense   \n",
       "\n",
       "   g4byg1g2g3_ratio  \n",
       "0         33.666667  \n",
       "1          1.361233  \n",
       "2          0.009877  \n",
       "3          0.019755  \n",
       "4         33.666667  \n",
       "\n",
       "[5 rows x 63 columns]"
      ]
     },
     "execution_count": 91,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_out_rfiltered_merged30_inner.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "metadata": {},
   "outputs": [],
   "source": [
    "#top_g1plusg2byg4 = df_out_rfiltered_merged30_inner.groupby(['length_groups','overlap_group']).apply(lambda grp: grp.nlargest(3000, 'g1plusg2byg4_ratio'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "metadata": {},
   "outputs": [],
   "source": [
    "top_g1plusg2byg4_under100 = df_out_rfiltered_merged30_inner[df_out_rfiltered_merged30_inner.length_groups != '0len100plus'].groupby(['length_groups','overlap_group']).apply(lambda grp: grp.nlargest(3000, 'g1plusg2byg4_ratio'))\n",
    "top_g1plusg2byg4_over100 = df_out_rfiltered_merged30_inner[df_out_rfiltered_merged30_inner.length_groups == '0len100plus'].groupby(['length_groups','overlap_group']).apply(lambda grp: grp.nlargest(1500, 'g1plusg2byg4_ratio'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "metadata": {},
   "outputs": [],
   "source": [
    "top_g1plusg2byg4 = pd.concat([top_g1plusg2byg4_under100, top_g1plusg2byg4_over100])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(27000, 4500, 31500)"
      ]
     },
     "execution_count": 95,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(top_g1plusg2byg4_under100), len(top_g1plusg2byg4_over100), len(top_g1plusg2byg4_under100) + len(top_g1plusg2byg4_over100)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 96,
   "metadata": {},
   "outputs": [],
   "source": [
    "top_g1plusg2byg4 = pd.concat([top_g1plusg2byg4_under100, top_g1plusg2byg4_over100])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "metadata": {},
   "outputs": [],
   "source": [
    "top_g1plusg2byg4['ratio_group'] = 'UG'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>name</th>\n",
       "      <th>score</th>\n",
       "      <th>strand</th>\n",
       "      <th>region</th>\n",
       "      <th>biotype</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>...</th>\n",
       "      <th>total_overlap</th>\n",
       "      <th>overlap_g1_pseudoc</th>\n",
       "      <th>overlap_g2_pseudoc</th>\n",
       "      <th>overlap_g3_pseudoc</th>\n",
       "      <th>overlap_g4_pseudoc</th>\n",
       "      <th>length_groups</th>\n",
       "      <th>g1plusg2byg4_ratio</th>\n",
       "      <th>overlap_group</th>\n",
       "      <th>g4byg1g2g3_ratio</th>\n",
       "      <th>ratio_group</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>length_groups</th>\n",
       "      <th>overlap_group</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"5\" valign=\"top\">1len60-100</th>\n",
       "      <th rowspan=\"5\" valign=\"top\">sparse</th>\n",
       "      <th>12685</th>\n",
       "      <td>chr10</td>\n",
       "      <td>74936663</td>\n",
       "      <td>74936725</td>\n",
       "      <td>.</td>\n",
       "      <td>92.0</td>\n",
       "      <td>+</td>\n",
       "      <td>intron</td>\n",
       "      <td>\"pre-mRNA\"</td>\n",
       "      <td>\"ENSG00000156650.12\"</td>\n",
       "      <td>\"KAT6B\"</td>\n",
       "      <td>...</td>\n",
       "      <td>22.580645</td>\n",
       "      <td>12.290323</td>\n",
       "      <td>12.290323</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1len60-100</td>\n",
       "      <td>24.580645</td>\n",
       "      <td>sparse</td>\n",
       "      <td>0.039092</td>\n",
       "      <td>UG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>73927</th>\n",
       "      <td>chr4</td>\n",
       "      <td>92611082</td>\n",
       "      <td>92611144</td>\n",
       "      <td>.</td>\n",
       "      <td>21.0</td>\n",
       "      <td>+</td>\n",
       "      <td>intron</td>\n",
       "      <td>\"lncRNA,pre-mRNA\"</td>\n",
       "      <td>\"ENSG00000152208.12\"</td>\n",
       "      <td>\"GRID2\"</td>\n",
       "      <td>...</td>\n",
       "      <td>22.580645</td>\n",
       "      <td>23.580645</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1len60-100</td>\n",
       "      <td>24.580645</td>\n",
       "      <td>sparse</td>\n",
       "      <td>0.039092</td>\n",
       "      <td>UG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15074</th>\n",
       "      <td>chr11</td>\n",
       "      <td>1405308</td>\n",
       "      <td>1405379</td>\n",
       "      <td>.</td>\n",
       "      <td>105.0</td>\n",
       "      <td>+</td>\n",
       "      <td>intron</td>\n",
       "      <td>\"pre-mRNA\"</td>\n",
       "      <td>\"ENSG00000174672.15\"</td>\n",
       "      <td>\"BRSK2\"</td>\n",
       "      <td>...</td>\n",
       "      <td>22.535211</td>\n",
       "      <td>23.535211</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1len60-100</td>\n",
       "      <td>24.535211</td>\n",
       "      <td>sparse</td>\n",
       "      <td>0.039162</td>\n",
       "      <td>UG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64656</th>\n",
       "      <td>chr22</td>\n",
       "      <td>48754450</td>\n",
       "      <td>48754521</td>\n",
       "      <td>.</td>\n",
       "      <td>22.0</td>\n",
       "      <td>+</td>\n",
       "      <td>intron</td>\n",
       "      <td>\"pre-mRNA\"</td>\n",
       "      <td>\"ENSG00000219438.8\"</td>\n",
       "      <td>\"FAM19A5\"</td>\n",
       "      <td>...</td>\n",
       "      <td>22.535211</td>\n",
       "      <td>23.535211</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1len60-100</td>\n",
       "      <td>24.535211</td>\n",
       "      <td>sparse</td>\n",
       "      <td>0.039162</td>\n",
       "      <td>UG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>114984</th>\n",
       "      <td>chr19</td>\n",
       "      <td>47759937</td>\n",
       "      <td>47760004</td>\n",
       "      <td>.</td>\n",
       "      <td>356.0</td>\n",
       "      <td>+</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>\"\"</td>\n",
       "      <td>\".\"</td>\n",
       "      <td>\"None\"</td>\n",
       "      <td>...</td>\n",
       "      <td>22.388060</td>\n",
       "      <td>23.388060</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1len60-100</td>\n",
       "      <td>24.388060</td>\n",
       "      <td>sparse</td>\n",
       "      <td>0.039389</td>\n",
       "      <td>UG</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 64 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                    chrom     start       end name  score  \\\n",
       "length_groups overlap_group                                                 \n",
       "1len60-100    sparse        12685   chr10  74936663  74936725    .   92.0   \n",
       "                            73927    chr4  92611082  92611144    .   21.0   \n",
       "                            15074   chr11   1405308   1405379    .  105.0   \n",
       "                            64656   chr22  48754450  48754521    .   22.0   \n",
       "                            114984  chr19  47759937  47760004    .  356.0   \n",
       "\n",
       "                                   strand      region            biotype  \\\n",
       "length_groups overlap_group                                                \n",
       "1len60-100    sparse        12685       +      intron         \"pre-mRNA\"   \n",
       "                            73927       +      intron  \"lncRNA,pre-mRNA\"   \n",
       "                            15074       +      intron         \"pre-mRNA\"   \n",
       "                            64656       +      intron         \"pre-mRNA\"   \n",
       "                            114984      +  intergenic                 \"\"   \n",
       "\n",
       "                                                 gene_id  gene_name  ...  \\\n",
       "length_groups overlap_group                                          ...   \n",
       "1len60-100    sparse        12685   \"ENSG00000156650.12\"    \"KAT6B\"  ...   \n",
       "                            73927   \"ENSG00000152208.12\"    \"GRID2\"  ...   \n",
       "                            15074   \"ENSG00000174672.15\"    \"BRSK2\"  ...   \n",
       "                            64656    \"ENSG00000219438.8\"  \"FAM19A5\"  ...   \n",
       "                            114984                   \".\"     \"None\"  ...   \n",
       "\n",
       "                                    total_overlap  overlap_g1_pseudoc  \\\n",
       "length_groups overlap_group                                             \n",
       "1len60-100    sparse        12685       22.580645           12.290323   \n",
       "                            73927       22.580645           23.580645   \n",
       "                            15074       22.535211           23.535211   \n",
       "                            64656       22.535211           23.535211   \n",
       "                            114984      22.388060           23.388060   \n",
       "\n",
       "                                    overlap_g2_pseudoc  overlap_g3_pseudoc  \\\n",
       "length_groups overlap_group                                                  \n",
       "1len60-100    sparse        12685            12.290323                 1.0   \n",
       "                            73927             1.000000                 1.0   \n",
       "                            15074             1.000000                 1.0   \n",
       "                            64656             1.000000                 1.0   \n",
       "                            114984            1.000000                 1.0   \n",
       "\n",
       "                                    overlap_g4_pseudoc  length_groups  \\\n",
       "length_groups overlap_group                                             \n",
       "1len60-100    sparse        12685                  1.0     1len60-100   \n",
       "                            73927                  1.0     1len60-100   \n",
       "                            15074                  1.0     1len60-100   \n",
       "                            64656                  1.0     1len60-100   \n",
       "                            114984                 1.0     1len60-100   \n",
       "\n",
       "                                    g1plusg2byg4_ratio  overlap_group  \\\n",
       "length_groups overlap_group                                             \n",
       "1len60-100    sparse        12685            24.580645         sparse   \n",
       "                            73927            24.580645         sparse   \n",
       "                            15074            24.535211         sparse   \n",
       "                            64656            24.535211         sparse   \n",
       "                            114984           24.388060         sparse   \n",
       "\n",
       "                                    g4byg1g2g3_ratio  ratio_group  \n",
       "length_groups overlap_group                                        \n",
       "1len60-100    sparse        12685           0.039092           UG  \n",
       "                            73927           0.039092           UG  \n",
       "                            15074           0.039162           UG  \n",
       "                            64656           0.039162           UG  \n",
       "                            114984          0.039389           UG  \n",
       "\n",
       "[5 rows x 64 columns]"
      ]
     },
     "execution_count": 98,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "top_g1plusg2byg4.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "metadata": {},
   "outputs": [],
   "source": [
    "top_g1plusg2byg4 = top_g1plusg2byg4.reset_index(drop='True')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 100,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['chrom', 'start', 'end', 'name', 'score', 'strand', 'region', 'biotype',\n",
       "       'gene_id', 'gene_name',\n",
       "       'tardbp-316del346-egfp-hek293-hd1-6-4-20180829-ju',\n",
       "       'tardbp-egfp-hek293-hd2-5-3-20180829-ju',\n",
       "       'tardbp-egfp-hek293-hd2-5-on-bead-rt-1-20180829-ju',\n",
       "       'tardbp-316del346-egfp-hek293-hd2-5-3-20180829-ju',\n",
       "       'tardbp-egfp-hek293-3-20181001-ju',\n",
       "       'tardbp-q331k-egfp-hek293-no-uv-1-20181001-ju',\n",
       "       'tardbp-320del414-egfp-hek293-hd1-6-on-bead-rt-1-20180829-ju',\n",
       "       'tardbp-316del346-egfp-hek293-4-20181001-ju',\n",
       "       'tardbp-egfp-hek293-no-uv-1-20180829-ju',\n",
       "       'tardbp-egfp-hek293-4-20181001-ju',\n",
       "       'tardbp-egfp-hek293-hd1-6-4-20180829-ju',\n",
       "       'tardbp-a326p-egfp-hek293-4-20181001-ju',\n",
       "       'tardbp-egfp-hek293-hd1-6-3-20180829-ju',\n",
       "       'tardbp-320del414-egfp-hek293-hd1-6-4-20180829-ju',\n",
       "       'tardbp-q331k-egfp-hek293-3-20181001-ju',\n",
       "       'tardbp-316del346-egfp-hek293-hd1-6-3-20180829-ju',\n",
       "       'tardbp-320del414-egfp-hek293-hd1-6-3-20180829-ju',\n",
       "       'tardbp-egfp-hek293-hd1-6-on-bead-rt-1-20180829-ju',\n",
       "       'tardbp-316del346-egfp-hek293-hd1-6-on-bead-rt-1-20180829-ju',\n",
       "       'tardbp-m337p-egfp-hek293-4-20181001-ju',\n",
       "       'tardbp-m337p-egfp-hek293-3-20181001-ju',\n",
       "       'tardbp-a326p-egfp-hek293-3-20181001-ju',\n",
       "       'tardbp-316del346-egfp-hek293-hd2-5-4-20180829-ju',\n",
       "       'tardbp-q331k-egfp-hek293-4-20181001-ju',\n",
       "       'tardbp-g335a-egfp-hek293-3-20181001-ju',\n",
       "       'tardbp-g335a-egfp-hek293-4-20181001-ju',\n",
       "       'tardbp-g294a-egfp-hek293-3-20181001-ju',\n",
       "       'tardbp-316del346-egfp-hek293-hd2-5-on-bead-rt-1-20180829-ju',\n",
       "       'tardbp-316del346-egfp-hek293-3-20181001-ju',\n",
       "       'tardbp-egfp-hek293-hd2-5-4-20180829-ju',\n",
       "       'tardbp-g294a-egfp-hek293-4-20181001-ju',\n",
       "       'tardbp-egfp-293flp-l2-20171010-ju',\n",
       "       'tardbp-egfp-293flp-l1-20171010-ju',\n",
       "       'tardbp-316del346-egfp-293flp-l1-20171010-ju',\n",
       "       'tardbp-316del346-egfp-293flp-l2-20171010-ju',\n",
       "       'tardbp-egfp-293flp-m1-20171010-ju',\n",
       "       'tardbp-316del346-egfp-293flp-m2-20171010-ju',\n",
       "       'tardbp-316del346-egfp-293flp-m1-20171010-ju',\n",
       "       'tardbp-egfp-293flp-m2-20171010-ju', 'len', 'overlap_g1', 'overlap_g2',\n",
       "       'overlap_g3', 'overlap_g4', 'total_overlap', 'overlap_g1_pseudoc',\n",
       "       'overlap_g2_pseudoc', 'overlap_g3_pseudoc', 'overlap_g4_pseudoc',\n",
       "       'length_groups', 'g1plusg2byg4_ratio', 'overlap_group',\n",
       "       'g4byg1g2g3_ratio', 'ratio_group'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 100,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "top_g1plusg2byg4.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 101,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_1 = pd.merge(df_out_rfiltered_merged30_inner, top_g1plusg2byg4, \n",
    "        on=['chrom', 'start', 'end', 'name', 'score', 'strand', 'region', 'biotype',\n",
    "       'gene_id', 'gene_name', 'tardbp-316del346-egfp-hek293-4-20181001-ju',\n",
    "       #'tardbp-293flp-2-20181129-ju',\n",
    "       'tardbp-316del346-egfp-hek293-3-20181001-ju',\n",
    "       'tardbp-egfp-hek293-hd2-5-4-20180829-ju',\n",
    "       'tardbp-a326p-egfp-hek293-4-20181001-ju',\n",
    "       'tardbp-egfp-hek293-hd1-6-on-bead-rt-1-20180829-ju',\n",
    "       'tardbp-g294a-egfp-hek293-4-20181001-ju',\n",
    "       'tardbp-316del346-egfp-hek293-hd1-6-3-20180829-ju',\n",
    "       'tardbp-320del414-egfp-hek293-hd1-6-3-20180829-ju',\n",
    "       'tardbp-egfp-hek293-no-uv-1-20180829-ju',\n",
    "       'tardbp-egfp-hek293-hd2-5-on-bead-rt-1-20180829-ju',\n",
    "       'tardbp-egfp-hek293-hd1-6-3-20180829-ju',\n",
    "       'tardbp-316del346-egfp-hek293-hd1-6-4-20180829-ju',\n",
    "       'tardbp-316del346-egfp-hek293-hd2-5-4-20180829-ju',\n",
    "       'tardbp-a326p-egfp-hek293-3-20181001-ju',\n",
    "       'tardbp-g335a-egfp-hek293-3-20181001-ju',\n",
    "       'tardbp-g294a-egfp-hek293-3-20181001-ju',\n",
    "       'tardbp-320del414-egfp-hek293-hd1-6-on-bead-rt-1-20180829-ju',\n",
    "       'tardbp-316del346-egfp-hek293-hd1-6-on-bead-rt-1-20180829-ju',\n",
    "       'tardbp-m337p-egfp-hek293-3-20181001-ju',\n",
    "       'tardbp-q331k-egfp-hek293-4-20181001-ju',\n",
    "       'tardbp-egfp-hek293-hd2-5-3-20180829-ju',\n",
    "       'tardbp-316del346-egfp-hek293-hd2-5-3-20180829-ju',\n",
    "       'tardbp-320del414-egfp-hek293-hd1-6-4-20180829-ju',\n",
    "       'tardbp-egfp-hek293-3-20181001-ju', #'tardbp-293flp-1-20181129-ju',\n",
    "       'tardbp-q331k-egfp-hek293-no-uv-1-20181001-ju',\n",
    "       'tardbp-g335a-egfp-hek293-4-20181001-ju',\n",
    "       'tardbp-316del346-egfp-hek293-hd2-5-on-bead-rt-1-20180829-ju',\n",
    "       'tardbp-q331k-egfp-hek293-3-20181001-ju',\n",
    "       'tardbp-egfp-hek293-hd1-6-4-20180829-ju',\n",
    "       'tardbp-egfp-hek293-4-20181001-ju',\n",
    "       'tardbp-m337p-egfp-hek293-4-20181001-ju',\n",
    "       'tardbp-egfp-293flp-m2-20171010-ju',\n",
    "       'tardbp-316del346-egfp-293flp-l1-20171010-ju',\n",
    "       'tardbp-316del346-egfp-293flp-l2-20171010-ju',\n",
    "       #'tardbp-293flp-2-20181129-ju', \n",
    "       'tardbp-egfp-293flp-l2-20171010-ju',\n",
    "       'tardbp-316del346-egfp-293flp-m1-20171010-ju',\n",
    "       'tardbp-egfp-293flp-l1-20171010-ju',\n",
    "       'tardbp-316del346-egfp-293flp-m2-20171010-ju',\n",
    "       'tardbp-egfp-293flp-m1-20171010-ju', #'tardbp-293flp-1-20181129-ju',\n",
    "       'len', 'overlap_g1', 'overlap_g2', 'overlap_g3', 'overlap_g4',\n",
    "       'total_overlap', 'overlap_g1_pseudoc', 'overlap_g2_pseudoc',\n",
    "       'overlap_g3_pseudoc', 'overlap_g4_pseudoc', 'length_groups',\n",
    "       'g1plusg2byg4_ratio', 'overlap_group', 'g4byg1g2g3_ratio',],\n",
    "               how='left')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 102,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>name</th>\n",
       "      <th>score</th>\n",
       "      <th>strand</th>\n",
       "      <th>region</th>\n",
       "      <th>biotype</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>...</th>\n",
       "      <th>total_overlap</th>\n",
       "      <th>overlap_g1_pseudoc</th>\n",
       "      <th>overlap_g2_pseudoc</th>\n",
       "      <th>overlap_g3_pseudoc</th>\n",
       "      <th>overlap_g4_pseudoc</th>\n",
       "      <th>length_groups</th>\n",
       "      <th>g1plusg2byg4_ratio</th>\n",
       "      <th>overlap_group</th>\n",
       "      <th>g4byg1g2g3_ratio</th>\n",
       "      <th>ratio_group</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GL000194.1</td>\n",
       "      <td>75522</td>\n",
       "      <td>75545</td>\n",
       "      <td>.</td>\n",
       "      <td>22.0</td>\n",
       "      <td>-</td>\n",
       "      <td>intron</td>\n",
       "      <td>\"pre-mRNA\"</td>\n",
       "      <td>\"ENSG00000274847.1\"</td>\n",
       "      <td>\"AC145212.1\"</td>\n",
       "      <td>...</td>\n",
       "      <td>100.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>101.000000</td>\n",
       "      <td>3len30</td>\n",
       "      <td>0.019802</td>\n",
       "      <td>dense</td>\n",
       "      <td>33.666667</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GL000194.1</td>\n",
       "      <td>81461</td>\n",
       "      <td>81479</td>\n",
       "      <td>.</td>\n",
       "      <td>13.0</td>\n",
       "      <td>-</td>\n",
       "      <td>intron</td>\n",
       "      <td>\"pre-mRNA\"</td>\n",
       "      <td>\"ENSG00000274847.1\"</td>\n",
       "      <td>\"AC145212.1\"</td>\n",
       "      <td>...</td>\n",
       "      <td>55.555556</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.0</td>\n",
       "      <td>23.222222</td>\n",
       "      <td>34.333333</td>\n",
       "      <td>3len30</td>\n",
       "      <td>0.058252</td>\n",
       "      <td>medium</td>\n",
       "      <td>1.361233</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GL000194.1</td>\n",
       "      <td>92228</td>\n",
       "      <td>92285</td>\n",
       "      <td>.</td>\n",
       "      <td>68.0</td>\n",
       "      <td>-</td>\n",
       "      <td>intron</td>\n",
       "      <td>\"pre-mRNA\"</td>\n",
       "      <td>\"ENSG00000274847.1\"</td>\n",
       "      <td>\"AC145212.1\"</td>\n",
       "      <td>...</td>\n",
       "      <td>98.245614</td>\n",
       "      <td>99.245614</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>2len30-60</td>\n",
       "      <td>100.245614</td>\n",
       "      <td>dense</td>\n",
       "      <td>0.009877</td>\n",
       "      <td>UG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GL000194.1</td>\n",
       "      <td>99578</td>\n",
       "      <td>99599</td>\n",
       "      <td>.</td>\n",
       "      <td>23.0</td>\n",
       "      <td>-</td>\n",
       "      <td>intron</td>\n",
       "      <td>\"pre-mRNA\"</td>\n",
       "      <td>\"ENSG00000274847.1\"</td>\n",
       "      <td>\"AC145212.1\"</td>\n",
       "      <td>...</td>\n",
       "      <td>47.619048</td>\n",
       "      <td>48.619048</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>3len30</td>\n",
       "      <td>49.619048</td>\n",
       "      <td>sparse</td>\n",
       "      <td>0.019755</td>\n",
       "      <td>UG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GL000008.2</td>\n",
       "      <td>154464</td>\n",
       "      <td>154475</td>\n",
       "      <td>.</td>\n",
       "      <td>113.0</td>\n",
       "      <td>+</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>\"\"</td>\n",
       "      <td>\".\"</td>\n",
       "      <td>\"None\"</td>\n",
       "      <td>...</td>\n",
       "      <td>100.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>101.000000</td>\n",
       "      <td>3len30</td>\n",
       "      <td>0.019802</td>\n",
       "      <td>dense</td>\n",
       "      <td>33.666667</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 64 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        chrom   start     end name  score strand      region     biotype  \\\n",
       "0  GL000194.1   75522   75545    .   22.0      -      intron  \"pre-mRNA\"   \n",
       "1  GL000194.1   81461   81479    .   13.0      -      intron  \"pre-mRNA\"   \n",
       "2  GL000194.1   92228   92285    .   68.0      -      intron  \"pre-mRNA\"   \n",
       "3  GL000194.1   99578   99599    .   23.0      -      intron  \"pre-mRNA\"   \n",
       "4  GL000008.2  154464  154475    .  113.0      +  intergenic          \"\"   \n",
       "\n",
       "               gene_id     gene_name  ...  total_overlap  overlap_g1_pseudoc  \\\n",
       "0  \"ENSG00000274847.1\"  \"AC145212.1\"  ...     100.000000            1.000000   \n",
       "1  \"ENSG00000274847.1\"  \"AC145212.1\"  ...      55.555556            1.000000   \n",
       "2  \"ENSG00000274847.1\"  \"AC145212.1\"  ...      98.245614           99.245614   \n",
       "3  \"ENSG00000274847.1\"  \"AC145212.1\"  ...      47.619048           48.619048   \n",
       "4                  \".\"        \"None\"  ...     100.000000            1.000000   \n",
       "\n",
       "   overlap_g2_pseudoc  overlap_g3_pseudoc  overlap_g4_pseudoc  length_groups  \\\n",
       "0                 1.0            1.000000          101.000000         3len30   \n",
       "1                 1.0           23.222222           34.333333         3len30   \n",
       "2                 1.0            1.000000            1.000000      2len30-60   \n",
       "3                 1.0            1.000000            1.000000         3len30   \n",
       "4                 1.0            1.000000          101.000000         3len30   \n",
       "\n",
       "   g1plusg2byg4_ratio  overlap_group  g4byg1g2g3_ratio  ratio_group  \n",
       "0            0.019802          dense         33.666667          NaN  \n",
       "1            0.058252         medium          1.361233          NaN  \n",
       "2          100.245614          dense          0.009877           UG  \n",
       "3           49.619048         sparse          0.019755           UG  \n",
       "4            0.019802          dense         33.666667          NaN  \n",
       "\n",
       "[5 rows x 64 columns]"
      ]
     },
     "execution_count": 102,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_1.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 103,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "84832"
      ]
     },
     "execution_count": 103,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(df_1[df_1.ratio_group.isna()])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "metadata": {},
   "outputs": [],
   "source": [
    "top_g4byg1g2g3_ratio_under100 = df_1[(df_1.ratio_group.isna()) & (df_1.length_groups != '0len100plus')].groupby(['length_groups','overlap_group']).apply(lambda grp: grp.nlargest(3000, 'g4byg1g2g3_ratio'))\n",
    "top_g4byg1g2g3_ratio_over100 = df_1[(df_1.ratio_group.isna()) & (df_1.length_groups == '0len100plus')].groupby(['length_groups','overlap_group']).apply(lambda grp: grp.nlargest(1500, 'g4byg1g2g3_ratio'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "metadata": {},
   "outputs": [],
   "source": [
    "top_g4byg1g2g3_ratio = pd.concat([top_g4byg1g2g3_ratio_under100, top_g4byg1g2g3_ratio_over100])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 106,
   "metadata": {},
   "outputs": [],
   "source": [
    "top_g4byg1g2g3_ratio.ratio_group = 'AA'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "metadata": {},
   "outputs": [],
   "source": [
    "top_g4byg1g2g3_ratio = top_g4byg1g2g3_ratio.reset_index(drop='True')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 108,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>name</th>\n",
       "      <th>score</th>\n",
       "      <th>strand</th>\n",
       "      <th>region</th>\n",
       "      <th>biotype</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>...</th>\n",
       "      <th>total_overlap</th>\n",
       "      <th>overlap_g1_pseudoc</th>\n",
       "      <th>overlap_g2_pseudoc</th>\n",
       "      <th>overlap_g3_pseudoc</th>\n",
       "      <th>overlap_g4_pseudoc</th>\n",
       "      <th>length_groups</th>\n",
       "      <th>g1plusg2byg4_ratio</th>\n",
       "      <th>overlap_group</th>\n",
       "      <th>g4byg1g2g3_ratio</th>\n",
       "      <th>ratio_group</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>17429892</td>\n",
       "      <td>17429954</td>\n",
       "      <td>.</td>\n",
       "      <td>188.0</td>\n",
       "      <td>-</td>\n",
       "      <td>intron</td>\n",
       "      <td>\"pre-mRNA\"</td>\n",
       "      <td>\"ENSG00000179051.13\"</td>\n",
       "      <td>\"RCC2\"</td>\n",
       "      <td>...</td>\n",
       "      <td>22.580645</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>23.580645</td>\n",
       "      <td>1len60-100</td>\n",
       "      <td>0.084815</td>\n",
       "      <td>sparse</td>\n",
       "      <td>7.860215</td>\n",
       "      <td>AA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>177032932</td>\n",
       "      <td>177032994</td>\n",
       "      <td>.</td>\n",
       "      <td>28.0</td>\n",
       "      <td>-</td>\n",
       "      <td>intron</td>\n",
       "      <td>\"lncRNA,pre-mRNA\"</td>\n",
       "      <td>\"ENSG00000152092.15\"</td>\n",
       "      <td>\"ASTN1\"</td>\n",
       "      <td>...</td>\n",
       "      <td>22.580645</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>23.580645</td>\n",
       "      <td>1len60-100</td>\n",
       "      <td>0.084815</td>\n",
       "      <td>sparse</td>\n",
       "      <td>7.860215</td>\n",
       "      <td>AA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr12</td>\n",
       "      <td>12361647</td>\n",
       "      <td>12361709</td>\n",
       "      <td>.</td>\n",
       "      <td>73.0</td>\n",
       "      <td>+</td>\n",
       "      <td>intron</td>\n",
       "      <td>\"pre-mRNA\"</td>\n",
       "      <td>\"ENSG00000165714.10\"</td>\n",
       "      <td>\"BORCS5\"</td>\n",
       "      <td>...</td>\n",
       "      <td>22.580645</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>23.580645</td>\n",
       "      <td>1len60-100</td>\n",
       "      <td>0.084815</td>\n",
       "      <td>sparse</td>\n",
       "      <td>7.860215</td>\n",
       "      <td>AA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr2</td>\n",
       "      <td>241426522</td>\n",
       "      <td>241426584</td>\n",
       "      <td>.</td>\n",
       "      <td>59.0</td>\n",
       "      <td>+</td>\n",
       "      <td>intron</td>\n",
       "      <td>\"pre-mRNA\"</td>\n",
       "      <td>\"ENSG00000006607.13\"</td>\n",
       "      <td>\"FARP2\"</td>\n",
       "      <td>...</td>\n",
       "      <td>22.580645</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>23.580645</td>\n",
       "      <td>1len60-100</td>\n",
       "      <td>0.084815</td>\n",
       "      <td>sparse</td>\n",
       "      <td>7.860215</td>\n",
       "      <td>AA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr5</td>\n",
       "      <td>142796184</td>\n",
       "      <td>142796246</td>\n",
       "      <td>.</td>\n",
       "      <td>207.0</td>\n",
       "      <td>+</td>\n",
       "      <td>intron</td>\n",
       "      <td>\"pre-mRNA\"</td>\n",
       "      <td>\"ENSG00000145819.15\"</td>\n",
       "      <td>\"ARHGAP26\"</td>\n",
       "      <td>...</td>\n",
       "      <td>22.580645</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>23.580645</td>\n",
       "      <td>1len60-100</td>\n",
       "      <td>0.084815</td>\n",
       "      <td>sparse</td>\n",
       "      <td>7.860215</td>\n",
       "      <td>AA</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 64 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   chrom      start        end name  score strand  region            biotype  \\\n",
       "0   chr1   17429892   17429954    .  188.0      -  intron         \"pre-mRNA\"   \n",
       "1   chr1  177032932  177032994    .   28.0      -  intron  \"lncRNA,pre-mRNA\"   \n",
       "2  chr12   12361647   12361709    .   73.0      +  intron         \"pre-mRNA\"   \n",
       "3   chr2  241426522  241426584    .   59.0      +  intron         \"pre-mRNA\"   \n",
       "4   chr5  142796184  142796246    .  207.0      +  intron         \"pre-mRNA\"   \n",
       "\n",
       "                gene_id   gene_name  ...  total_overlap  overlap_g1_pseudoc  \\\n",
       "0  \"ENSG00000179051.13\"      \"RCC2\"  ...      22.580645                 1.0   \n",
       "1  \"ENSG00000152092.15\"     \"ASTN1\"  ...      22.580645                 1.0   \n",
       "2  \"ENSG00000165714.10\"    \"BORCS5\"  ...      22.580645                 1.0   \n",
       "3  \"ENSG00000006607.13\"     \"FARP2\"  ...      22.580645                 1.0   \n",
       "4  \"ENSG00000145819.15\"  \"ARHGAP26\"  ...      22.580645                 1.0   \n",
       "\n",
       "   overlap_g2_pseudoc  overlap_g3_pseudoc  overlap_g4_pseudoc  length_groups  \\\n",
       "0                 1.0                 1.0           23.580645     1len60-100   \n",
       "1                 1.0                 1.0           23.580645     1len60-100   \n",
       "2                 1.0                 1.0           23.580645     1len60-100   \n",
       "3                 1.0                 1.0           23.580645     1len60-100   \n",
       "4                 1.0                 1.0           23.580645     1len60-100   \n",
       "\n",
       "   g1plusg2byg4_ratio  overlap_group  g4byg1g2g3_ratio  ratio_group  \n",
       "0            0.084815         sparse          7.860215           AA  \n",
       "1            0.084815         sparse          7.860215           AA  \n",
       "2            0.084815         sparse          7.860215           AA  \n",
       "3            0.084815         sparse          7.860215           AA  \n",
       "4            0.084815         sparse          7.860215           AA  \n",
       "\n",
       "[5 rows x 64 columns]"
      ]
     },
     "execution_count": 108,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "top_g4byg1g2g3_ratio.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 109,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['chrom', 'start', 'end', 'name', 'score', 'strand', 'region', 'biotype',\n",
       "       'gene_id', 'gene_name',\n",
       "       'tardbp-316del346-egfp-hek293-hd1-6-4-20180829-ju',\n",
       "       'tardbp-egfp-hek293-hd2-5-3-20180829-ju',\n",
       "       'tardbp-egfp-hek293-hd2-5-on-bead-rt-1-20180829-ju',\n",
       "       'tardbp-316del346-egfp-hek293-hd2-5-3-20180829-ju',\n",
       "       'tardbp-egfp-hek293-3-20181001-ju',\n",
       "       'tardbp-q331k-egfp-hek293-no-uv-1-20181001-ju',\n",
       "       'tardbp-320del414-egfp-hek293-hd1-6-on-bead-rt-1-20180829-ju',\n",
       "       'tardbp-316del346-egfp-hek293-4-20181001-ju',\n",
       "       'tardbp-egfp-hek293-no-uv-1-20180829-ju',\n",
       "       'tardbp-egfp-hek293-4-20181001-ju',\n",
       "       'tardbp-egfp-hek293-hd1-6-4-20180829-ju',\n",
       "       'tardbp-a326p-egfp-hek293-4-20181001-ju',\n",
       "       'tardbp-egfp-hek293-hd1-6-3-20180829-ju',\n",
       "       'tardbp-320del414-egfp-hek293-hd1-6-4-20180829-ju',\n",
       "       'tardbp-q331k-egfp-hek293-3-20181001-ju',\n",
       "       'tardbp-316del346-egfp-hek293-hd1-6-3-20180829-ju',\n",
       "       'tardbp-320del414-egfp-hek293-hd1-6-3-20180829-ju',\n",
       "       'tardbp-egfp-hek293-hd1-6-on-bead-rt-1-20180829-ju',\n",
       "       'tardbp-316del346-egfp-hek293-hd1-6-on-bead-rt-1-20180829-ju',\n",
       "       'tardbp-m337p-egfp-hek293-4-20181001-ju',\n",
       "       'tardbp-m337p-egfp-hek293-3-20181001-ju',\n",
       "       'tardbp-a326p-egfp-hek293-3-20181001-ju',\n",
       "       'tardbp-316del346-egfp-hek293-hd2-5-4-20180829-ju',\n",
       "       'tardbp-q331k-egfp-hek293-4-20181001-ju',\n",
       "       'tardbp-g335a-egfp-hek293-3-20181001-ju',\n",
       "       'tardbp-g335a-egfp-hek293-4-20181001-ju',\n",
       "       'tardbp-g294a-egfp-hek293-3-20181001-ju',\n",
       "       'tardbp-316del346-egfp-hek293-hd2-5-on-bead-rt-1-20180829-ju',\n",
       "       'tardbp-316del346-egfp-hek293-3-20181001-ju',\n",
       "       'tardbp-egfp-hek293-hd2-5-4-20180829-ju',\n",
       "       'tardbp-g294a-egfp-hek293-4-20181001-ju',\n",
       "       'tardbp-egfp-293flp-l2-20171010-ju',\n",
       "       'tardbp-egfp-293flp-l1-20171010-ju',\n",
       "       'tardbp-316del346-egfp-293flp-l1-20171010-ju',\n",
       "       'tardbp-316del346-egfp-293flp-l2-20171010-ju',\n",
       "       'tardbp-egfp-293flp-m1-20171010-ju',\n",
       "       'tardbp-316del346-egfp-293flp-m2-20171010-ju',\n",
       "       'tardbp-316del346-egfp-293flp-m1-20171010-ju',\n",
       "       'tardbp-egfp-293flp-m2-20171010-ju', 'len', 'overlap_g1', 'overlap_g2',\n",
       "       'overlap_g3', 'overlap_g4', 'total_overlap', 'overlap_g1_pseudoc',\n",
       "       'overlap_g2_pseudoc', 'overlap_g3_pseudoc', 'overlap_g4_pseudoc',\n",
       "       'length_groups', 'g1plusg2byg4_ratio', 'overlap_group',\n",
       "       'g4byg1g2g3_ratio', 'ratio_group'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 109,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "top_g4byg1g2g3_ratio.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 110,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_2 = pd.merge(df_1, top_g4byg1g2g3_ratio, \n",
    "        on=['chrom', 'start', 'end', 'name', 'score', 'strand', 'region', 'biotype',\n",
    "       'gene_id', 'gene_name', 'tardbp-316del346-egfp-hek293-4-20181001-ju',\n",
    "       #'tardbp-293flp-2-20181129-ju',\n",
    "       'tardbp-316del346-egfp-hek293-3-20181001-ju',\n",
    "       'tardbp-egfp-hek293-hd2-5-4-20180829-ju',\n",
    "       'tardbp-a326p-egfp-hek293-4-20181001-ju',\n",
    "       'tardbp-egfp-hek293-hd1-6-on-bead-rt-1-20180829-ju',\n",
    "       'tardbp-g294a-egfp-hek293-4-20181001-ju',\n",
    "       'tardbp-316del346-egfp-hek293-hd1-6-3-20180829-ju',\n",
    "       'tardbp-320del414-egfp-hek293-hd1-6-3-20180829-ju',\n",
    "       'tardbp-egfp-hek293-no-uv-1-20180829-ju',\n",
    "       'tardbp-egfp-hek293-hd2-5-on-bead-rt-1-20180829-ju',\n",
    "       'tardbp-egfp-hek293-hd1-6-3-20180829-ju',\n",
    "       'tardbp-316del346-egfp-hek293-hd1-6-4-20180829-ju',\n",
    "       'tardbp-316del346-egfp-hek293-hd2-5-4-20180829-ju',\n",
    "       'tardbp-a326p-egfp-hek293-3-20181001-ju',\n",
    "       'tardbp-g335a-egfp-hek293-3-20181001-ju',\n",
    "       'tardbp-g294a-egfp-hek293-3-20181001-ju',\n",
    "       'tardbp-320del414-egfp-hek293-hd1-6-on-bead-rt-1-20180829-ju',\n",
    "       'tardbp-316del346-egfp-hek293-hd1-6-on-bead-rt-1-20180829-ju',\n",
    "       'tardbp-m337p-egfp-hek293-3-20181001-ju',\n",
    "       'tardbp-q331k-egfp-hek293-4-20181001-ju',\n",
    "       'tardbp-egfp-hek293-hd2-5-3-20180829-ju',\n",
    "       'tardbp-316del346-egfp-hek293-hd2-5-3-20180829-ju',\n",
    "       'tardbp-320del414-egfp-hek293-hd1-6-4-20180829-ju',\n",
    "       'tardbp-egfp-hek293-3-20181001-ju', \n",
    "            #'tardbp-293flp-1-20181129-ju',\n",
    "       'tardbp-q331k-egfp-hek293-no-uv-1-20181001-ju',\n",
    "       'tardbp-g335a-egfp-hek293-4-20181001-ju',\n",
    "       'tardbp-316del346-egfp-hek293-hd2-5-on-bead-rt-1-20180829-ju',\n",
    "       'tardbp-q331k-egfp-hek293-3-20181001-ju',\n",
    "       'tardbp-egfp-hek293-hd1-6-4-20180829-ju',\n",
    "       'tardbp-egfp-hek293-4-20181001-ju',\n",
    "       'tardbp-m337p-egfp-hek293-4-20181001-ju',\n",
    "       'tardbp-egfp-293flp-m2-20171010-ju',\n",
    "       'tardbp-316del346-egfp-293flp-l1-20171010-ju',\n",
    "       'tardbp-316del346-egfp-293flp-l2-20171010-ju',\n",
    "       #'tardbp-293flp-2-20181129-ju', \n",
    "       'tardbp-egfp-293flp-l2-20171010-ju',\n",
    "       'tardbp-316del346-egfp-293flp-m1-20171010-ju',\n",
    "       'tardbp-egfp-293flp-l1-20171010-ju',\n",
    "       'tardbp-316del346-egfp-293flp-m2-20171010-ju',\n",
    "       'tardbp-egfp-293flp-m1-20171010-ju', \n",
    "            #'tardbp-293flp-1-20181129-ju',\n",
    "       'len', 'overlap_g1', 'overlap_g2', 'overlap_g3', 'overlap_g4',\n",
    "       'total_overlap', 'overlap_g1_pseudoc', 'overlap_g2_pseudoc',\n",
    "       'overlap_g3_pseudoc', 'overlap_g4_pseudoc', 'length_groups',\n",
    "       'g1plusg2byg4_ratio', 'overlap_group', 'g4byg1g2g3_ratio'],\n",
    "               how='left')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 111,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>name</th>\n",
       "      <th>score</th>\n",
       "      <th>strand</th>\n",
       "      <th>region</th>\n",
       "      <th>biotype</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>...</th>\n",
       "      <th>overlap_g1_pseudoc</th>\n",
       "      <th>overlap_g2_pseudoc</th>\n",
       "      <th>overlap_g3_pseudoc</th>\n",
       "      <th>overlap_g4_pseudoc</th>\n",
       "      <th>length_groups</th>\n",
       "      <th>g1plusg2byg4_ratio</th>\n",
       "      <th>overlap_group</th>\n",
       "      <th>g4byg1g2g3_ratio</th>\n",
       "      <th>ratio_group_x</th>\n",
       "      <th>ratio_group_y</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GL000194.1</td>\n",
       "      <td>75522</td>\n",
       "      <td>75545</td>\n",
       "      <td>.</td>\n",
       "      <td>22.0</td>\n",
       "      <td>-</td>\n",
       "      <td>intron</td>\n",
       "      <td>\"pre-mRNA\"</td>\n",
       "      <td>\"ENSG00000274847.1\"</td>\n",
       "      <td>\"AC145212.1\"</td>\n",
       "      <td>...</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>101.000000</td>\n",
       "      <td>3len30</td>\n",
       "      <td>0.019802</td>\n",
       "      <td>dense</td>\n",
       "      <td>33.666667</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GL000194.1</td>\n",
       "      <td>81461</td>\n",
       "      <td>81479</td>\n",
       "      <td>.</td>\n",
       "      <td>13.0</td>\n",
       "      <td>-</td>\n",
       "      <td>intron</td>\n",
       "      <td>\"pre-mRNA\"</td>\n",
       "      <td>\"ENSG00000274847.1\"</td>\n",
       "      <td>\"AC145212.1\"</td>\n",
       "      <td>...</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.0</td>\n",
       "      <td>23.222222</td>\n",
       "      <td>34.333333</td>\n",
       "      <td>3len30</td>\n",
       "      <td>0.058252</td>\n",
       "      <td>medium</td>\n",
       "      <td>1.361233</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GL000194.1</td>\n",
       "      <td>92228</td>\n",
       "      <td>92285</td>\n",
       "      <td>.</td>\n",
       "      <td>68.0</td>\n",
       "      <td>-</td>\n",
       "      <td>intron</td>\n",
       "      <td>\"pre-mRNA\"</td>\n",
       "      <td>\"ENSG00000274847.1\"</td>\n",
       "      <td>\"AC145212.1\"</td>\n",
       "      <td>...</td>\n",
       "      <td>99.245614</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>2len30-60</td>\n",
       "      <td>100.245614</td>\n",
       "      <td>dense</td>\n",
       "      <td>0.009877</td>\n",
       "      <td>UG</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GL000194.1</td>\n",
       "      <td>99578</td>\n",
       "      <td>99599</td>\n",
       "      <td>.</td>\n",
       "      <td>23.0</td>\n",
       "      <td>-</td>\n",
       "      <td>intron</td>\n",
       "      <td>\"pre-mRNA\"</td>\n",
       "      <td>\"ENSG00000274847.1\"</td>\n",
       "      <td>\"AC145212.1\"</td>\n",
       "      <td>...</td>\n",
       "      <td>48.619048</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>3len30</td>\n",
       "      <td>49.619048</td>\n",
       "      <td>sparse</td>\n",
       "      <td>0.019755</td>\n",
       "      <td>UG</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GL000008.2</td>\n",
       "      <td>154464</td>\n",
       "      <td>154475</td>\n",
       "      <td>.</td>\n",
       "      <td>113.0</td>\n",
       "      <td>+</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>\"\"</td>\n",
       "      <td>\".\"</td>\n",
       "      <td>\"None\"</td>\n",
       "      <td>...</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>101.000000</td>\n",
       "      <td>3len30</td>\n",
       "      <td>0.019802</td>\n",
       "      <td>dense</td>\n",
       "      <td>33.666667</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AA</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 65 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        chrom   start     end name  score strand      region     biotype  \\\n",
       "0  GL000194.1   75522   75545    .   22.0      -      intron  \"pre-mRNA\"   \n",
       "1  GL000194.1   81461   81479    .   13.0      -      intron  \"pre-mRNA\"   \n",
       "2  GL000194.1   92228   92285    .   68.0      -      intron  \"pre-mRNA\"   \n",
       "3  GL000194.1   99578   99599    .   23.0      -      intron  \"pre-mRNA\"   \n",
       "4  GL000008.2  154464  154475    .  113.0      +  intergenic          \"\"   \n",
       "\n",
       "               gene_id     gene_name  ...  overlap_g1_pseudoc  \\\n",
       "0  \"ENSG00000274847.1\"  \"AC145212.1\"  ...            1.000000   \n",
       "1  \"ENSG00000274847.1\"  \"AC145212.1\"  ...            1.000000   \n",
       "2  \"ENSG00000274847.1\"  \"AC145212.1\"  ...           99.245614   \n",
       "3  \"ENSG00000274847.1\"  \"AC145212.1\"  ...           48.619048   \n",
       "4                  \".\"        \"None\"  ...            1.000000   \n",
       "\n",
       "   overlap_g2_pseudoc  overlap_g3_pseudoc  overlap_g4_pseudoc  length_groups  \\\n",
       "0                 1.0            1.000000          101.000000         3len30   \n",
       "1                 1.0           23.222222           34.333333         3len30   \n",
       "2                 1.0            1.000000            1.000000      2len30-60   \n",
       "3                 1.0            1.000000            1.000000         3len30   \n",
       "4                 1.0            1.000000          101.000000         3len30   \n",
       "\n",
       "   g1plusg2byg4_ratio  overlap_group  g4byg1g2g3_ratio  ratio_group_x  \\\n",
       "0            0.019802          dense         33.666667            NaN   \n",
       "1            0.058252         medium          1.361233            NaN   \n",
       "2          100.245614          dense          0.009877             UG   \n",
       "3           49.619048         sparse          0.019755             UG   \n",
       "4            0.019802          dense         33.666667            NaN   \n",
       "\n",
       "   ratio_group_y  \n",
       "0             AA  \n",
       "1            NaN  \n",
       "2            NaN  \n",
       "3            NaN  \n",
       "4             AA  \n",
       "\n",
       "[5 rows x 65 columns]"
      ]
     },
     "execution_count": 111,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_2.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 112,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_2.loc[df_2.ratio_group_x.isna(),'ratio_group_x'] = df_2.ratio_group_y"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 113,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_2.loc[df_2.ratio_group_x.isna(),'ratio_group_x'] = 'UA'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 114,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "length_groups  overlap_group  ratio_group_x\n",
       "0len100plus    sparse         AA               1500\n",
       "                              UA                951\n",
       "                              UG               1500\n",
       "               medium         AA               1500\n",
       "                              UA                949\n",
       "                              UG               1500\n",
       "               dense          AA               1500\n",
       "                              UA                950\n",
       "                              UG               1500\n",
       "1len60-100     sparse         AA               3000\n",
       "                              UA               2375\n",
       "                              UG               3000\n",
       "               medium         AA               3000\n",
       "                              UA               2352\n",
       "                              UG               3000\n",
       "               dense          AA               3000\n",
       "                              UA               2329\n",
       "                              UG               3000\n",
       "2len30-60      sparse         AA               3000\n",
       "                              UA               7731\n",
       "                              UG               3000\n",
       "               medium         AA               3000\n",
       "                              UA               7768\n",
       "                              UG               3000\n",
       "               dense          AA               3000\n",
       "                              UA               7629\n",
       "                              UG               3000\n",
       "3len30         sparse         AA               3000\n",
       "                              UA               6768\n",
       "                              UG               3000\n",
       "               medium         AA               3000\n",
       "                              UA               6781\n",
       "                              UG               3000\n",
       "               dense          AA               3000\n",
       "                              UA               6749\n",
       "                              UG               3000\n",
       "dtype: int64"
      ]
     },
     "execution_count": 114,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_2.groupby(['length_groups', 'overlap_group', 'ratio_group_x']).size()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 115,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "ratio_group_x  length_groups  overlap_group\n",
       "AA             0len100plus    sparse           1500\n",
       "                              medium           1500\n",
       "                              dense            1500\n",
       "               1len60-100     sparse           3000\n",
       "                              medium           3000\n",
       "                              dense            3000\n",
       "               2len30-60      sparse           3000\n",
       "                              medium           3000\n",
       "                              dense            3000\n",
       "               3len30         sparse           3000\n",
       "                              medium           3000\n",
       "                              dense            3000\n",
       "UA             0len100plus    sparse            951\n",
       "                              medium            949\n",
       "                              dense             950\n",
       "               1len60-100     sparse           2375\n",
       "                              medium           2352\n",
       "                              dense            2329\n",
       "               2len30-60      sparse           7731\n",
       "                              medium           7768\n",
       "                              dense            7629\n",
       "               3len30         sparse           6768\n",
       "                              medium           6781\n",
       "                              dense            6749\n",
       "UG             0len100plus    sparse           1500\n",
       "                              medium           1500\n",
       "                              dense            1500\n",
       "               1len60-100     sparse           3000\n",
       "                              medium           3000\n",
       "                              dense            3000\n",
       "               2len30-60      sparse           3000\n",
       "                              medium           3000\n",
       "                              dense            3000\n",
       "               3len30         sparse           3000\n",
       "                              medium           3000\n",
       "                              dense            3000\n",
       "dtype: int64"
      ]
     },
     "execution_count": 115,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_2.groupby(['ratio_group_x', 'length_groups', 'overlap_group',]).size()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 116,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>name</th>\n",
       "      <th>score</th>\n",
       "      <th>strand</th>\n",
       "      <th>region</th>\n",
       "      <th>biotype</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>...</th>\n",
       "      <th>overlap_g1_pseudoc</th>\n",
       "      <th>overlap_g2_pseudoc</th>\n",
       "      <th>overlap_g3_pseudoc</th>\n",
       "      <th>overlap_g4_pseudoc</th>\n",
       "      <th>length_groups</th>\n",
       "      <th>g1plusg2byg4_ratio</th>\n",
       "      <th>overlap_group</th>\n",
       "      <th>g4byg1g2g3_ratio</th>\n",
       "      <th>ratio_group_x</th>\n",
       "      <th>ratio_group_y</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GL000194.1</td>\n",
       "      <td>75522</td>\n",
       "      <td>75545</td>\n",
       "      <td>.</td>\n",
       "      <td>22.0</td>\n",
       "      <td>-</td>\n",
       "      <td>intron</td>\n",
       "      <td>\"pre-mRNA\"</td>\n",
       "      <td>\"ENSG00000274847.1\"</td>\n",
       "      <td>\"AC145212.1\"</td>\n",
       "      <td>...</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>101.000000</td>\n",
       "      <td>3len30</td>\n",
       "      <td>0.019802</td>\n",
       "      <td>dense</td>\n",
       "      <td>33.666667</td>\n",
       "      <td>AA</td>\n",
       "      <td>AA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GL000194.1</td>\n",
       "      <td>81461</td>\n",
       "      <td>81479</td>\n",
       "      <td>.</td>\n",
       "      <td>13.0</td>\n",
       "      <td>-</td>\n",
       "      <td>intron</td>\n",
       "      <td>\"pre-mRNA\"</td>\n",
       "      <td>\"ENSG00000274847.1\"</td>\n",
       "      <td>\"AC145212.1\"</td>\n",
       "      <td>...</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.0</td>\n",
       "      <td>23.222222</td>\n",
       "      <td>34.333333</td>\n",
       "      <td>3len30</td>\n",
       "      <td>0.058252</td>\n",
       "      <td>medium</td>\n",
       "      <td>1.361233</td>\n",
       "      <td>UA</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GL000194.1</td>\n",
       "      <td>92228</td>\n",
       "      <td>92285</td>\n",
       "      <td>.</td>\n",
       "      <td>68.0</td>\n",
       "      <td>-</td>\n",
       "      <td>intron</td>\n",
       "      <td>\"pre-mRNA\"</td>\n",
       "      <td>\"ENSG00000274847.1\"</td>\n",
       "      <td>\"AC145212.1\"</td>\n",
       "      <td>...</td>\n",
       "      <td>99.245614</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>2len30-60</td>\n",
       "      <td>100.245614</td>\n",
       "      <td>dense</td>\n",
       "      <td>0.009877</td>\n",
       "      <td>UG</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GL000194.1</td>\n",
       "      <td>99578</td>\n",
       "      <td>99599</td>\n",
       "      <td>.</td>\n",
       "      <td>23.0</td>\n",
       "      <td>-</td>\n",
       "      <td>intron</td>\n",
       "      <td>\"pre-mRNA\"</td>\n",
       "      <td>\"ENSG00000274847.1\"</td>\n",
       "      <td>\"AC145212.1\"</td>\n",
       "      <td>...</td>\n",
       "      <td>48.619048</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>3len30</td>\n",
       "      <td>49.619048</td>\n",
       "      <td>sparse</td>\n",
       "      <td>0.019755</td>\n",
       "      <td>UG</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GL000008.2</td>\n",
       "      <td>154464</td>\n",
       "      <td>154475</td>\n",
       "      <td>.</td>\n",
       "      <td>113.0</td>\n",
       "      <td>+</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>\"\"</td>\n",
       "      <td>\".\"</td>\n",
       "      <td>\"None\"</td>\n",
       "      <td>...</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>101.000000</td>\n",
       "      <td>3len30</td>\n",
       "      <td>0.019802</td>\n",
       "      <td>dense</td>\n",
       "      <td>33.666667</td>\n",
       "      <td>AA</td>\n",
       "      <td>AA</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 65 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        chrom   start     end name  score strand      region     biotype  \\\n",
       "0  GL000194.1   75522   75545    .   22.0      -      intron  \"pre-mRNA\"   \n",
       "1  GL000194.1   81461   81479    .   13.0      -      intron  \"pre-mRNA\"   \n",
       "2  GL000194.1   92228   92285    .   68.0      -      intron  \"pre-mRNA\"   \n",
       "3  GL000194.1   99578   99599    .   23.0      -      intron  \"pre-mRNA\"   \n",
       "4  GL000008.2  154464  154475    .  113.0      +  intergenic          \"\"   \n",
       "\n",
       "               gene_id     gene_name  ...  overlap_g1_pseudoc  \\\n",
       "0  \"ENSG00000274847.1\"  \"AC145212.1\"  ...            1.000000   \n",
       "1  \"ENSG00000274847.1\"  \"AC145212.1\"  ...            1.000000   \n",
       "2  \"ENSG00000274847.1\"  \"AC145212.1\"  ...           99.245614   \n",
       "3  \"ENSG00000274847.1\"  \"AC145212.1\"  ...           48.619048   \n",
       "4                  \".\"        \"None\"  ...            1.000000   \n",
       "\n",
       "   overlap_g2_pseudoc  overlap_g3_pseudoc  overlap_g4_pseudoc  length_groups  \\\n",
       "0                 1.0            1.000000          101.000000         3len30   \n",
       "1                 1.0           23.222222           34.333333         3len30   \n",
       "2                 1.0            1.000000            1.000000      2len30-60   \n",
       "3                 1.0            1.000000            1.000000         3len30   \n",
       "4                 1.0            1.000000          101.000000         3len30   \n",
       "\n",
       "   g1plusg2byg4_ratio  overlap_group  g4byg1g2g3_ratio  ratio_group_x  \\\n",
       "0            0.019802          dense         33.666667             AA   \n",
       "1            0.058252         medium          1.361233             UA   \n",
       "2          100.245614          dense          0.009877             UG   \n",
       "3           49.619048         sparse          0.019755             UG   \n",
       "4            0.019802          dense         33.666667             AA   \n",
       "\n",
       "   ratio_group_y  \n",
       "0             AA  \n",
       "1            NaN  \n",
       "2            NaN  \n",
       "3            NaN  \n",
       "4             AA  \n",
       "\n",
       "[5 rows x 65 columns]"
      ]
     },
     "execution_count": 116,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_2.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 117,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['chrom', 'start', 'end', 'name', 'score', 'strand', 'region', 'biotype',\n",
       "       'gene_id', 'gene_name',\n",
       "       'tardbp-316del346-egfp-hek293-hd1-6-4-20180829-ju',\n",
       "       'tardbp-egfp-hek293-hd2-5-3-20180829-ju',\n",
       "       'tardbp-egfp-hek293-hd2-5-on-bead-rt-1-20180829-ju',\n",
       "       'tardbp-316del346-egfp-hek293-hd2-5-3-20180829-ju',\n",
       "       'tardbp-egfp-hek293-3-20181001-ju',\n",
       "       'tardbp-q331k-egfp-hek293-no-uv-1-20181001-ju',\n",
       "       'tardbp-320del414-egfp-hek293-hd1-6-on-bead-rt-1-20180829-ju',\n",
       "       'tardbp-316del346-egfp-hek293-4-20181001-ju',\n",
       "       'tardbp-egfp-hek293-no-uv-1-20180829-ju',\n",
       "       'tardbp-egfp-hek293-4-20181001-ju',\n",
       "       'tardbp-egfp-hek293-hd1-6-4-20180829-ju',\n",
       "       'tardbp-a326p-egfp-hek293-4-20181001-ju',\n",
       "       'tardbp-egfp-hek293-hd1-6-3-20180829-ju',\n",
       "       'tardbp-320del414-egfp-hek293-hd1-6-4-20180829-ju',\n",
       "       'tardbp-q331k-egfp-hek293-3-20181001-ju',\n",
       "       'tardbp-316del346-egfp-hek293-hd1-6-3-20180829-ju',\n",
       "       'tardbp-320del414-egfp-hek293-hd1-6-3-20180829-ju',\n",
       "       'tardbp-egfp-hek293-hd1-6-on-bead-rt-1-20180829-ju',\n",
       "       'tardbp-316del346-egfp-hek293-hd1-6-on-bead-rt-1-20180829-ju',\n",
       "       'tardbp-m337p-egfp-hek293-4-20181001-ju',\n",
       "       'tardbp-m337p-egfp-hek293-3-20181001-ju',\n",
       "       'tardbp-a326p-egfp-hek293-3-20181001-ju',\n",
       "       'tardbp-316del346-egfp-hek293-hd2-5-4-20180829-ju',\n",
       "       'tardbp-q331k-egfp-hek293-4-20181001-ju',\n",
       "       'tardbp-g335a-egfp-hek293-3-20181001-ju',\n",
       "       'tardbp-g335a-egfp-hek293-4-20181001-ju',\n",
       "       'tardbp-g294a-egfp-hek293-3-20181001-ju',\n",
       "       'tardbp-316del346-egfp-hek293-hd2-5-on-bead-rt-1-20180829-ju',\n",
       "       'tardbp-316del346-egfp-hek293-3-20181001-ju',\n",
       "       'tardbp-egfp-hek293-hd2-5-4-20180829-ju',\n",
       "       'tardbp-g294a-egfp-hek293-4-20181001-ju',\n",
       "       'tardbp-egfp-293flp-l2-20171010-ju',\n",
       "       'tardbp-egfp-293flp-l1-20171010-ju',\n",
       "       'tardbp-316del346-egfp-293flp-l1-20171010-ju',\n",
       "       'tardbp-316del346-egfp-293flp-l2-20171010-ju',\n",
       "       'tardbp-egfp-293flp-m1-20171010-ju',\n",
       "       'tardbp-316del346-egfp-293flp-m2-20171010-ju',\n",
       "       'tardbp-316del346-egfp-293flp-m1-20171010-ju',\n",
       "       'tardbp-egfp-293flp-m2-20171010-ju', 'len', 'overlap_g1', 'overlap_g2',\n",
       "       'overlap_g3', 'overlap_g4', 'total_overlap', 'overlap_g1_pseudoc',\n",
       "       'overlap_g2_pseudoc', 'overlap_g3_pseudoc', 'overlap_g4_pseudoc',\n",
       "       'length_groups', 'g1plusg2byg4_ratio', 'overlap_group',\n",
       "       'g4byg1g2g3_ratio', 'ratio_group_x', 'ratio_group_y'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 117,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_2.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 118,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "del df_2['ratio_group_y']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_ug_100plus_dense = df_2[(df_2.ratio_group_x == 'UG') & (df_2.length_groups == '0len100plus') & (df_2.overlap_group == 'dense')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_ug_100plus_medium = df_2[(df_2.ratio_group_x == 'UG') & (df_2.length_groups == '0len100plus') & (df_2.overlap_group == 'medium')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_ua_0len100plus_dense = df_2[(df_2.ratio_group_x == 'UA') & (df_2.length_groups == '0len100plus') & (df_2.overlap_group == 'dense')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_ug_3len30_dense = df_2[(df_2.ratio_group_x == 'UG') & (df_2.length_groups == '3len30') & (df_2.overlap_group == 'dense')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_ug_len60to100_dense = df_2[(df_2.ratio_group_x == 'UG') & (df_2.length_groups == '1len60-100') & (df_2.overlap_group == 'dense')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_ug_len30to60_dense = df_2[(df_2.ratio_group_x == 'UG') & (df_2.length_groups == '2len30-60') & (df_2.overlap_group == 'dense')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "row_order = pd.read_excel('/home/aram/Dropbox (UCL-MN Team)/aram/martina_tdp43/reanalysis_all_samples/row_order.xlsx')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "row_order"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 119,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_2_all_samples_rfiltered = pd.read_csv('/home/aram/Dropbox (The Francis Crick)/ucl/aram/martina_tdp43/reanalysis_all_samples/df_2_all_samples_rfiltered_unnormalised.tsv', sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 120,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>name</th>\n",
       "      <th>score</th>\n",
       "      <th>strand</th>\n",
       "      <th>region</th>\n",
       "      <th>biotype</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>...</th>\n",
       "      <th>total_overlap</th>\n",
       "      <th>overlap_g1_pseudoc</th>\n",
       "      <th>overlap_g2_pseudoc</th>\n",
       "      <th>overlap_g3_pseudoc</th>\n",
       "      <th>overlap_g4_pseudoc</th>\n",
       "      <th>length_groups</th>\n",
       "      <th>g1plusg2byg4_ratio</th>\n",
       "      <th>overlap_group</th>\n",
       "      <th>g4byg1g2g3_ratio</th>\n",
       "      <th>ratio_group_x</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GL000194.1</td>\n",
       "      <td>75522</td>\n",
       "      <td>75545</td>\n",
       "      <td>.</td>\n",
       "      <td>22.0</td>\n",
       "      <td>-</td>\n",
       "      <td>intron</td>\n",
       "      <td>\"pre-mRNA\"</td>\n",
       "      <td>\"ENSG00000274847.1\"</td>\n",
       "      <td>\"AC145212.1\"</td>\n",
       "      <td>...</td>\n",
       "      <td>100.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>101.000000</td>\n",
       "      <td>3len30</td>\n",
       "      <td>0.019802</td>\n",
       "      <td>dense</td>\n",
       "      <td>33.666667</td>\n",
       "      <td>AA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GL000194.1</td>\n",
       "      <td>81461</td>\n",
       "      <td>81479</td>\n",
       "      <td>.</td>\n",
       "      <td>13.0</td>\n",
       "      <td>-</td>\n",
       "      <td>intron</td>\n",
       "      <td>\"pre-mRNA\"</td>\n",
       "      <td>\"ENSG00000274847.1\"</td>\n",
       "      <td>\"AC145212.1\"</td>\n",
       "      <td>...</td>\n",
       "      <td>55.555556</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.0</td>\n",
       "      <td>23.222222</td>\n",
       "      <td>34.333333</td>\n",
       "      <td>3len30</td>\n",
       "      <td>0.058252</td>\n",
       "      <td>medium</td>\n",
       "      <td>1.361233</td>\n",
       "      <td>UA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GL000194.1</td>\n",
       "      <td>92228</td>\n",
       "      <td>92285</td>\n",
       "      <td>.</td>\n",
       "      <td>68.0</td>\n",
       "      <td>-</td>\n",
       "      <td>intron</td>\n",
       "      <td>\"pre-mRNA\"</td>\n",
       "      <td>\"ENSG00000274847.1\"</td>\n",
       "      <td>\"AC145212.1\"</td>\n",
       "      <td>...</td>\n",
       "      <td>98.245614</td>\n",
       "      <td>99.245614</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>2len30-60</td>\n",
       "      <td>100.245614</td>\n",
       "      <td>dense</td>\n",
       "      <td>0.009877</td>\n",
       "      <td>UG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GL000194.1</td>\n",
       "      <td>99578</td>\n",
       "      <td>99599</td>\n",
       "      <td>.</td>\n",
       "      <td>23.0</td>\n",
       "      <td>-</td>\n",
       "      <td>intron</td>\n",
       "      <td>\"pre-mRNA\"</td>\n",
       "      <td>\"ENSG00000274847.1\"</td>\n",
       "      <td>\"AC145212.1\"</td>\n",
       "      <td>...</td>\n",
       "      <td>47.619048</td>\n",
       "      <td>48.619048</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>3len30</td>\n",
       "      <td>49.619048</td>\n",
       "      <td>sparse</td>\n",
       "      <td>0.019755</td>\n",
       "      <td>UG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GL000008.2</td>\n",
       "      <td>154464</td>\n",
       "      <td>154475</td>\n",
       "      <td>.</td>\n",
       "      <td>113.0</td>\n",
       "      <td>+</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>\"\"</td>\n",
       "      <td>\".\"</td>\n",
       "      <td>\"None\"</td>\n",
       "      <td>...</td>\n",
       "      <td>100.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>101.000000</td>\n",
       "      <td>3len30</td>\n",
       "      <td>0.019802</td>\n",
       "      <td>dense</td>\n",
       "      <td>33.666667</td>\n",
       "      <td>AA</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 64 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        chrom   start     end name  score strand      region     biotype  \\\n",
       "0  GL000194.1   75522   75545    .   22.0      -      intron  \"pre-mRNA\"   \n",
       "1  GL000194.1   81461   81479    .   13.0      -      intron  \"pre-mRNA\"   \n",
       "2  GL000194.1   92228   92285    .   68.0      -      intron  \"pre-mRNA\"   \n",
       "3  GL000194.1   99578   99599    .   23.0      -      intron  \"pre-mRNA\"   \n",
       "4  GL000008.2  154464  154475    .  113.0      +  intergenic          \"\"   \n",
       "\n",
       "               gene_id     gene_name  ...  total_overlap  overlap_g1_pseudoc  \\\n",
       "0  \"ENSG00000274847.1\"  \"AC145212.1\"  ...     100.000000            1.000000   \n",
       "1  \"ENSG00000274847.1\"  \"AC145212.1\"  ...      55.555556            1.000000   \n",
       "2  \"ENSG00000274847.1\"  \"AC145212.1\"  ...      98.245614           99.245614   \n",
       "3  \"ENSG00000274847.1\"  \"AC145212.1\"  ...      47.619048           48.619048   \n",
       "4                  \".\"        \"None\"  ...     100.000000            1.000000   \n",
       "\n",
       "   overlap_g2_pseudoc  overlap_g3_pseudoc  overlap_g4_pseudoc  length_groups  \\\n",
       "0                 1.0            1.000000          101.000000         3len30   \n",
       "1                 1.0           23.222222           34.333333         3len30   \n",
       "2                 1.0            1.000000            1.000000      2len30-60   \n",
       "3                 1.0            1.000000            1.000000         3len30   \n",
       "4                 1.0            1.000000          101.000000         3len30   \n",
       "\n",
       "   g1plusg2byg4_ratio  overlap_group  g4byg1g2g3_ratio  ratio_group_x  \n",
       "0            0.019802          dense         33.666667             AA  \n",
       "1            0.058252         medium          1.361233             UA  \n",
       "2          100.245614          dense          0.009877             UG  \n",
       "3           49.619048         sparse          0.019755             UG  \n",
       "4            0.019802          dense         33.666667             AA  \n",
       "\n",
       "[5 rows x 64 columns]"
      ]
     },
     "execution_count": 120,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_2_all_samples_rfiltered.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 121,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['chrom', 'start', 'end', 'name', 'score', 'strand', 'region', 'biotype',\n",
       "       'gene_id', 'gene_name', 'tardbp-316del346-egfp-hek293-4-20181001-ju',\n",
       "       'tardbp-316del346-egfp-hek293-3-20181001-ju',\n",
       "       'tardbp-egfp-hek293-hd2-5-4-20180829-ju',\n",
       "       'tardbp-a326p-egfp-hek293-4-20181001-ju',\n",
       "       'tardbp-egfp-hek293-hd1-6-on-bead-rt-1-20180829-ju',\n",
       "       'tardbp-g294a-egfp-hek293-4-20181001-ju',\n",
       "       'tardbp-316del346-egfp-hek293-hd1-6-3-20180829-ju',\n",
       "       'tardbp-320del414-egfp-hek293-hd1-6-3-20180829-ju',\n",
       "       'tardbp-egfp-hek293-no-uv-1-20180829-ju',\n",
       "       'tardbp-egfp-hek293-hd2-5-on-bead-rt-1-20180829-ju',\n",
       "       'tardbp-egfp-hek293-hd1-6-3-20180829-ju',\n",
       "       'tardbp-316del346-egfp-hek293-hd1-6-4-20180829-ju',\n",
       "       'tardbp-316del346-egfp-hek293-hd2-5-4-20180829-ju',\n",
       "       'tardbp-a326p-egfp-hek293-3-20181001-ju',\n",
       "       'tardbp-g335a-egfp-hek293-3-20181001-ju',\n",
       "       'tardbp-g294a-egfp-hek293-3-20181001-ju',\n",
       "       'tardbp-320del414-egfp-hek293-hd1-6-on-bead-rt-1-20180829-ju',\n",
       "       'tardbp-316del346-egfp-hek293-hd1-6-on-bead-rt-1-20180829-ju',\n",
       "       'tardbp-m337p-egfp-hek293-3-20181001-ju',\n",
       "       'tardbp-q331k-egfp-hek293-4-20181001-ju',\n",
       "       'tardbp-egfp-hek293-hd2-5-3-20180829-ju',\n",
       "       'tardbp-316del346-egfp-hek293-hd2-5-3-20180829-ju',\n",
       "       'tardbp-320del414-egfp-hek293-hd1-6-4-20180829-ju',\n",
       "       'tardbp-egfp-hek293-3-20181001-ju',\n",
       "       'tardbp-q331k-egfp-hek293-no-uv-1-20181001-ju',\n",
       "       'tardbp-g335a-egfp-hek293-4-20181001-ju',\n",
       "       'tardbp-316del346-egfp-hek293-hd2-5-on-bead-rt-1-20180829-ju',\n",
       "       'tardbp-q331k-egfp-hek293-3-20181001-ju',\n",
       "       'tardbp-egfp-hek293-hd1-6-4-20180829-ju',\n",
       "       'tardbp-egfp-hek293-4-20181001-ju',\n",
       "       'tardbp-m337p-egfp-hek293-4-20181001-ju',\n",
       "       'tardbp-egfp-293flp-m2-20171010-ju',\n",
       "       'tardbp-316del346-egfp-293flp-l1-20171010-ju',\n",
       "       'tardbp-316del346-egfp-293flp-l2-20171010-ju',\n",
       "       'tardbp-egfp-293flp-l2-20171010-ju',\n",
       "       'tardbp-316del346-egfp-293flp-m1-20171010-ju',\n",
       "       'tardbp-egfp-293flp-l1-20171010-ju',\n",
       "       'tardbp-316del346-egfp-293flp-m2-20171010-ju',\n",
       "       'tardbp-egfp-293flp-m1-20171010-ju', 'len', 'overlap_g1', 'overlap_g2',\n",
       "       'overlap_g3', 'overlap_g4', 'total_overlap', 'overlap_g1_pseudoc',\n",
       "       'overlap_g2_pseudoc', 'overlap_g3_pseudoc', 'overlap_g4_pseudoc',\n",
       "       'length_groups', 'g1plusg2byg4_ratio', 'overlap_group',\n",
       "       'g4byg1g2g3_ratio', 'ratio_group_x'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 121,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_2_all_samples_rfiltered.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_2_all_samples_rfiltered[['chrom', 'start', 'end', 'name', 'score', 'strand', \n",
    "                            'length_groups', 'overlap_group', 'ratio_group_x']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_2_all_samples_rfiltered[['chrom', 'start', 'end', 'name', 'score', 'strand', \n",
    "                            'length_groups', 'overlap_group', 'ratio_group_x']].to_csv(\n",
    "    'tardbp_reference_binding_sites.bed', sep='\\t', index=None, header=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 122,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>count</th>\n",
       "      <th>mean</th>\n",
       "      <th>std</th>\n",
       "      <th>min</th>\n",
       "      <th>25%</th>\n",
       "      <th>50%</th>\n",
       "      <th>75%</th>\n",
       "      <th>max</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>length_groups</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0len100plus</th>\n",
       "      <td>11850.0</td>\n",
       "      <td>20.447389</td>\n",
       "      <td>10.386551</td>\n",
       "      <td>1.039169</td>\n",
       "      <td>13.709677</td>\n",
       "      <td>18.238994</td>\n",
       "      <td>24.236506</td>\n",
       "      <td>98.701299</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1len60-100</th>\n",
       "      <td>25056.0</td>\n",
       "      <td>30.597177</td>\n",
       "      <td>15.089638</td>\n",
       "      <td>6.060606</td>\n",
       "      <td>20.634921</td>\n",
       "      <td>26.562500</td>\n",
       "      <td>35.483871</td>\n",
       "      <td>100.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2len30-60</th>\n",
       "      <td>41128.0</td>\n",
       "      <td>40.080057</td>\n",
       "      <td>19.535335</td>\n",
       "      <td>3.030303</td>\n",
       "      <td>26.666667</td>\n",
       "      <td>35.294118</td>\n",
       "      <td>47.500000</td>\n",
       "      <td>100.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3len30</th>\n",
       "      <td>38298.0</td>\n",
       "      <td>67.672596</td>\n",
       "      <td>26.098560</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>45.833333</td>\n",
       "      <td>66.666667</td>\n",
       "      <td>100.000000</td>\n",
       "      <td>100.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                 count       mean        std       min        25%        50%  \\\n",
       "length_groups                                                                  \n",
       "0len100plus    11850.0  20.447389  10.386551  1.039169  13.709677  18.238994   \n",
       "1len60-100     25056.0  30.597177  15.089638  6.060606  20.634921  26.562500   \n",
       "2len30-60      41128.0  40.080057  19.535335  3.030303  26.666667  35.294118   \n",
       "3len30         38298.0  67.672596  26.098560  0.000000  45.833333  66.666667   \n",
       "\n",
       "                      75%         max  \n",
       "length_groups                          \n",
       "0len100plus     24.236506   98.701299  \n",
       "1len60-100      35.483871  100.000000  \n",
       "2len30-60       47.500000  100.000000  \n",
       "3len30         100.000000  100.000000  "
      ]
     },
     "execution_count": 122,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_out_rfiltered_merged30_inner.groupby(['length_groups'], observed=False).total_overlap.describe()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 123,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>count</th>\n",
       "      <th>mean</th>\n",
       "      <th>std</th>\n",
       "      <th>min</th>\n",
       "      <th>25%</th>\n",
       "      <th>50%</th>\n",
       "      <th>75%</th>\n",
       "      <th>max</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>length_groups</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0len100plus</th>\n",
       "      <td>11850.0</td>\n",
       "      <td>20.447389</td>\n",
       "      <td>10.386551</td>\n",
       "      <td>1.039169</td>\n",
       "      <td>13.709677</td>\n",
       "      <td>18.238994</td>\n",
       "      <td>24.236506</td>\n",
       "      <td>98.701299</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1len60-100</th>\n",
       "      <td>25056.0</td>\n",
       "      <td>30.597177</td>\n",
       "      <td>15.089638</td>\n",
       "      <td>6.060606</td>\n",
       "      <td>20.634921</td>\n",
       "      <td>26.562500</td>\n",
       "      <td>35.483871</td>\n",
       "      <td>100.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2len30-60</th>\n",
       "      <td>41128.0</td>\n",
       "      <td>40.080057</td>\n",
       "      <td>19.535335</td>\n",
       "      <td>3.030303</td>\n",
       "      <td>26.666667</td>\n",
       "      <td>35.294118</td>\n",
       "      <td>47.500000</td>\n",
       "      <td>100.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3len30</th>\n",
       "      <td>38298.0</td>\n",
       "      <td>67.672596</td>\n",
       "      <td>26.098560</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>45.833333</td>\n",
       "      <td>66.666667</td>\n",
       "      <td>100.000000</td>\n",
       "      <td>100.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                 count       mean        std       min        25%        50%  \\\n",
       "length_groups                                                                  \n",
       "0len100plus    11850.0  20.447389  10.386551  1.039169  13.709677  18.238994   \n",
       "1len60-100     25056.0  30.597177  15.089638  6.060606  20.634921  26.562500   \n",
       "2len30-60      41128.0  40.080057  19.535335  3.030303  26.666667  35.294118   \n",
       "3len30         38298.0  67.672596  26.098560  0.000000  45.833333  66.666667   \n",
       "\n",
       "                      75%         max  \n",
       "length_groups                          \n",
       "0len100plus     24.236506   98.701299  \n",
       "1len60-100      35.483871  100.000000  \n",
       "2len30-60       47.500000  100.000000  \n",
       "3len30         100.000000  100.000000  "
      ]
     },
     "execution_count": 123,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_2_all_samples_rfiltered.groupby(['length_groups'], observed=False).total_overlap.describe()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 125,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "df_2_all_samples_rfiltered[['chrom', 'start', 'end', 'name', 'score', 'strand', \n",
    "                            'length_groups', 'overlap_group', 'ratio_group_x']].to_csv(\n",
    "    'tardbp_reference_binding_sites_reproduced.bed', sep='\\t', index=None, header=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
